<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="MLagisz">
<meta name="dcterms.date" content="2023-06-10">

<title>Impacts and opportunities of geographically focused data integration – a case study of the Atlas of Living Australia</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="SI_methods_analyses_figures_files/libs/clipboard/clipboard.min.js"></script>
<script src="SI_methods_analyses_figures_files/libs/quarto-html/quarto.js"></script>
<script src="SI_methods_analyses_figures_files/libs/quarto-html/popper.min.js"></script>
<script src="SI_methods_analyses_figures_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="SI_methods_analyses_figures_files/libs/quarto-html/anchor.min.js"></script>
<link href="SI_methods_analyses_figures_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="SI_methods_analyses_figures_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="SI_methods_analyses_figures_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="SI_methods_analyses_figures_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="SI_methods_analyses_figures_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<script src="SI_methods_analyses_figures_files/libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="SI_methods_analyses_figures_files/libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="SI_methods_analyses_figures_files/libs/datatables-binding-0.27/datatables.js"></script>
<script src="SI_methods_analyses_figures_files/libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="SI_methods_analyses_figures_files/libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="SI_methods_analyses_figures_files/libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="SI_methods_analyses_figures_files/libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="SI_methods_analyses_figures_files/libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="SI_methods_analyses_figures_files/libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#load-and-clean-initial-dataset-for-ala-impact-assessment" id="toc-load-and-clean-initial-dataset-for-ala-impact-assessment" class="nav-link active" data-scroll-target="#load-and-clean-initial-dataset-for-ala-impact-assessment">Load and clean initial dataset for ALA impact assessment</a></li>
  <li><a href="#author-affiliations---articles." id="toc-author-affiliations---articles." class="nav-link" data-scroll-target="#author-affiliations---articles.">Author affiliations - articles.</a>
  <ul class="collapse">
  <li><a href="#openalex." id="toc-openalex." class="nav-link" data-scroll-target="#openalex.">OpenAlex.</a></li>
  <li><a href="#final-plots" id="toc-final-plots" class="nav-link" data-scroll-target="#final-plots">Final plots</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Impacts and opportunities of geographically focused data integration – a case study of the Atlas of Living Australia</h1>
<p class="subtitle lead">Supplementary Information - data and code</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>MLagisz </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">June 10, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<p><strong>Analyses details</strong></p>
<p>This documents includes complete processing code for analyses presented in the main text, as well as additional details and results. This document is also intended to be rendered into an html file, but for logistical reasons not all analyses results are shown in that document (especially collaboration networks which take long time to run and plot and thus are not automatically evaluated. Instead, relevant subsequent code chunks can be run by loading already fetched and/or preprocessed data objects saved as intermediate files).</p>
<p>The main data source used in this project is an ALA-curated impact literature collection, hosted on Zotero and exported as .csv file on 2023/03/30 into data/2023-03-30_ALA_cited.csv. Additional information was added consecutively from several other sources (OpenAlex, Unpaywall, Crossref, Altmetrics, Scimago), as described in this document.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.3 (2023-03-15)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] htmlwidgets_1.6.2 compiler_4.2.3    fastmap_1.1.1     cli_3.6.1        
 [5] tools_4.2.3       htmltools_0.5.5   rstudioapi_0.14   yaml_2.3.7       
 [9] rmarkdown_2.21    knitr_1.42        xfun_0.38         digest_0.6.31    
[13] jsonlite_1.8.4    rlang_1.1.0       evaluate_0.20    </code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo =</span> <span class="cn">TRUE</span>, <span class="at">message =</span> <span class="cn">FALSE</span>, <span class="at">warning =</span> <span class="cn">FALSE</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::install_github("massimoaria/bibliometrix") # to install the bibliometrix most recent version from GITHUB</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages('rAltmetric') #devtools::install_github("ropensci/rAltmetric") </span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>               here,</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>               rgbif,</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>               visdat, <span class="co">#for vis_miss</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>               roadoi, <span class="co">#https://cran.r-project.org/web/packages/roadoi/vignettes/intro.html</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>               corpcor,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>               naniar,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>               rvest,</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>               httr,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>               bibliometrix,</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>               rcrossref,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>               networkD3,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>               stringr,</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>               tibble,</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>               tidystringdist,</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>               synthesisr,</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>               ggplot2,</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>               RColorBrewer,</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>               rAltmetric,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>               maps,</span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>               mapproj,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>               purrr,</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>               ggthemr,</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>               igraph,</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>               circlize,</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>               patchwork,</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>               ggraph,</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>               ggridges,</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>               ggthemes)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("remotes")</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="co">#remotes::install_github("massimoaria/openalexR")</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("openalexR")</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openalexR) <span class="co">#not on CRAN</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("devtools")</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a><span class="co">#devtools::install_github("haleyjeppson/ggmosaic")</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggmosaic)</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="fu">ggthemr</span>(<span class="st">'pale'</span>) <span class="co">#select one ggplot theme to be used</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("galah")</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("remotes")</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a><span class="co">#remotes::install_github("AtlasOfLivingAustralia/galah")</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(galah) <span class="co">#for gathering record counts from Living Atlases</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'galah'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    filter</code></pre>
</div>
</div>
<section id="load-and-clean-initial-dataset-for-ala-impact-assessment" class="level3">
<h3 class="anchored" data-anchor-id="load-and-clean-initial-dataset-for-ala-impact-assessment">Load and clean initial dataset for ALA impact assessment</h3>
<p>Load original Zotero reference collection from a csv file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"2023-03-30_ALA_cited.csv"</span>))</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat) #3866 87</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#names(dat)</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Publication.Year, useNA = "always") #92 in 2023, 10 missing year (webpages)</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Publication.Year <span class="sc">!=</span> <span class="dv">2023</span>) <span class="co">#remove 2023 and records with missing year</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat) #3764 87</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#add a column in the main data frame article vs. grey</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>dat <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">article_grey =</span> <span class="fu">case_when</span>(Item.Type <span class="sc">==</span> <span class="st">"journalArticle"</span> <span class="sc">~</span> <span class="st">"article"</span>, </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>                                Item.Type <span class="sc">!=</span> <span class="st">"journalArticle"</span> <span class="sc">~</span> <span class="st">"grey"</span>)) <span class="ot">-&gt;</span> dat</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Make simple tables summarizing key selected variables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>article_grey) <span class="co">#2406 articles and 1358 grey literature</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>article_grey, dat<span class="sc">$</span>Abstract.Note<span class="sc">==</span><span class="st">""</span>) <span class="co"># only 758 of grey have abstract, 2059 articles have abstract</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>Item.Type, <span class="at">useNA =</span> <span class="st">"always"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(dat$Publication.Year) #make graph by year, below</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>Language, <span class="at">useNA =</span> <span class="st">"always"</span>) <span class="co">#mosly en_AU, needs cleaning, below</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Place, useNA = "always") #mostly empty, super messy - not usable</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Manual.Tags, useNA = "always") #often more than one tag (different types) per item - split later</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">str_starts</span>(dat<span class="sc">$</span>DOI, <span class="st">"10."</span>)) <span class="co">#2139 are proper DOI numbers (starting with 10.), other are "" or "-"</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">str_starts</span>(dat<span class="sc">$</span>DOI, <span class="st">"10."</span>), dat<span class="sc">$</span>article_grey) <span class="co">#2070 out of 2406 articles have proper DOI numbers (starting with 10.)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Clean DOI data column.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>DOI[dat<span class="sc">$</span>DOI <span class="sc">==</span> <span class="st">"-"</span>] <span class="ot">&lt;-</span> <span class="st">""</span> <span class="co">#replace with empty value for consistency</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>DOI[dat<span class="sc">$</span>DOI <span class="sc">==</span> <span class="st">"n/a"</span>] <span class="ot">&lt;-</span> <span class="st">""</span> <span class="co">#replace with empty value for consistency</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>DOI <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"https://doi.org/"</span>, <span class="st">""</span>, dat<span class="sc">$</span>DOI, <span class="at">fixed =</span> <span class="cn">TRUE</span>)  <span class="co">#replace with ""</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>DOI <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"http://dx.doi.org/"</span>, <span class="st">""</span>, dat<span class="sc">$</span>DOI, <span class="at">fixed =</span> <span class="cn">TRUE</span>)  <span class="co">#replace with ""</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(dat<span class="sc">$</span>DOI)) <span class="co">#2172 unique values</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#table(str_starts(dat$DOI, "10."), dat$Item.Type) #DOI by Item.Type</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Clean Language data column.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Language, useNA = "always") #needs cleaning, below</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"en"</span>,<span class="st">"EN"</span>, <span class="st">"ENG"</span>, <span class="st">"en_AU"</span>, <span class="st">"en_US"</span>, <span class="st">"en-AU"</span>, <span class="st">"en-GB"</span>, <span class="st">"en_GB"</span>, <span class="st">"en-IN"</span>, <span class="st">"en-us"</span>, <span class="st">"en-US"</span>, <span class="st">"eng"</span>, <span class="st">"dcterms.RFC4646; en-AU"</span>), <span class="st">"English"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"de"</span>), <span class="st">"German"</span>)</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"fr"</span>), <span class="st">"French"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"es"</span>), <span class="st">"Spanish"</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"por"</span>), <span class="st">"Portugese"</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"id"</span>), <span class="st">"Indonesian"</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"IT"</span>), <span class="st">"Italian"</span>)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Language, useNA = "always") #much better, but 1847 records have no language info</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dat<span class="sc">$</span>Language)[<span class="st">"English"</span>] <span class="co">#1902 records with English</span></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="co">#sum(dat$Language=="", na.rm  = TRUE) #1847 records without language info (extra 12 are NA)</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, dat<span class="sc">$</span>Language <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">""</span>), <span class="st">"not specified"</span>)</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Language <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Language, <span class="fu">is.na</span>(dat<span class="sc">$</span>Language), <span class="st">"not specified"</span>)</span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Language, useNA = "always") #much better, but 2003 records as "not specified"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Make a simple bar plot for languages.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>dat <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Language) <span class="ot">-&gt;</span> count_Language</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_Language, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Language, n), <span class="at">y =</span> n)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Language: 1904 out of 3764 are in English, but for 1847 records the language info is missing.</p>
<p>Place information - 3049 recorded (messily, mostly for for grey literature), but majority is missing (715).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Place, useNA = "always") #needs cleaning, below</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Place == "") #number of records that have recorded any info on the place - not very useful column</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>all_countries <span class="ot">&lt;-</span> <span class="fu">str_c</span>(<span class="fu">unique</span>(iso3166<span class="sc">$</span>sovereignty), <span class="at">collapse =</span> <span class="st">"|"</span>) <span class="co">#use iso3166 data set</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> dat <span class="co">#make a copy</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#all_countries &lt;- str_c(unique(world.cities$country.etc), collapse = "|")</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>country <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">str_extract_all</span>(df<span class="sc">$</span>Place, all_countries), toString)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(df<span class="sc">$</span>country, <span class="at">useNA =</span> <span class="st">"always"</span>) <span class="co">#needs much more cleaning</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>country <span class="ot">&lt;-</span> <span class="fu">replace</span>(df<span class="sc">$</span>country, df<span class="sc">$</span>country <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Australia, Australia"</span>), <span class="st">"Australia"</span>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co">#table(df$country=="", df$article_grey, useNA = "always") #place is recorded mostly for 509 grey lit (847 missing), 0 for articles</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Where Country is recorded, 374 are from Australia.</p>
<p>Simple bar plot for places recorded for grey literature.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(article_grey <span class="sc">==</span> <span class="st">"grey"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(country) <span class="ot">-&gt;</span> count_country</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_country, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(country, n), <span class="at">y =</span> n)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-8-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Simple bar plot for literature type.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>dat <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Item.Type) <span class="ot">-&gt;</span> count_lit_type</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_lit_type, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Item.Type, n), <span class="at">y =</span> n)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Clean string characters - standardise common foreign and formatting characters in the Author, Title and Abstract.Note columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#only use latin-ascii characters</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author <span class="ot">&lt;-</span> stringi<span class="sc">::</span><span class="fu">stri_trans_general</span>(dat<span class="sc">$</span>Author, <span class="st">'latin-ascii'</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Title <span class="ot">&lt;-</span> stringi<span class="sc">::</span><span class="fu">stri_trans_general</span>(dat<span class="sc">$</span>Title, <span class="st">'latin-ascii'</span>)</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Abstract.Note <span class="ot">&lt;-</span> stringi<span class="sc">::</span><span class="fu">stri_trans_general</span>(dat<span class="sc">$</span>Abstract.Note, <span class="st">'latin-ascii'</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co">#remove some leftover html formatting</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"&lt;i&gt;"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Title)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"&lt;/i&gt;"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Title)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Clean manual tags (Manual.Tags column contains various manually assigned tags to identify how ALA was used and what was done).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>mtags <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat<span class="sc">$</span>Manual.Tags, <span class="st">"; "</span>) </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#table(unlist(mtags))</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#keep only the 6 tags starting with numerics (main use)</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>mtags_num <span class="ot">&lt;-</span> <span class="fu">unlist</span>(mtags)[<span class="fu">str_detect</span>(<span class="fu">unlist</span>(mtags), <span class="st">"^[0-9]"</span>)]</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>count_mtags_num <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(mtags_num))</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>count_mtags_num</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             mtags_num Freq
1         1 - ALA used 1007
2        2 - ALA cited 1418
3    3 - ALA discussed  143
4 4 - ALA acknowledged   51
5    5 - ALA mentioned 1023
6    6 - ALA published   31</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#str(count_mtags_num)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot frequencies of manual tags for ALA use type.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_mtags_num, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(mtags_num, Freq), <span class="at">y =</span> Freq)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Plot manual tags counts for the categories of used ALA resources.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Non-numeric manual tags assigned only to records where ALA was used ("1 - ALA used") - they provide details of use</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>mtags_det <span class="ot">&lt;-</span> <span class="fu">unlist</span>(mtags)[<span class="fu">unlist</span>(mtags) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Species occurrence records"</span>, <span class="st">"Map"</span> , <span class="st">"Modelling"</span>, <span class="st">"Climate Data"</span>, <span class="st">"Spatial Portal"</span>, <span class="st">"Species lists"</span>, <span class="st">"Profiles"</span>)]</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>count_mtags_det <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(mtags_det))</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co">#count_mtags_det</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_mtags_det, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(mtags_det, Freq), <span class="at">y =</span> Freq)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<section id="australia-related-content." class="level4">
<h4 class="anchored" data-anchor-id="australia-related-content.">Australia-related content.</h4>
<p>Create columns indicating if Australia-related terms (e.g.&nbsp;Tasmania, Victoria, Queensland, NSW, Sydney) were mentioned in study title, abstract, or both.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#create a new logical column with TRUE when AU-relevant terms mentioned in titles </span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>AU_title <span class="ot">&lt;-</span> <span class="fu">str_detect</span>(<span class="at">string =</span> <span class="fu">str_to_lower</span>(dat<span class="sc">$</span>Title), <span class="at">pattern =</span> <span class="st">"austral|tasman|victoria|queensland|northern territiory|new south wales|nsw|vic|qld|sunshine coast|canberra|mudgee|pilbara|illawarra|sydney|melbourne|perth|adelaide|brisbane|great barrier reef|snowy mountains|murray-darling|gippsland|aboriginal|kangaroo|koala|platypus|echidna|quoll|eucalypt"</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#dat$Title[dat$AU_title == FALSE] lists that that did not match</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat$AU_title)</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co">#create a new logical column with TRUE when AU-relevant terms mentioned in abstracts </span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>AU_abs <span class="ot">&lt;-</span> <span class="fu">str_detect</span>(<span class="at">string =</span> <span class="fu">str_to_lower</span>(dat<span class="sc">$</span>Abstract.Note), <span class="at">pattern =</span> <span class="st">"austral|tasman|victoria|queensland|northern territiory|new south wales|nsw|vic|qld|sunshine coast|canberra|mudgee|pilbara|illawarra|sydney|melbourne|perth|adelaide|brisbane|great barrier reef|snowy mountains|murray-darling|gippsland|aboriginal|kangaroo|koala|platypus|echidna|quoll|eucalypt"</span>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat$AU_abs)</span></span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co">#create a new logical column with TRUE when AU-relevant terms mentioned in title or abstract </span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>AU_title_abs <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(dat<span class="sc">$</span>AU_title <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> dat<span class="sc">$</span>AU_abs <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>)</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat$AU_title_abs)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot where (column / reference field) Australian context is mentioned.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#summarise and merge for plotting for 3 new columns across all records</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>count_AU <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rbind</span>(<span class="st">"AU in title"</span> <span class="ot">=</span> <span class="fu">table</span>(dat<span class="sc">$</span>AU_title), <span class="st">"AU in abstract"</span> <span class="ot">=</span> <span class="fu">table</span>(dat<span class="sc">$</span>AU_abs), <span class="st">"AU in title or abstract"</span> <span class="ot">=</span> <span class="fu">table</span>(dat<span class="sc">$</span>AU_title_abs)))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(count_AU) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"not metioned"</span>, <span class="st">"mentioned"</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>count_AU<span class="sc">$</span>Australia <span class="ot">&lt;-</span> <span class="fu">rownames</span>(count_AU)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_AU, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Australia, mentioned), <span class="at">y =</span> mentioned)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="citizen-science-focus---by-literature-type." class="level4">
<h4 class="anchored" data-anchor-id="citizen-science-focus---by-literature-type.">Citizen science focus - by literature type.</h4>
<p>Create columns recording if citizen science (CS) or related terms (e.g.&nbsp;volunteer, crowdsourcing) were mentioned in study title, abstract, or both (cannot use “community” because it often means “ecological community” (of various species) not “human community” (general public or local residents, etc.)).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">#create a new logical column with TRUE when AU-relevant terms mentioned in title s</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>CS_title <span class="ot">&lt;-</span> <span class="fu">str_detect</span>(<span class="at">string =</span> <span class="fu">str_to_lower</span>(dat<span class="sc">$</span>Title), <span class="at">pattern =</span> <span class="st">"citizen|volunteer|crowdsourc"</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co">#dat$Title[dat$CS_title == FALSE] lists that that did not match</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat$CS_title)</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="co">#create a new logical column with TRUE when CS-relevant terms mentioned in abstracts </span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>CS_abs <span class="ot">&lt;-</span> <span class="fu">str_detect</span>(<span class="at">string =</span> <span class="fu">str_to_lower</span>(dat<span class="sc">$</span>Abstract.Note), <span class="at">pattern =</span> <span class="st">"citizen|volunteer|crowdsourc"</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat$CS_abs)</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#create a new logical column with TRUE when CS-relevant terms mentioned in titles or abstracts </span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>CS_title_abs <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(dat<span class="sc">$</span>CS_title <span class="sc">==</span> <span class="st">"TRUE"</span> <span class="sc">|</span> dat<span class="sc">$</span>CS_abs <span class="sc">==</span> <span class="st">"TRUE"</span>, <span class="st">"TRUE"</span>, <span class="st">"FALSE"</span>)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat$CS_title_abs)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot where citizen science(CS)-related terms are mentioned.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co">#summarise and merge for plotting for 3 new columns across all records</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>count_CS <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rbind</span>(<span class="st">"CS in title"</span> <span class="ot">=</span> <span class="fu">table</span>(dat<span class="sc">$</span>CS_title), <span class="st">"CS in abstract"</span> <span class="ot">=</span> <span class="fu">table</span>(dat<span class="sc">$</span>CS_abs), <span class="st">"CS in title or abstract"</span> <span class="ot">=</span> <span class="fu">table</span>(dat<span class="sc">$</span>CS_title_abs)))</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(count_CS) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"not metioned"</span>, <span class="st">"mentioned"</span>)</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>count_CS<span class="sc">$</span>CS <span class="ot">&lt;-</span> <span class="fu">rownames</span>(count_CS)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_CS, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(CS, mentioned), <span class="at">y =</span> mentioned)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"count"</span>)  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="authors" class="level4">
<h4 class="anchored" data-anchor-id="authors">Authors</h4>
<p>Process author infomation from Zotero dataset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Author column contains a mixture of different formats - try to Abbreviate author first names</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="co">#split column with author names to find most common names</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>authors_all <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unlist</span>(<span class="fu">str_split</span>(dat<span class="sc">$</span>Author, <span class="st">"; "</span>))) <span class="co">#mix of full and initialised first names</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co">#remove first word from each string (family name)</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>authors_all <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*, "</span>,<span class="st">""</span>, authors_all)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co">#extract next first word from each string (first name or initial)</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>authors_all <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">word</span>(authors_all, <span class="dv">1</span>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co">#check if string contains "." (use escape symbols!) and remove these strings</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>authors_all <span class="ot">&lt;-</span> authors_all[<span class="sc">!</span>stringr<span class="sc">::</span><span class="fu">str_detect</span>(authors_all, <span class="st">"</span><span class="sc">\\</span><span class="st">."</span>)]</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>count_authors_all <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(authors_all))</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="co">#remove numeric, comma, dot, single char, CSIRO and ID - in a new column</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>count_authors_all2 <span class="ot">&lt;-</span> count_authors_all <span class="sc">%&gt;%</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">authors_all2 =</span> <span class="fu">str_replace_all</span>(authors_all, <span class="st">".[:digit:]."</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span> <span class="co">#remove all numbers</span></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">authors_all2 =</span> <span class="fu">str_replace_all</span>(authors_all2, <span class="st">","</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">authors_all2 =</span> <span class="fu">str_replace_all</span>(authors_all2, <span class="st">"ANTONY"</span>, <span class="st">"Antony"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">authors_all2 =</span> <span class="fu">str_replace_all</span>(authors_all2, <span class="st">"&gt;"</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset</span>(<span class="fu">nchar</span>(<span class="fu">as.character</span>(authors_all2)) <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="co">#remove 1-character strings</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">subset</span>(authors_all2 <span class="sc">!=</span> <span class="st">""</span>)  <span class="sc">%&gt;%</span>  <span class="co">#remove empty strings  </span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">name_length =</span> <span class="fu">str_length</span>(authors_all2)) <span class="sc">%&gt;%</span> <span class="co">#add column with name lengths</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(name_length)) <span class="sc">%&gt;%</span> <span class="co">#arrange from the longest names to the shortest</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">authors_abbr =</span> <span class="fu">str_replace_all</span>(authors_all2, <span class="st">"[:lower:]"</span>, <span class="st">""</span>)) <span class="sc">%&gt;%</span> <span class="co">#remove all lowercase characters and save in a new column (abbreviated names)</span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">authors_abbr =</span> <span class="fu">str_replace_all</span>(authors_abbr, <span class="st">"-"</span>, <span class="st">""</span>)) <span class="co">#remove hyphens</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a>count_authors_all2 <span class="ot">&lt;-</span> count_authors_all2[<span class="sc">!</span>count_authors_all2<span class="sc">$</span>authors_all <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Foundation"</span>, <span class="st">"Australian"</span>, <span class="st">"Electrical"</span>, <span class="st">"Foundation"</span>, <span class="st">"Department"</span>, <span class="st">"Terrestrial"</span>, <span class="st">"Tecnologico"</span>, <span class="st">"Environmental"</span>, <span class="st">"GrrlScientist"</span>, <span class="st">"Contributing"</span>, <span class="st">"EcoCommons"</span>, <span class="st">"University"</span>, <span class="st">"Computing"</span>, <span class="st">"Research"</span>, <span class="st">"Defence"</span>, <span class="st">"EPNRM"</span>, <span class="st">"SANBI"</span>, <span class="st">"ANPC"</span>, <span class="st">"EPPO"</span>, <span class="st">"OGTR"</span>, <span class="st">"TERN"</span>, <span class="st">"GSN"</span>, <span class="st">"NSW"</span>, <span class="st">"OEH"</span>, <span class="st">"UNSW"</span>, <span class="st">"CSIRO"</span>, <span class="st">"this"</span>, <span class="st">"Water"</span>, <span class="st">"Olkola"</span>, <span class="st">"et"</span>, <span class="st">"an"</span>, <span class="st">"the"</span>, <span class="st">"The"</span>, <span class="st">"Joint"</span>, <span class="st">"address=50"</span>, <span class="st">"Queensland"</span>, <span class="st">"Biodiversity"</span>), ] <span class="co">#remove names that are not personal names</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a><span class="co">#authors_all_abbr &lt;- setNames(object = count_authors_all2$authors_abbr, count_authors_all2$authors_all) #create a named vector with patterns to be replaced to abbreviate names</span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>authors_all_abbr <span class="ot">&lt;-</span> <span class="fu">setNames</span>(<span class="at">object =</span> <span class="fu">paste</span>(<span class="st">","</span>, count_authors_all2<span class="sc">$</span>authors_abbr), <span class="fu">paste</span>(<span class="st">","</span>, count_authors_all2<span class="sc">$</span>authors_all)) <span class="co">#create a named vector with patterns to be replaced to abbreviate names, but obly for words following "," if more than one full first name, only first one will be abbrevaietd (otherwise the family names get soetimes messed up)</span></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(dat<span class="sc">$</span>Author, authors_all_abbr) <span class="co">#replace authors first names in the main data frame (as new column) with initials using the table above </span></span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a><span class="co">#tidy up column with abbreviated first names (exclude names of groups and orgabisations)</span></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">."</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr) <span class="co">#remove dots</span></span>
<span id="cb24-35"><a href="#cb24-35" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr) <span class="co">#remove white spaces</span></span>
<span id="cb24-36"><a href="#cb24-36" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"scheme=AGLSTERMSAglsAgent"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-37"><a href="#cb24-37" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"WatershedLandcare"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-38"><a href="#cb24-38" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"AustralianGovernment-DepartmentofDefence,DefenceScience"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-39"><a href="#cb24-39" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Technology,G"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-40"><a href="#cb24-40" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CanonInformationSystemsResearchAustraliaPty,L"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-41"><a href="#cb24-41" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Griffith,University"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-42"><a href="#cb24-42" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">";ElectronicsEngineersSignalProcessing,S"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-43"><a href="#cb24-43" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Aprs;AustralianNational,University;IeeeComputerSociety;Nicta;TheComputerVision,Foundation;etal"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-44"><a href="#cb24-44" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CountryManagers,Y"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-45"><a href="#cb24-45" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Corporation,WAboriginal"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-46"><a href="#cb24-46" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NgukurrYangbalarangers"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-47"><a href="#cb24-47" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"DirectorofNationalParks"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-48"><a href="#cb24-48" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"TheInstituteof,Electrical"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-49"><a href="#cb24-49" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NativeConservationCouncil"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-50"><a href="#cb24-50" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"DepartmentofEnvironment"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-51"><a href="#cb24-51" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"TerrestrialEcosystems"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-52"><a href="#cb24-52" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CharlesSturtUniversity"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-53"><a href="#cb24-53" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NativeVegetationBranch"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-54"><a href="#cb24-54" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NSW,NConservationCouncil"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-55"><a href="#cb24-55" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"address=50MarcusClarkeSt"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-56"><a href="#cb24-56" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"EnvironmentalProtectionAuthority"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-57"><a href="#cb24-57" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"OrganisationforEconomicCooperationandDevelopment"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-58"><a href="#cb24-58" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NSWThreatenedSpeciesScientificCommittee"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-59"><a href="#cb24-59" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"TheGreatEasternRangesInitiative(GER)"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-60"><a href="#cb24-60" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"AustralianGovernmentDeptofHealth:OfficeoftheGeneTechnologyRegulator"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-61"><a href="#cb24-61" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"DepartmentofAgriculture,WaterandtheEnvironment"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-62"><a href="#cb24-62" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"RylstoneRegionCoalFreeCommunity"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-63"><a href="#cb24-63" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"SouthernHighlandNews"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-64"><a href="#cb24-64" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"TERNEco-Informatics"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-65"><a href="#cb24-65" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"EPNRMBoard"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-66"><a href="#cb24-66" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"GrrlScientist"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-67"><a href="#cb24-67" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CanadensysNetwork"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-68"><a href="#cb24-68" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"WarrnamboolCouncil"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-69"><a href="#cb24-69" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Tribune,TheNational"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-70"><a href="#cb24-70" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CityofJoondalup"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-71"><a href="#cb24-71" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NSWRoadsandMaritimeServices"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-72"><a href="#cb24-72" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"RoyalBotanicGardensBoardVictoria"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-73"><a href="#cb24-73" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"EnvironmentalProtectionAuthority"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-74"><a href="#cb24-74" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"LitoriaEcoservices"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-75"><a href="#cb24-75" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"DepartmentoftheEnvironment"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-76"><a href="#cb24-76" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"DepartmentofAgricultureandWaterResources"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-77"><a href="#cb24-77" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NorthQueenslandBulkPortsCorporation"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-78"><a href="#cb24-78" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"PhoenixEnvironmentalSciences"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-79"><a href="#cb24-79" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CSIRO"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-80"><a href="#cb24-80" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"WADPW"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-81"><a href="#cb24-81" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"SANBI"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-82"><a href="#cb24-82" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"UNSW"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-83"><a href="#cb24-83" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Queensl,DRossDwyerRossDwyerisapost-doctoralresearchfellowatTheUniversityof"</span>, <span class="st">"Dwyer,R"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-84"><a href="#cb24-84" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"fundamental,Hisaresearcheracross"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-85"><a href="#cb24-85" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"OzTrack,beenakeypartoftheteamdeveloping"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-86"><a href="#cb24-86" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ConselhoNacionaldeDesenvolvimentoCientificoe,Tecnologico"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-87"><a href="#cb24-87" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"SADepartmentforEnvironment,Water&amp;NaturalResources"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-88"><a href="#cb24-88" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NSWCatchmentManagementAuthority-NorthernRivers"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-89"><a href="#cb24-89" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"EuropeanCommission,JintResearchCentre"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-90"><a href="#cb24-90" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"AustralianNationalWildlifeCollection,NonalResearchCollectionsAustralia,,CanberraACT,A"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-91"><a href="#cb24-91" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"AustralianWildlifeConservancy,CairnsQLD4870,A"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-92"><a href="#cb24-92" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ResearchandRecoveryofEndangeredSpecies,SchoolofEarthandEnvironmentalSciences,UniversityofQueensland,StLuciaQLD4072,A"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-93"><a href="#cb24-93" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"OlkolaAboriginalCorporation(Cairns)"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-94"><a href="#cb24-94" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"corporateName=DepartmentofEducation,SandEmployment"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-95"><a href="#cb24-95" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"TheGreatEasternRangesInitiative(GER)"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-96"><a href="#cb24-96" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Linktoexternalsite,thislinkwillopeninanewwindow"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-97"><a href="#cb24-97" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"FisheriesQueensland,DepartmentofAgricultureandFisheries"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-98"><a href="#cb24-98" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CentreforEnvironmentalRiskManagementofBushfires,SchoolofEarth,AtmosphericandLifeSciences,UniversityofWollongong,WollongongNSW2522,A"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-99"><a href="#cb24-99" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"CentreforEcosystemScience,SchoolofBiologicalEarthandEnvironmentalSciences,UniversityofNewSouthWales,S,KensingtonNSW2033,A"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-100"><a href="#cb24-100" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"ResearchandRecoveryofEndangeredSpecies,SchoolofEarthandEnvironmentalSciences,UniversityofQueensland,StLuciaQLD4072,A"</span>, <span class="st">""</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-101"><a href="#cb24-101" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Ramirez-Cabral,NYakovlevaZitz"</span>, <span class="st">"Ramirez-Cabral,NYZ"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-102"><a href="#cb24-102" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Platell,GAnneMarieMarguerite"</span>, <span class="st">"Platell,GAMM"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-103"><a href="#cb24-103" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"daSilvaSantos,KCristineBezerra"</span>, <span class="st">"daSilvaSantos,KCB"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-104"><a href="#cb24-104" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Ladino,PMiguelOliveira"</span>, <span class="st">"Ladino,PMO"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-105"><a href="#cb24-105" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Blanco,CMariaRodrigues"</span>, <span class="st">"Blanco,CMR"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-106"><a href="#cb24-106" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Mora-Cross,MAuxiliadora"</span>, <span class="st">"Mora-Cross,MA"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-107"><a href="#cb24-107" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"SalazarCarrillo,JCarlos"</span>, <span class="st">"SalazarCarrillo,JC"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-108"><a href="#cb24-108" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Ezquivelzeta,ERobredo"</span>, <span class="st">"Ezquivelzeta,ER"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-109"><a href="#cb24-109" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Bickerstaff,JRonaldMunro"</span>, <span class="st">"Bickerstaff,JRM"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-110"><a href="#cb24-110" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Arias-Penna,DCarolina"</span>, <span class="st">"Arias-Penna,DC"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-111"><a href="#cb24-111" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Gerhardinger,LCavaleri"</span> , <span class="st">"Gerhardinger,LC"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-112"><a href="#cb24-112" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"AlmeydaZambrano,AMaria"</span> , <span class="st">"lmeydaZambrano,AM"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-113"><a href="#cb24-113" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Wijewardhana,UAbhisheka"</span> , <span class="st">"Wijewardhana,UA"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-114"><a href="#cb24-114" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Vasconcelos,HCristina"</span> , <span class="st">"Vasconcelos,HC"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-115"><a href="#cb24-115" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Notarnicola,RFabrizio"</span> , <span class="st">"Notarnicola,RF"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-116"><a href="#cb24-116" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_abbr <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"([[;]])</span><span class="sc">\\</span><span class="st">1+"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, dat<span class="sc">$</span>Author_abbr)</span>
<span id="cb24-117"><a href="#cb24-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-118"><a href="#cb24-118" aria-hidden="true" tabindex="-1"></a><span class="co">#readr::write_csv(dat, file = here("data","dat_all_processed.csv")) #saving intermediate version of the processed data</span></span>
<span id="cb24-119"><a href="#cb24-119" aria-hidden="true" tabindex="-1"></a><span class="co">#dat &lt;- read.csv(here("data", "dat_all_processed.csv")) #loading all data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="co-authorships" class="level4">
<h4 class="anchored" data-anchor-id="co-authorships">Co-authorships</h4>
<p>Count authors.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#all</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unlist(str_split(string = dat$Author_abbr, pattern = ";"))) # total number of authors (might be slightly biased)</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unique(unlist(str_split(string = dat$Author_abbr, pattern = ";")))) # total number of unique authors (might be slightly biased)</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Author_abbr=="", dat$article_grey) #missing authorship (either not captured in Zotero or removed organisations names)</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="co">#all - split and clean before counting again</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>authors_all <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">str_split</span>(<span class="at">string =</span> dat<span class="sc">$</span>Author_abbr, <span class="at">pattern =</span> <span class="st">";"</span>))</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(nchar(authors_all)) #check distribution of name lengths - some odd with &gt;25 characters</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co">#authors_all[nchar(authors_all) &gt; 25]</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>authors_all <span class="ot">&lt;-</span> authors_all[<span class="fu">nchar</span>(authors_all) <span class="sc">&lt;</span> <span class="dv">26</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(authors_all)] <span class="co">#remove any odd or empty long strings left</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co"># hist(nchar(authors_all)) #check distribution of name lengths   </span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co"># authors_all[nchar(authors_all) &gt; 20]</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="co"># length(authors_all) # total number of author names (should be correct)</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(authors_all)) #  total number of unique authors names</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="co">#articles</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>authors_articles <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">str_split</span>(<span class="at">string =</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>))</span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>authors_articles <span class="ot">&lt;-</span> authors_articles[<span class="fu">nchar</span>(authors_articles) <span class="sc">&lt;</span> <span class="dv">26</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(authors_articles)]</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a><span class="co"># length(authors_articles) # total number of authors</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(authors_articles)) # total number of unique authors</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a><span class="co">#grey</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>authors_grey <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">str_split</span>(<span class="at">string =</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"grey"</span>, <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>))</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>authors_grey <span class="ot">&lt;-</span> authors_grey[<span class="fu">nchar</span>(authors_grey) <span class="sc">&lt;</span> <span class="dv">26</span> <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(authors_grey)]</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a><span class="co"># length(authors_grey) # total number of authors</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(authors_grey)) # total number of unique authors</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Number of records with missing authorship information: 149 total, 3 for articles, 146 for grey literature.<br>
Total number of unikwe known author names: 9813 total, 7827 for articles, 2742 for grey literature.</p>
<p>Author overlaps bytween literature types and more processing code.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Authorship overlaps - are there same author names in articles and grey subsets?</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co"># length(intersect(unique(authors_articles), unique(authors_grey))) /  length(unique(authors_all)) *100 # % authors in both articles and grey literature</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># length(setdiff(unique(authors_grey), unique(authors_articles))) #count in grey but not in articles - calculate precentages of authors publishing both</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co"># length(setdiff(unique(authors_articles), unique(authors_grey))) #7071 in articles but not in grey literature</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Multiple papers by the same author -  Top10 most productive authors for articles</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co"># data.frame(table(authors_articles)) %&gt;% rename(Names = authors_articles, Count = Freq) %&gt;% arrange(-Count) %&gt;% slice_head(n = 10) %&gt;% </span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(aes(x = reorder(Names, Count), y = Count)) + geom_bar(stat = "identity") +</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   coord_flip() +</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   xlab("")  +</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   ylab("article count")</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Top10 most productive authors for grey</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a><span class="co"># data.frame(table(authors_grey)) %&gt;% rename(Names = authors_grey, Count = Freq) %&gt;% arrange(-Count) %&gt;% filter(Names != "") %&gt;% slice_head(n = 10) %&gt;% </span></span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(aes(x = reorder(Names, Count), y = Count)) + geom_bar(stat = "identity") +</span></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   coord_flip() +</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   xlab("")  +</span></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   ylab("article count")</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a><span class="do">## Separate author names for overall summary per paper (list of vectors format)</span></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>Author_split <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat<span class="sc">$</span>Author_abbr, <span class="st">";"</span>) <span class="co">#list of vectors of authors</span></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a><span class="co">#length(Author_split) #3774 elements (outputs) in the list</span></span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_counts <span class="ot">&lt;-</span> <span class="fu">lengths</span>(Author_split) <span class="co">#numbers of elements in the vectors</span></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a><span class="co">#View(data.frame(dat$article_grey, dat$Author, dat$Author_abbr, dat$Author_counts)) #visual checks</span></span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$Author_counts, useNA = "always")</span></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(dat$Author_counts, breaks = 50) #most have just 1 author</span></span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a><span class="co">#check for empty values</span></span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$article_grey, dat$Author_abbr == "") #146/1356\8 (10%) of grey has missing (individual) authorship, &lt;1% for articles</span></span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$article_grey, dat$Author_counts == "") #no empty values, i.e. missing ones count as 1</span></span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>Author_counts <span class="ot">&lt;-</span> <span class="fu">replace</span>(dat<span class="sc">$</span>Author_counts, dat<span class="sc">$</span>Author_abbr <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">""</span>, <span class="cn">NA</span>), <span class="dv">0</span>) <span class="co">#replace empty ones with 0</span></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat$article_grey, dat$Author_counts == "0") </span></span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a><span class="co">#dat[dat$Author_counts == 0, "Author_abbr"]</span></span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Summarise author number per output by literature type</span></span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a><span class="co"># dat %&gt;%</span></span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a><span class="co">#   group_by(article_grey) %&gt;%</span></span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a><span class="co">#   summarise(mean_AC = mean(Author_counts),</span></span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a><span class="co">#            median_AC = median(Author_counts),</span></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a><span class="co">#            sd_AC = sd(Author_counts),</span></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a><span class="co">#            IQR_AC = IQR(Author_counts),</span></span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a><span class="co">#            min_AC = min(Author_counts),</span></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a><span class="co">#            max_AC = max(Author_counts)</span></span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a><span class="co">#            ) #does not capture that much of the grey lit is by teams, but under organisational name</span></span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a><span class="do">## Papers with multiple authors </span></span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a><span class="co"># #create a dataframe of author counts and literature type</span></span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a><span class="co">#  dat %&gt;% </span></span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a><span class="co">#    group_by(article_grey) %&gt;%</span></span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a><span class="co">#    reframe(article_grey, Author_counts) %&gt;% </span></span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a><span class="co">#    drop_na() %&gt;% filter(Author_counts &gt; 50) -&gt; Author_counts_df</span></span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a><span class="co"># table(Author_counts_df$article_grey) #number of papers with &gt;50 authors</span></span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a><span class="co"># dat %&gt;% </span></span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a><span class="co">#   reframe(article_grey, Title, Author_counts) %&gt;% </span></span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(Author_counts &gt; 50) #display paper titles with &gt;50 authors</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Authors that contributed to both articles and grey literature as % of the total number of unique author names: 7.7</p>
<p>Models for historical author team size changes for articles and grey literature.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ## Author numbers per paper across years</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="co"># #create a data frame of author counts and literature type</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>dat <span class="sc">%&gt;%</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(article_grey) <span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">reframe</span>(Publication.Year, Author_counts) <span class="sc">%&gt;%</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="ot">-&gt;</span> count_year_authors</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="co"># #plot for articles</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a><span class="co"># count_year_authors %&gt;%</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(article_grey == "article") %&gt;% # articles only</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot(aes(x = Publication.Year, y = Author_counts)) +</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_point(alpha = 0.2, color = "#00cc99") +</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   ylim(0, 220) +</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_minimal() +</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme(legend.position = "none") +</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   ylab("author count") +</span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_smooth(method = lm)</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a><span class="co"># #plot for grey</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a><span class="co"># count_year_authors %&gt;%</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   filter(article_grey == "grey") %&gt;% # grey only</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   ggplot(aes(x = Publication.Year, y = Author_counts)) +</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_point(alpha = 0.2, color = "#000000") +</span></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   ylim(0, 220) +</span></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_minimal() +</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme(legend.position = "none") +</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   ylab("author count") +</span></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_smooth(method = lm)</span></span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a><span class="co"># ggsave(plot = p, here("plots", "plot.authors_peroutput_years.pdf"), width = 16, height = 6, units = "cm",  dpi = "retina", scale = 2)</span></span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a><span class="co"># #linear regression for articles with Year</span></span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>model_a <span class="ot">&lt;-</span> <span class="fu">lm</span>(Author_counts <span class="sc">~</span> <span class="fu">scale</span>(Publication.Year), <span class="at">data =</span> count_year_authors[count_year_authors<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span> <span class="sc">&amp;</span> count_year_authors<span class="sc">$</span>Author_counts <span class="sc">!=</span> <span class="dv">0</span>, ]) <span class="co">#run for articles, without 0 values (missing data)</span></span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_a) <span class="co">#slope 0.7204     0.1484   4.854 1.29e-06 ***</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Author_counts ~ scale(Publication.Year), data = count_year_authors[count_year_authors$article_grey == 
    "article" &amp; count_year_authors$Author_counts != 0, ])

Residuals:
    Min      1Q  Median      3Q     Max 
 -4.902  -2.698  -1.217   0.579 212.339 

Coefficients:
                        Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               4.9051     0.1484  33.053  &lt; 2e-16 ***
scale(Publication.Year)   0.7204     0.1484   4.854 1.29e-06 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 7.275 on 2401 degrees of freedom
Multiple R-squared:  0.009716,  Adjusted R-squared:  0.009303 
F-statistic: 23.56 on 1 and 2401 DF,  p-value: 1.291e-06</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co"># #linear regression for grey with Year</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>model_g <span class="ot">&lt;-</span> <span class="fu">lm</span>(Author_counts <span class="sc">~</span> <span class="fu">scale</span>(Publication.Year), <span class="at">data =</span> count_year_authors[count_year_authors<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"grey"</span> <span class="sc">&amp;</span> count_year_authors<span class="sc">$</span>Author_counts <span class="sc">!=</span> <span class="dv">0</span>, ]) <span class="co">#run for grey, without 0 values (missing data)</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_g) <span class="co">#slope 0.1411     0.1901   0.742    0.458</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Author_counts ~ scale(Publication.Year), data = count_year_authors[count_year_authors$article_grey == 
    "grey" &amp; count_year_authors$Author_counts != 0, ])

Residuals:
    Min      1Q  Median      3Q     Max 
 -2.078  -1.861  -1.645   0.355 208.009 

Coefficients:
                        Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               2.8556     0.1900  15.028   &lt;2e-16 ***
scale(Publication.Year)   0.1411     0.1901   0.742    0.458    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 6.615 on 1210 degrees of freedom
Multiple R-squared:  0.0004551, Adjusted R-squared:  -0.000371 
F-statistic: 0.5509 on 1 and 1210 DF,  p-value: 0.4581</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Stats for comparing number of authors per article vs. grey</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>dat <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Author_counts <span class="sc">!=</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="co">#dont use 0 values (missing data)</span></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">reframe</span>(article_grey, Author_counts) <span class="ot">-&gt;</span> Author_counts_df</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># poisson - less likely to fit to real biological data well because of overdispersion</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(Author_counts <span class="sc">~</span> article_grey, <span class="at">family =</span> <span class="st">"poisson"</span>, <span class="at">data =</span> Author_counts_df)</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = Author_counts ~ article_grey, family = "poisson", 
    data = Author_counts_df)

Deviance Residuals: 
   Min      1Q  Median      3Q     Max  
-2.152  -1.270  -0.535   0.477  37.409  

Coefficients:
                  Estimate Std. Error z value Pr(&gt;|z|)    
(Intercept)       1.590279   0.009211  172.65   &lt;2e-16 ***
article_greygrey -0.540994   0.019333  -27.98   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 14645  on 3614  degrees of freedom
Residual deviance: 13787  on 3613  degrees of freedom
AIC: 24489

Number of Fisher Scoring iterations: 5</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># #An alternative to overdispersed poisson fit - quasipoisson -&gt; same pattern</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co"># fit &lt;- glm(Author_counts ~ article_grey, family = "quasipoisson", data = Author_counts_df)</span></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="co"># #An alternative to overdispersed poisson fit -  negative binomial - more likely to fit to real biological data well -&gt; same pattern</span></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="co"># fit &lt;- MASS::glm.nb(Author_counts ~ article_grey, data = Author_counts_df)</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="co">#readr::write_csv(dat, file = here("data","dat_all_processed.csv")) #saving intermediate version of the processed data</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="co">#dat &lt;- read.csv(here("data", "dat_all_processed.csv")) #loading all data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="co-authorship-network" class="level4">
<h4 class="anchored" data-anchor-id="co-authorship-network">Co-authorship network</h4>
<p>Note: not executing the code below due to long time required to plot networks with so many nodes and connections.</p>
<p>Author networks calculations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Separate author names for overall summary per paper (list of vectors format) for article</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>Author_split <span class="ot">&lt;-</span> <span class="fu">str_split</span>(<span class="at">string =</span> dat[ , <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>) <span class="co">#list of vectors of authors</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(Author_split) <span class="co"># elements (outputs) in the list</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>df_auth_collab <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ref.id =</span> <span class="fu">rep</span>(dat[, <span class="st">"Key"</span>], </span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">lengths</span>(Author_split)),</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">auth_names =</span> <span class="fu">unlist</span>(Author_split))</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a><span class="co">#table(df_auth_collab$ref.id == "") #no empty</span></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a><span class="co">#table(df_auth_collab$auth_names == "") # empty ""</span></span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co">#remove empty nodes </span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co">#df_auth_collab &lt;- df_auth_collab[!is.na(df_auth_collab$auth_names), ]</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>df_auth_collab <span class="ot">&lt;-</span> df_auth_collab[df_auth_collab<span class="sc">$</span>auth_names <span class="sc">!=</span> <span class="st">""</span>, ]</span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unique(df_auth_collab$ref.id))</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unique(df_auth_collab$auth_names))</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>df_auth_collab <span class="sc">%&gt;%</span></span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_auth_collab, <span class="at">by =</span> <span class="st">"auth_names"</span>, <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="co">#remove many-to-many duplicates</span></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ref.id.x <span class="sc">&lt;=</span> ref.id.y) <span class="sc">%&gt;%</span> <span class="co">#use &lt;= to include unconnected outcomes (self-connections counted as 1)</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(ref.id.x, ref.id.y) <span class="sc">%&gt;%</span> </span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">graph_from_data_frame</span>(<span class="at">directed =</span> <span class="cn">FALSE</span>) <span class="ot">-&gt;</span> all_outputs_graph <span class="co">#with output ID (Key) names as vertices</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a><span class="co">#str(all_outputs_graph)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot co-authorship network for articles and grey literature.<br>
Points are items (articles/grey) and edges/connections are shared authors between published items.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># #add color attribute showing if article or grey</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>name <span class="sc">%in%</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Key"</span>], <span class="st">"article"</span>, <span class="st">"grey"</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co"># #table(V(all_outputs_graph)$color)</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>name <span class="sc">%in%</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Key"</span>], <span class="st">"#0FD19D"</span>, <span class="st">"#808080"</span>)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co"># #table(V(all_outputs_graph)$color)</span></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="co"># #plot - each vertex is an output and edges reflect shared authors</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="co"># all_outputs_graph_layout &lt;- layout_with_graphopt(all_outputs_graph) #layout_with_graphopt(all_outputs_graph) #layout_nicely(all_outputs_graph) #layout.kamada.kawai(all_outputs_graph)  # choose an appropriate graph layout algorithm for the graph automatically</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="co"># all_outputs_graph_plot &lt;- plot(all_outputs_graph, layout = all_outputs_graph_layout, vertex.label = NA, vertex.size = 1, vertex.frame.color = NA)</span></span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a><span class="co"># #, vertex.color = "grey"</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a><span class="co"># legend("left", legend=c("grey", "article"), col = levels(as.factor(V(all_outputs_graph)$color)) , bty = "n", pch = 20 , pt.cex = 3, cex = 1.5, text.col="black" , horiz = FALSE, inset = c(0.1, 0.1))</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>type <span class="ot">&lt;-</span> <span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>type <span class="co">#save as an object for easier plotting</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot graph using library(ggraph) - outcomes as nodes (single-author articles shown)</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a><span class="co">#https://www.r-bloggers.com/2020/03/ggraph-tricks-for-common-problems/</span></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>p_network <span class="ot">&lt;-</span> <span class="fu">ggraph</span>(all_outputs_graph, <span class="st">'igraph'</span>, <span class="at">algorithm =</span> <span class="st">'kk'</span>) <span class="sc">+</span>   <span class="co">#, colour = 'forestgreen'</span></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_edge_link0(aes(width = 0.01), edge_alpha = 0.1) + </span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_edge_fan</span>(<span class="at">color =</span> <span class="st">"gray50"</span>, <span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_node_point</span>(<span class="fu">aes</span>(<span class="at">linewidth =</span> <span class="fl">0.001</span>, <span class="at">fill =</span> type, <span class="at">color =</span> type, <span class="at">alpha =</span> <span class="fl">0.01</span>)) <span class="sc">+</span></span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">"article"</span>, <span class="st">"grey"</span>), <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#66c2a5"</span>, <span class="st">"#808080"</span>)) <span class="sc">+</span> <span class="co"># using directly $color for geom_node_point does not work</span></span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">alpha =</span> <span class="st">"none"</span>, <span class="at">size =</span> <span class="st">"none"</span>, <span class="at">type =</span> <span class="st">"none"</span>) <span class="sc">+</span> <span class="co">#remove unnecessary legend elements</span></span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">colour =</span> <span class="st">"type"</span>) <span class="sc">+</span> <span class="co">#change legend title</span></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="collaboration-networks." class="level4">
<h4 class="anchored" data-anchor-id="collaboration-networks.">Collaboration networks.</h4>
<p>Authors are nodes and edges/connections are items they published together.<br>
Again, not executing the plots due to the large size of the networks.</p>
<p>Collaboration networks (using igraph) - articles subset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="do">### article literature subset</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Separate author names for overall summary per paper (list of vectors format) for article</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>Author_split_article <span class="ot">&lt;-</span> <span class="fu">str_split</span>(<span class="at">string =</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>) <span class="co">#list of vectors of authors</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(Author_split_article) <span class="co">#2406 elements (outputs) in the list</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ref.id =</span> <span class="fu">rep</span>(dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Key"</span>], <span class="fu">lengths</span>(Author_split_article)),</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">auth_names =</span> <span class="fu">unlist</span>(Author_split_article))</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co">#str(df_auth_collab_article)</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="co">#df_auth_collab_article[1:20, ]</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="co">#remove empty ones </span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="ot">&lt;-</span> df_auth_collab_article[<span class="sc">!</span><span class="fu">is.na</span>(df_auth_collab_article<span class="sc">$</span>auth_names), ]</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="ot">&lt;-</span> df_auth_collab_article[df_auth_collab_article<span class="sc">$</span>auth_names <span class="sc">!=</span> <span class="st">""</span>, ]</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a><span class="co">#find 25 most productive authors</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="co"># df_auth_collab_article %&gt;% count(auth_names) %&gt;% arrange(-n) %&gt;% slice_head(n = 25)</span></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a><span class="co">#compute graph of collaborations for all authors</span></span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="sc">%&gt;%</span></span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_auth_collab_article, <span class="at">by =</span> <span class="st">"ref.id"</span>, <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="co">#remove many-to-many duplicates</span></span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(auth_names.x <span class="sc">!=</span> auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(auth_names.x, auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">graph_from_data_frame</span>(<span class="at">directed =</span> <span class="cn">FALSE</span>) <span class="ot">-&gt;</span> a1 <span class="co">#with author names as vertices</span></span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(a1) <span class="co">#IGRAPH 2b42183 UN-- 7650 166630 -- attr: name (v/c), n (e/n)</span></span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a><span class="co">#an undirected multigraph (parallel edges) with 7650 authors and 166630 scientific collaborations. Each node (author) in the network has 1 attribute: name. Each edge has 1 attributes: n (collaborations).</span></span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a><span class="co">#overall graph statistics</span></span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a><span class="fu">graph.density</span>(a1) <span class="co">#0.006 = quite sparse</span></span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a><span class="fu">transitivity</span>(a1, <span class="at">type =</span> <span class="st">"global"</span>) <span class="co"># 0.894 = probability that the neighbors of a node are also connected (also called the clustering coefficient)</span></span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a><span class="co">#see components of the igraph object</span></span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a><span class="co"># E(a1)</span></span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a><span class="co"># V(a1)</span></span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a><span class="co"># a1[]</span></span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a><span class="co"># edge_attr(a1)</span></span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a><span class="co"># vertex_attr(a1)</span></span>
<span id="cb36-39"><a href="#cb36-39" aria-hidden="true" tabindex="-1"></a><span class="co"># as_data_frame(a1, what = "edges")</span></span>
<span id="cb36-40"><a href="#cb36-40" aria-hidden="true" tabindex="-1"></a><span class="co"># get.edgelist(a1)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot article author collaboration network.<br>
Points are authors and edges/connections are shared articles they published together.<br>
(Note: not plotting due to this taking long time - the graph is very dense!)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#visualize our co-authorship graph network (https://stackoverflow.com/questions/9759484/vertex-border-color-width-in-r-graph-plot)</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>a1_graph_layout <span class="ot">&lt;-</span> <span class="fu">layout_with_graphopt</span>(a1) <span class="co">#layout_nicely(a1) #layout.kamada.kawai(a1)  # choose an appropriate graph layout algorithm for the graph automatically</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>a1_graph_plot <span class="ot">&lt;-</span> <span class="fu">plot</span>(a1, <span class="at">layout =</span> a1_graph_layout, <span class="at">vertex.label =</span> <span class="cn">NA</span>, <span class="at">vertex.size =</span> <span class="dv">1</span>, <span class="at">vertex.color =</span> <span class="st">"#66c2a5"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Collaboration network graph metrics from a saved network object - articles.<br>
(Can be run manually from a saved data object)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"data_a1.RData"</span>)) <span class="co">#load saved data object</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co">#try to simplify the graph</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>a1is <span class="ot">&lt;-</span> <span class="fu">simplify</span>(a1, <span class="at">remove.multiple =</span> T, <span class="at">remove.loops =</span> T, </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">edge.attr.comb=</span><span class="fu">list</span>(<span class="at">weight=</span><span class="st">"sum"</span>, <span class="st">"ignore"</span>) ) <span class="co">#simplify</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(a1is, layout = a1_graph_layout, edge.arrow.size=0, vertex.color="gold", vertex.size=1, vertex.frame.color="gray", </span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co">#      vertex.label.color="black", vertex.label.cex=0.8, vertex.label.dist=2, edge.curved=0.2, vertex.label = NA)</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="co"># #E(a1is)</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co"># #V(a1is)</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a><span class="do">##Find cliques (complete subgraphs of an undirected graph) - not good - picks up the 209-authors paper</span></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a><span class="co"># cliques(as.undirected(a1)) # compute list of cliques       </span></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(cliques(as.undirected(a1)), length) # clique sizes</span></span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a><span class="co"># largest_cliques(as.undirected(a1)) # clique with max number of nodes - 2010 - single paper</span></span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="co"># vcol &lt;- rep("article80", vcount(as.undirected(a1)))</span></span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="co"># vcol[unlist(largest_cliques(as.undirected(a1)))] &lt;- "gold"</span></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(as.undirected(a1), vertex.label=V(a1)$name, vertex.color=vcol)</span></span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculating more centrality measures (degree, betweeness, closeness, eigenvector centrality, coreness; see https://ona-book.org/vertex-importance.html and https://users.dimi.uniud.it/~massimo.franceschet/R/communities.html)</span></span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>metrics_article <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">type =</span> <span class="fu">rep</span>(<span class="st">"article"</span>, <span class="fu">length</span>(igraph<span class="sc">::</span><span class="fu">degree</span>(a1is))),</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">deg =</span> igraph<span class="sc">::</span><span class="fu">degree</span>(a1is),</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">bet =</span> igraph<span class="sc">::</span><span class="fu">betweenness</span>(a1is),</span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">clo =</span> igraph<span class="sc">::</span><span class="fu">closeness</span>(a1is),</span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">eig =</span> igraph<span class="sc">::</span><span class="fu">evcent</span>(a1is)<span class="sc">$</span>vector,</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">cor =</span> igraph<span class="sc">::</span><span class="fu">graph.coreness</span>(a1is)</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(metrics_article<span class="sc">$</span>deg, metrics_article<span class="sc">$</span>clo) <span class="co">#clo signif correlated with deg</span></span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(metrics_article$bet)</span></span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(metrics_article$bet, breaks = 100) # mostly extremely low values and a few high ones</span></span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a><span class="do">## Greedy community detection</span></span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a><span class="co"># greedy method (hiearchical, fast method)</span></span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a>comm_a1 <span class="ot">&lt;-</span> <span class="fu">cluster_fast_greedy</span>(a1is)</span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a><span class="co"># modularity measure</span></span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a><span class="fu">modularity</span>(comm_a1) <span class="co">#0.76</span></span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a><span class="co"># memberships of nodes</span></span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a><span class="co">#membership(comm_a1)</span></span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a><span class="co"># number of communities</span></span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(comm_a1) </span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a><span class="co"># size of communities</span></span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(sizes(comm_a1), breaks = 100) #mostly very small, but a few really large ones</span></span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a><span class="co"># plot communities with shaded regions</span></span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(a1c, a1, layout = a1_graph_layout, vertex.label = NA, vertex.size = 2)</span></span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a><span class="co"># plot communities without shaded regions</span></span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(a1, vertex.color = membership(c1), layout = a1_graph_layout, vertex.label = NA, vertex.size = 2)</span></span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a><span class="co">#save network for articles</span></span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(a1, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"data_a1.RData"</span>))</span>
<span id="cb38-55"><a href="#cb38-55" aria-hidden="true" tabindex="-1"></a><span class="co"># To load the data again</span></span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a><span class="co">#load(here("data", "data_a1.RData"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Collaboration networks (using igraph) - grey literature subset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Grey literature subset</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Separate author names for overall summary per paper (list of vectors format) for grey</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>Author_split_grey <span class="ot">&lt;-</span> <span class="fu">str_split</span>(<span class="at">string =</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"grey"</span>, <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>) <span class="co">#list of vectors of authors</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(Author_split_grey) <span class="co">#1358 elements (outputs) in the list</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ref.id =</span> <span class="fu">rep</span>(dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"grey"</span>, <span class="st">"Key"</span>], <span class="fu">lengths</span>(Author_split_grey)),</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">auth_names =</span> <span class="fu">unlist</span>(Author_split_grey))</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="co">#remove empty ones </span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="ot">&lt;-</span> df_auth_collab_grey[<span class="sc">!</span><span class="fu">is.na</span>(df_auth_collab_grey<span class="sc">$</span>auth_names), ]</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="ot">&lt;-</span> df_auth_collab_grey[df_auth_collab_grey<span class="sc">$</span>auth_names <span class="sc">!=</span> <span class="st">""</span>, ]</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a><span class="co">#find 25 most productive authors</span></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a><span class="co"># df_auth_collab_grey %&gt;% count(auth_names) %&gt;% arrange(-n) %&gt;% slice_head(n = 25)</span></span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a><span class="co">#compute graph of coauthorships for all authors</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="sc">%&gt;%</span></span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_auth_collab_grey, <span class="at">by =</span> <span class="st">"ref.id"</span>, <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="co">#remove many-to-many duplicates) %&gt;%</span></span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(auth_names.x <span class="sc">!=</span> auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(auth_names.x, auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">graph_from_data_frame</span>(<span class="at">directed =</span> <span class="cn">FALSE</span>) <span class="ot">-&gt;</span> g1 <span class="co">#with author names as vertices</span></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(g1) <span class="co">#IGRAPH b8acb1a UN-- 2235 57722 -- attr: name (v/c), n (e/n)</span></span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a><span class="co">#an undirected multigraph (parallel edges) with 2235 authors and 57722 scientific collaborations. Each node (author) in the network has 1 attribute: name. Each edge has 1 attributes: n (collaborations).</span></span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a><span class="co">#overall graph statistics</span></span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a><span class="fu">graph.density</span>(g1) <span class="co">#0.023 = quite sparse</span></span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a><span class="fu">transitivity</span>(g1, <span class="at">type =</span> <span class="st">"global"</span>) <span class="co"># 0.993 = probability that the neighbors of a node are also connected (also called the clustering coefficient)</span></span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a><span class="co">#see components of the igraph object</span></span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a><span class="co"># E(g1)</span></span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a><span class="co"># V(g1)</span></span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a><span class="co"># g1[]</span></span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a><span class="co"># edge_attr(g1)</span></span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a><span class="co"># vertex_attr(g1)</span></span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a><span class="co"># as_data_frame(g1, what = "edges")</span></span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a><span class="co"># get.edgelist(g1)</span></span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a><span class="co">#save network for articles</span></span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(g1, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"data_g1.RData"</span>))</span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a><span class="co"># To load the data again</span></span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a><span class="co">#load(here("data", "data_g1.RData"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot grey author collaboration network.<br>
Points are authors and edges/connections are shared articles they published together.<br>
(Note: not plotting due to this taking long time - the graph is very dense!)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load(here("data", "data_g1.RData")) #load saved data object</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co">#visualize our co-authorship graph network</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>g1_graph_layout <span class="ot">&lt;-</span> <span class="fu">layout_with_graphopt</span>(g1) <span class="co">#layout_nicely(g1) #layout.kamada.kawai(g1)  # choose an appropriate graph layout algorithm for the graph automatically</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>g1_graph_plot <span class="ot">&lt;-</span> <span class="fu">plot</span>(g1, <span class="at">layout =</span> g1_graph_layout, <span class="at">vertex.label =</span> <span class="cn">NA</span>, <span class="at">vertex.size =</span> <span class="dv">1</span>, <span class="at">vertex.color =</span> <span class="st">"grey"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Collaboration network graph metrics.<br>
(Can be run from a saved network object.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"data_g1.RData"</span>)) <span class="co">#load saved data object</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="co">#try to simplify the graph</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>g1is <span class="ot">&lt;-</span> <span class="fu">simplify</span>(g1, <span class="at">remove.multiple =</span> T, <span class="at">remove.loops =</span> T, </span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">edge.attr.comb=</span><span class="fu">list</span>(<span class="at">weight=</span><span class="st">"sum"</span>, <span class="st">"ignore"</span>) ) <span class="co">#simplify</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(g1is, layout = g1_graph_layout, edge.arrow.size=0, vertex.color="gold", vertex.size=1, vertex.frame.color="gray", </span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="co">#       vertex.label.color="black", vertex.label.cex=0.8, vertex.label.dist=2, edge.curved=0.2, vertex.label = NA)</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="co"># #E(g1is)</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a><span class="co"># #V(g1is)</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="do">##Find cliques (complete subgraphs of an undirected graph) - not good - picks up the 209-authors paper</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co"># cliques(as.undirected(g1)) # compute list of cliques       </span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(cliques(as.undirected(g1)), length) # clique sizes</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a><span class="co"># largest_cliques(as.undirected(g1)) # clique with max number of nodes - 2010 - single paper</span></span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a><span class="co"># vcol &lt;- rep("grey80", vcount(as.undirected(g1)))</span></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a><span class="co"># vcol[unlist(largest_cliques(as.undirected(g1)))] &lt;- "gold"</span></span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(as.undirected(g1), vertex.label=V(g1)$name, vertex.color=vcol)</span></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculating more centrality measures (degree, betweeness, closeness, eigenvector centrality, coreness)</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>metrics_grey <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">type =</span> <span class="fu">rep</span>(<span class="st">"grey"</span>, <span class="fu">length</span>(igraph<span class="sc">::</span><span class="fu">degree</span>(g1is))),</span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">deg =</span> igraph<span class="sc">::</span><span class="fu">degree</span>(g1is),</span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">bet =</span> igraph<span class="sc">::</span><span class="fu">betweenness</span>(g1is),</span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">clo =</span> igraph<span class="sc">::</span><span class="fu">closeness</span>(g1is),</span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">eig =</span> igraph<span class="sc">::</span><span class="fu">evcent</span>(g1is)<span class="sc">$</span>vector,</span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">cor =</span> igraph<span class="sc">::</span><span class="fu">graph.coreness</span>(g1is)</span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(metrics_grey<span class="sc">$</span>deg, metrics_grey<span class="sc">$</span>clo) <span class="co">#clo signif correlated with deg</span></span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(metrics_grey$bet)</span></span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(metrics_grey$bet, breaks = 100) # mostly extremely low values and a few high ones</span></span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(metrics_grey, aes(x = deg, fill = type)) +</span></span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_density(alpha = .3)</span></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">###</span><span class="co"> STATS  - COMPARE a1 and g1 graph degree centrality metrics</span></span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a><span class="co"># metrics_all &lt;- rbind(metrics_article, metrics_grey) #create a single data frame</span></span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a><span class="co"># str(metrics_all)</span></span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a><span class="co"># fit_metrics_all &lt;- glm(deg ~ type, family = "poisson", data = metrics_all) #node degree centrality</span></span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(fit_metrics_all) #grey slightly higher degree</span></span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a><span class="do">## Greedy community detection</span></span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a><span class="co"># greedy method (hiearchical, fast method)</span></span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a>comm_g1 <span class="ot">&lt;-</span> <span class="fu">cluster_fast_greedy</span>(g1is)</span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a><span class="co"># modularity measure </span></span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a><span class="fu">modularity</span>(comm_g1) <span class="co">#0.399</span></span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a><span class="co"># memberships of nodes</span></span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a><span class="co">#membership(comm_g1)</span></span>
<span id="cb41-52"><a href="#cb41-52" aria-hidden="true" tabindex="-1"></a><span class="co"># number of communities</span></span>
<span id="cb41-53"><a href="#cb41-53" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(comm_g1)</span>
<span id="cb41-54"><a href="#cb41-54" aria-hidden="true" tabindex="-1"></a><span class="co"># size of communities</span></span>
<span id="cb41-55"><a href="#cb41-55" aria-hidden="true" tabindex="-1"></a><span class="co"># sizes(comm_g1)</span></span>
<span id="cb41-56"><a href="#cb41-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-57"><a href="#cb41-57" aria-hidden="true" tabindex="-1"></a><span class="co"># COMPARE plot communities with shaded regions</span></span>
<span id="cb41-58"><a href="#cb41-58" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(comm_g1, g1is, layout = g1_graph_layout, vertex.label = NA, vertex.size = 2)</span></span>
<span id="cb41-59"><a href="#cb41-59" aria-hidden="true" tabindex="-1"></a><span class="co"># plot communities without shaded regions</span></span>
<span id="cb41-60"><a href="#cb41-60" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(g1is, vertex.color = membership(comm_g1), layout = g1_graph_layout, vertex.label = NA, vertex.size = 2)</span></span>
<span id="cb41-61"><a href="#cb41-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-62"><a href="#cb41-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-63"><a href="#cb41-63" aria-hidden="true" tabindex="-1"></a><span class="do">## Extract top4 components</span></span>
<span id="cb41-64"><a href="#cb41-64" aria-hidden="true" tabindex="-1"></a><span class="co">#str(g1)</span></span>
<span id="cb41-65"><a href="#cb41-65" aria-hidden="true" tabindex="-1"></a><span class="co">#comps_grey &lt;- components(g1)</span></span>
<span id="cb41-66"><a href="#cb41-66" aria-hidden="true" tabindex="-1"></a><span class="co">#groups(comps_grey)</span></span>
<span id="cb41-67"><a href="#cb41-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-68"><a href="#cb41-68" aria-hidden="true" tabindex="-1"></a><span class="co"># #alt way to see components:</span></span>
<span id="cb41-69"><a href="#cb41-69" aria-hidden="true" tabindex="-1"></a><span class="co"># comps_grey &lt;- decompose(g1, min.vertices=2)</span></span>
<span id="cb41-70"><a href="#cb41-70" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(comps_grey, gorder) #counts number of vertices of each component </span></span>
<span id="cb41-71"><a href="#cb41-71" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(comps_grey, gsize) #counts number of edges of each group </span></span>
<span id="cb41-72"><a href="#cb41-72" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(comps_grey, diameter) #calculates diameter of each group </span></span>
<span id="cb41-73"><a href="#cb41-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-74"><a href="#cb41-74" aria-hidden="true" tabindex="-1"></a><span class="co"># #a series of plots by component</span></span>
<span id="cb41-75"><a href="#cb41-75" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(decompose(g1), plot)</span></span>
<span id="cb41-76"><a href="#cb41-76" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(decompose(g1)[2], plot) #only second one</span></span>
<span id="cb41-77"><a href="#cb41-77" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply(decompose(g1)[2], vertex_attr)  #only second one, with names</span></span>
<span id="cb41-78"><a href="#cb41-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-79"><a href="#cb41-79" aria-hidden="true" tabindex="-1"></a><span class="co"># comps_grey &lt;- components(g1)</span></span>
<span id="cb41-80"><a href="#cb41-80" aria-hidden="true" tabindex="-1"></a><span class="co"># -sort(desc(comps_grey$csize)) #show descending sizes</span></span>
<span id="cb41-81"><a href="#cb41-81" aria-hidden="true" tabindex="-1"></a><span class="co"># table(comps_grey$csize &lt; 6) #70% of components have 5 or less authors</span></span>
<span id="cb41-82"><a href="#cb41-82" aria-hidden="true" tabindex="-1"></a><span class="co"># -sort(desc(comps_grey$csize))[1:4] #sizes of the top 3 components</span></span>
<span id="cb41-83"><a href="#cb41-83" aria-hidden="true" tabindex="-1"></a><span class="co"># comp_max &lt;- which(c$membership == which.max(c$csize)) # get the largest component group</span></span>
<span id="cb41-84"><a href="#cb41-84" aria-hidden="true" tabindex="-1"></a><span class="co"># comp_985 &lt;- which(comps_grey$membership == which(comps_grey$csize == 985)) # get the largest component group</span></span>
<span id="cb41-85"><a href="#cb41-85" aria-hidden="true" tabindex="-1"></a><span class="co"># comp_31 &lt;- which(comps_grey$membership == which(comps_grey$csize == 31)) # get the largest component group</span></span>
<span id="cb41-86"><a href="#cb41-86" aria-hidden="true" tabindex="-1"></a><span class="co"># comp_26 &lt;- which(comps_grey$membership == which(comps_grey$csize == 26)) # get the largest component group</span></span>
<span id="cb41-87"><a href="#cb41-87" aria-hidden="true" tabindex="-1"></a><span class="co"># comp_23 &lt;- which(comps_grey$membership == which(comps_grey$csize == 23)) # get the largest component group</span></span>
<span id="cb41-88"><a href="#cb41-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-89"><a href="#cb41-89" aria-hidden="true" tabindex="-1"></a><span class="co">#compute centrality measures for vertexes - degree (number of links to a given author)</span></span>
<span id="cb41-90"><a href="#cb41-90" aria-hidden="true" tabindex="-1"></a>g1_degree <span class="ot">&lt;-</span> <span class="fu">degree</span>(g1) </span>
<span id="cb41-91"><a href="#cb41-91" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(g1_degree) #a small number of high-degree vertexes - highly skewed (209 authors from the same paper)</span></span>
<span id="cb41-92"><a href="#cb41-92" aria-hidden="true" tabindex="-1"></a>g1_bet <span class="ot">&lt;-</span> <span class="fu">betweenness</span>(g1) <span class="co">#author betweenness measures</span></span>
<span id="cb41-93"><a href="#cb41-93" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(g1_bet) #highly skewed</span></span>
<span id="cb41-94"><a href="#cb41-94" aria-hidden="true" tabindex="-1"></a>authors_bet <span class="ot">&lt;-</span> g1_bet[g1_bet <span class="sc">&gt;</span> <span class="dv">25000</span>] <span class="co">#22 with betweenes score &gt;25000: length(names(authors_bet))</span></span>
<span id="cb41-95"><a href="#cb41-95" aria-hidden="true" tabindex="-1"></a><span class="co">#sort(authors_bet, decreasing = TRUE)</span></span>
<span id="cb41-96"><a href="#cb41-96" aria-hidden="true" tabindex="-1"></a>df_authors_bet <span class="ot">&lt;-</span> df_auth_collab_grey[df_auth_collab_grey<span class="sc">$</span>auth_names <span class="sc">%in%</span> <span class="fu">names</span>(authors_bet), ] <span class="co">#only keep ones that match names in authors_central_table, use ref.id  to see realted publications</span></span>
<span id="cb41-97"><a href="#cb41-97" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unique(df_authors_bet$ref.id)) #78 unique publications</span></span>
<span id="cb41-98"><a href="#cb41-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-99"><a href="#cb41-99" aria-hidden="true" tabindex="-1"></a>df_authors_bet <span class="sc">%&gt;%</span></span>
<span id="cb41-100"><a href="#cb41-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_authors_bet, <span class="at">by =</span> <span class="st">"ref.id"</span>, <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-101"><a href="#cb41-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="co">#remove many-to-many duplicates  </span></span>
<span id="cb41-102"><a href="#cb41-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(auth_names.x <span class="sc">&lt;</span> auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb41-103"><a href="#cb41-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(auth_names.x, auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb41-104"><a href="#cb41-104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">graph_from_data_frame</span>(<span class="at">directed =</span> <span class="cn">FALSE</span>) <span class="ot">-&gt;</span> g1b <span class="co">#with author names as vertices</span></span>
<span id="cb41-105"><a href="#cb41-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-106"><a href="#cb41-106" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(g1b) <span class="co">#IGRAPH a3211e2 UN-- 23 42 --  + attr: name (v/c), n (e/n)</span></span>
<span id="cb41-107"><a href="#cb41-107" aria-hidden="true" tabindex="-1"></a><span class="co">#overall graph statistics</span></span>
<span id="cb41-108"><a href="#cb41-108" aria-hidden="true" tabindex="-1"></a><span class="fu">graph.density</span>(g1b) <span class="co">#0.166 = quite sparse</span></span>
<span id="cb41-109"><a href="#cb41-109" aria-hidden="true" tabindex="-1"></a><span class="fu">transitivity</span>(g1b, <span class="at">type =</span> <span class="st">"global"</span>) <span class="co"># 0.457 = probability that the neighbors of a node are also connected (also called the clustering coefficient)</span></span>
<span id="cb41-110"><a href="#cb41-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-111"><a href="#cb41-111" aria-hidden="true" tabindex="-1"></a><span class="co">#see components of the igraph object</span></span>
<span id="cb41-112"><a href="#cb41-112" aria-hidden="true" tabindex="-1"></a><span class="co"># E(g1b)</span></span>
<span id="cb41-113"><a href="#cb41-113" aria-hidden="true" tabindex="-1"></a><span class="co"># V(g1b)</span></span>
<span id="cb41-114"><a href="#cb41-114" aria-hidden="true" tabindex="-1"></a><span class="co"># g1b[]</span></span>
<span id="cb41-115"><a href="#cb41-115" aria-hidden="true" tabindex="-1"></a><span class="co"># edge_attr(g1b)</span></span>
<span id="cb41-116"><a href="#cb41-116" aria-hidden="true" tabindex="-1"></a><span class="co"># vertex_attr(g1b)</span></span>
<span id="cb41-117"><a href="#cb41-117" aria-hidden="true" tabindex="-1"></a><span class="co"># as_data_frame(g1b, what = "edges")</span></span>
<span id="cb41-118"><a href="#cb41-118" aria-hidden="true" tabindex="-1"></a><span class="co"># get.edgelist(g1b)</span></span>
<span id="cb41-119"><a href="#cb41-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-120"><a href="#cb41-120" aria-hidden="true" tabindex="-1"></a><span class="co">#visualize our co-authorship graph network</span></span>
<span id="cb41-121"><a href="#cb41-121" aria-hidden="true" tabindex="-1"></a><span class="co">#g1b_graph_layout &lt;- layout_with_graphopt(g1b) #layout_as_tree(g1b) #layout_on_sphere(g1b) #layout_in_circle(g1b) #layout.kamada.kawai(g1b) #layout_nicely(g1b)</span></span>
<span id="cb41-122"><a href="#cb41-122" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(g1b, layout = g1b_graph_layout, vertex.size = betweenness(g1b), vertex.color = "grey", vertex.frame.color="grey") #vertex.label = NA, </span></span>
<span id="cb41-123"><a href="#cb41-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-124"><a href="#cb41-124" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(g1, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"data_g1.RData"</span>)) <span class="co"># save network for articles</span></span>
<span id="cb41-125"><a href="#cb41-125" aria-hidden="true" tabindex="-1"></a><span class="co">#load(here("data", "data_g1.RData")) # to load the data again</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="separate-subsets-for-peer-reviewed-articles-and-grey-literature-and-seva-intermittent-files" class="level4">
<h4 class="anchored" data-anchor-id="separate-subsets-for-peer-reviewed-articles-and-grey-literature-and-seva-intermittent-files">Separate subsets for peer-reviewed (articles) and grey literature and seva intermittent files</h4>
<p>Subset main data set.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>dat_articles <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Item.Type <span class="sc">==</span> <span class="st">"journalArticle"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Key<span class="sc">:</span>Manual.Tags)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles) #2406</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>dat_grey <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Item.Type <span class="sc">!=</span> <span class="st">"journalArticle"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Key<span class="sc">:</span>Manual.Tags)</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_grey) #1358</span></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Save main processed data and subsets: </span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_all_processed.csv"</span>))</span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat_articles, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_processed.csv"</span>))</span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat_grey, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_grey_processed.csv"</span>))</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a><span class="co">#dat &lt;- read.csv(here("data", "dat_all_processed.csv")) #loading all data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="author-affiliations---articles." class="level2">
<h2 class="anchored" data-anchor-id="author-affiliations---articles.">Author affiliations - articles.</h2>
<p>Zotero records do not contain information on authors affiliations.<br>
Trying to fetch bibliographic records from OpenAlex API using article DOI.</p>
<section id="openalex." class="level3">
<h3 class="anchored" data-anchor-id="openalex.">OpenAlex.</h3>
<p>Get affiliation information for journal articles (via doi) from OpenAlex.org platform using openalexR package.<br>
Note: do not run - skip this chunk and load locally saved fetched records.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>dois <span class="ot">&lt;-</span> <span class="fu">unique</span>(dat_articles<span class="sc">$</span>DOI) <span class="co">#some values need cleaning, as not proper DOI (empty strings) or extra chars</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>dois <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"https://doi.org/"</span>, <span class="st">""</span>, dois)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>dois <span class="ot">&lt;-</span> dois[dois <span class="sc">!=</span> <span class="st">""</span>]</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>dois <span class="ot">&lt;-</span> dois[dois <span class="sc">!=</span> <span class="st">"--"</span>]</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(dois) <span class="co">#2071</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a><span class="co">#A custom function allowing to fetch multiple records (usually works for up to 500 before throwing errors randomly) and convert the to bibliometrix-like format using oa2bibliometrix</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>block <span class="ot">&lt;-</span> <span class="cf">function</span>(i){</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"fetch request nr "</span>, i))</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Sys.sleep</span>(<span class="fl">0.2</span>)</span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  fetched <span class="ot">&lt;-</span> <span class="fu">oa_fetch</span>(</span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">doi =</span> dois[i],</span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">entity =</span> <span class="st">"works"</span>,</span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>) </span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">length</span>(fetched))</span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a><span class="co">#  return(fetched)</span></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(fetched) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="fu">oa2bibliometrix</span>(fetched) <span class="cf">else</span> <span class="fu">print</span>(<span class="st">"- not found"</span>) <span class="co">#convert or print message</span></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a><span class="co">#use the custom functions to run fetch with pauses and save individual results as a list of data frames - in chunks of around 500 records </span></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>block_test1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dois[<span class="dv">1</span><span class="sc">:</span><span class="dv">500</span>]), block)</span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>block_test2 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dois[<span class="dv">501</span><span class="sc">:</span><span class="dv">1000</span>]), block)</span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a>block_test3 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dois[<span class="dv">1001</span><span class="sc">:</span><span class="dv">1500</span>]), block)</span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>block_test4 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dois[<span class="dv">1501</span><span class="sc">:</span><span class="dv">2071</span>]), block)</span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a><span class="co">#note that json errors are thrown randomly and longer chunks are more likely to halt with # Error: lexical error: invalid char in json text</span></span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a><span class="co">#join block_test 1 to 4</span></span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a>block_test <span class="ot">&lt;-</span> <span class="fu">c</span>(block_test1, block_test2, block_test3, block_test4)</span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a><span class="co">#class(block_test) #list</span></span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(block_test) <span class="co">#2071</span></span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">lengths</span>(block_test)) <span class="co">#2063 records with 44 elements, 8 records with 1 element (empty)</span></span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(block_test, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"results_OpenAlex_full.RData"</span>)) <span class="co">#save OpenAlex output as a Rdata object</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Processing affiliation country information. (Can be run from a saved data object).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"results_OpenAlex_full.RData"</span>)) <span class="co">#loads block_test data object with full OpenAlex block query output</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="co">#extract two variables from the results</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>res_OpenA <span class="ot">&lt;-</span> <span class="fu">lapply</span>(block_test, <span class="st">"["</span>, <span class="fu">c</span>(<span class="st">"doi"</span>, <span class="st">"AU_CO"</span>))</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#data.frame(doi = sapply(block_test, "[", c("doi")), AU_CO = sapply(block_test, "[", c("AU_CO")) )</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="co">#str(res_OpenA)</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>res_OpenA <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind, res_OpenA))</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a><span class="co">#str(res_OpenA) #'data.frame':  2071 obs. of  2 variables: doi and AU_CO</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Processing affiliation country information into a data frame</span></span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>doi_country_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">doi =</span> <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">"https://doi.org/"</span>, <span class="st">""</span>, res_OpenA<span class="sc">$</span>doi)), <span class="at">AU_CO =</span> res_OpenA<span class="sc">$</span>AU_CO)</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a><span class="co"># table(is.na(doi_country_df)) #16/2 = 8 empty records</span></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a><span class="co"># doi_country_df[is.na(doi_country_df), ]</span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>doi_country_df <span class="ot">&lt;-</span> doi_country_df[<span class="sc">!</span><span class="fu">is.na</span>(doi_country_df<span class="sc">$</span>doi), ] <span class="co">#remove empty records</span></span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(doi_country_df) #2063 records = articles</span></span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>doi_country_list <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(doi_country_df<span class="sc">$</span>AU_CO, <span class="st">";"</span>) <span class="co">#creates a list of vectors of country names - many are NA</span></span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a><span class="co">#length(doi_country_list) #2063 </span></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>doi_country_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(doi_country_list, setdiff, <span class="st">'NA'</span>) <span class="co">#removes country names that are NA</span></span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>doi_country_list <span class="ot">&lt;-</span> doi_country_list[<span class="fu">lengths</span>(doi_country_list) <span class="sc">&gt;</span> <span class="dv">0</span>] <span class="co">#remove list elements that are empty - character(0)</span></span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a><span class="co">#length(doi_country_list) #1765 with non-NA country names, i.e. have any usable country info</span></span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a><span class="co">#sum(lengths(doi_country_list)) #2681 number of authors with affiliation country (after excluding NA values)</span></span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a><span class="co">#table(lengths(lapply(doi_country_list, unique)))  #table of counts of  unique country names per article (after excluding NA values)</span></span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a><span class="co">#1235/1765 #70% of articles with single-country authorship </span></span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>doi_country_all <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(doi_country_df<span class="sc">$</span>AU_CO, <span class="st">";"</span>)) <span class="co">#extract country names as a single vector</span></span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a><span class="co">#length(doi_country_all) #, including NA</span></span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a><span class="co">#table(doi_country_all) #table of country frequencies (note 2098 NA = missing)</span></span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(table(doi_country_all)) #54 different country names (including NA)</span></span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a>doi_country_all <span class="ot">&lt;-</span> doi_country_all[doi_country_all <span class="sc">!=</span> <span class="st">"NA"</span>] <span class="co">#remove NA</span></span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a><span class="co">#length(doi_country_all) #total 8672 author country values after removing NA</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>There are 1765 articles with author country information (for 2681 authors).<br>
Overall, authors are affiliated with 53 countries.<br>
List of top 10 countries in terms of the numbers of affiliated authors:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Count number of authors per country</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>country_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">table</span>(doi_country_all))</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co">#str(country_count)</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>country_count <span class="ot">&lt;-</span> <span class="fu">arrange</span>(country_count, <span class="sc">-</span>Freq)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>percent <span class="ot">&lt;-</span> <span class="fu">round</span>(country_count<span class="sc">$</span>Freq<span class="sc">/</span><span class="fu">sum</span>(country_count<span class="sc">$</span>Freq)<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>)</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>top10countries <span class="ot">&lt;-</span> country_count[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,]</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>top10countries</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   doi_country_all Freq percent
1        AUSTRALIA 5880    67.8
2              USA  695     8.0
3          GERMANY  329     3.8
4    UNITED KINDOM  200     2.3
5     SOUTH AFRICA  166     1.9
6           BRAZIL  138     1.6
7            SPAIN  130     1.5
8      NEW ZEALAND  127     1.5
9           CANADA   91     1.0
10          SWEDEN   77     0.9</code></pre>
</div>
</div>
<section id="country-level-collaboration-network." class="level4">
<h4 class="anchored" data-anchor-id="country-level-collaboration-network.">Country-level collaboration network.</h4>
<p>Process country of affiliation data - create collaboration matrices.<br>
(Output not shown due to very long processing time).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co">#create big matrix of coauthorship links among all countries (incl NA)</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>doi_country_df<span class="sc">$</span>DB <span class="ot">&lt;-</span> <span class="st">"SCOPUS"</span> <span class="co">#add extra column to enable processing with biblioNetwork function below</span></span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>NetMatrix <span class="ot">&lt;-</span> <span class="fu">biblioNetwork</span>(doi_country_df, <span class="at">analysis =</span> <span class="st">"collaboration"</span>, <span class="at">network =</span> <span class="st">"countries"</span>, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co">#str(NetMatrix)</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>net_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(NetMatrix)</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="co">#str(net_matrix)</span></span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="co">#country_sum &lt;- data.frame(countries = rownames(net_matrix), sum = rowSums(net_matrix)) #sum of links per country</span></span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="do">##use only top 10 countries to create a smaller matrix</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>small_matrix <span class="ot">&lt;-</span> net_matrix[<span class="fu">rownames</span>(net_matrix) <span class="sc">%in%</span> top10countries, <span class="fu">colnames</span>(net_matrix) <span class="sc">%in%</span> top10countries] <span class="co">#reduce matrix to most productive countries (with highest total number of authors)</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="fu">diag</span>(small_matrix) <span class="ot">&lt;-</span> <span class="dv">0</span> <span class="co">#get rid of collaboration counts within same country</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="co"># getting rid of lower triangle (as this is duplication of info)</span></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>small_matrix[<span class="fu">lower.tri</span>(small_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span> </span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="co">#colnames(small_matrix) - change to title case:</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(small_matrix) <span class="ot">&lt;-</span> <span class="fu">str_to_title</span>(<span class="fu">colnames</span>(small_matrix))</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="co">#rownames(small_matrix) - change to title case:</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(small_matrix) <span class="ot">&lt;-</span> <span class="fu">str_to_title</span>(<span class="fu">rownames</span>(small_matrix))</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Fix abbreviated country names :</span></span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(small_matrix)[<span class="fu">colnames</span>(small_matrix) <span class="sc">==</span> <span class="st">"Usa"</span>] <span class="ot">&lt;-</span> <span class="st">"USA"</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(small_matrix)[<span class="fu">rownames</span>(small_matrix) <span class="sc">==</span> <span class="st">"Usa"</span>] <span class="ot">&lt;-</span> <span class="st">"USA"</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(small_matrix)[<span class="fu">colnames</span>(small_matrix) <span class="sc">==</span> <span class="st">"United Kindom"</span>] <span class="ot">&lt;-</span> <span class="st">"United Kingdom"</span></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(small_matrix)[<span class="fu">rownames</span>(small_matrix) <span class="sc">==</span> <span class="st">"United Kindom"</span>] <span class="ot">&lt;-</span> <span class="st">"United Kingdom"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Article country collaboration chord network plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Plot country collaborations as a chord diagram for top 10 countries (using circlize package).   </span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="do">##Circumference are countries and edges/connections are shared articles they published together.   </span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.clear</span>()</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">canvas.xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">canvas.ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.2</span>, <span class="fl">1.5</span>)) <span class="co">#set margins</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co">#pdf(file = here("plots", "Figure_country_collaboration.pdf"), width=8, height=8, pointsize=10)</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a><span class="fu">chordDiagramFromMatrix</span>(small_matrix) <span class="co">#actual plotting</span></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="co">#dev.off()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot chord diagram to pdf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.chord_country_coauthorship.pdf"</span>), <span class="at">width=</span><span class="dv">8</span>, <span class="at">height=</span><span class="dv">8</span>, <span class="at">pointsize=</span><span class="dv">12</span>)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">oma=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.clear</span>()</span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">canvas.xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">canvas.ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.2</span>, <span class="fl">1.5</span>)) <span class="co">#set margins</span></span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a><span class="fu">chordDiagramFromMatrix</span>(small_matrix) <span class="co">#actual plotting</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot on the world map and save to pdf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="do">##load and filter world map data</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>world_map <span class="ot">&lt;-</span> <span class="fu">map_data</span>(<span class="st">"world"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> long <span class="sc">&gt;</span> <span class="dv">180</span>) <span class="co">#remove countries with longitude &gt;180 to make equal projection-like map without artifacts</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co">#table(world_map$region) #note that United Kingdom is UK and America is USA</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co">#table(world_map$region) #note that United Kingdom is UK and America is USA</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="do">##format country names to match regions on the world map</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">str_to_title</span>(<span class="fu">as.character</span>(country_count<span class="sc">$</span>doi_country_all)) <span class="co">#create title case names for matching</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Usa"</span>] <span class="ot">&lt;-</span> <span class="st">"USA"</span> <span class="co">#fix</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"United Kingdom"</span>] <span class="ot">&lt;-</span> <span class="st">"UK"</span> <span class="co">#fix</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"United Kindom"</span>] <span class="ot">&lt;-</span> <span class="st">"UK"</span> <span class="co">#fix</span></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Russian Federation"</span>] <span class="ot">&lt;-</span> <span class="st">"Russia"</span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Czechia"</span>] <span class="ot">&lt;-</span> <span class="st">"Czech Republic"</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Bolivia (Plurinational State Of)"</span>] <span class="ot">&lt;-</span> <span class="st">"Bolivia"</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Venezuela (Bolivarian Republic Of)"</span>] <span class="ot">&lt;-</span> <span class="st">"Venezuela"</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a><span class="co">#country_count$region[country_count$region == "Hong Kong"] &lt;- "China" #this messess up the counts?</span></span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Viet Nam"</span>] <span class="ot">&lt;-</span> <span class="st">"Vietnam"</span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Brunei Darussalam"</span>] <span class="ot">&lt;-</span> <span class="st">"Brunei"</span></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Congo"</span>] <span class="ot">&lt;-</span> <span class="st">"Republic of Congo"</span></span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Syrian Arab Republic"</span>] <span class="ot">&lt;-</span> <span class="st">"Syria"</span></span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Tanzania, United Republic Of"</span>] <span class="ot">&lt;-</span> <span class="st">"Tanzania"</span></span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a><span class="co">#country_count$region %in% world_map$region #check matching</span></span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="co">#(country_count$region)[!(country_count$region %in% world_map$region)] #check matching</span></span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="do">## colour all regions on the map:</span></span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a>emptymap <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">region =</span> <span class="fu">unique</span>(world_map<span class="sc">$</span>region), <span class="at">n =</span> <span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(<span class="fu">unique</span>(world_map<span class="sc">$</span>region)))) <span class="co">#create table with all counts as 0</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a>fullmap <span class="ot">&lt;-</span> <span class="fu">left_join</span>(emptymap, country_count, <span class="at">by =</span> <span class="st">"region"</span>) <span class="co">#join with actual counts table</span></span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a>fullmap<span class="sc">$</span>count <span class="ot">&lt;-</span> fullmap<span class="sc">$</span>n <span class="sc">+</span> fullmap<span class="sc">$</span>Freq <span class="co"># make new column for fixed counts</span></span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a>fullmap<span class="sc">$</span>count[<span class="fu">is.na</span>(fullmap<span class="sc">$</span>count)] <span class="ot">&lt;-</span> <span class="dv">0</span> <span class="co">#change NA to 0 for regions with no counts</span></span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a><span class="do">##plot</span></span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a>fullmap <span class="sc">%&gt;%</span> </span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> count, <span class="at">map_id =</span> region)) <span class="sc">+</span></span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_map</span>(<span class="at">map =</span> world_map) <span class="sc">+</span></span>
<span id="cb50-35"><a href="#cb50-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">x =</span> world_map<span class="sc">$</span>long, <span class="at">y =</span> world_map<span class="sc">$</span>lat) <span class="sc">+</span></span>
<span id="cb50-36"><a href="#cb50-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_map</span>(<span class="st">"moll"</span>) <span class="sc">+</span></span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a>  ggthemes<span class="sc">::</span><span class="fu">theme_map</span>() <span class="sc">+</span> </span>
<span id="cb50-38"><a href="#cb50-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"lightgrey"</span>, <span class="at">high =</span> <span class="st">"darkgreen"</span>, <span class="at">name =</span> <span class="st">"author count"</span>,</span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">direction =</span> <span class="st">"vertical."</span>)) <span class="sc">+</span></span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colourbar</span>(<span class="at">barwidth =</span> <span class="fu">unit</span>(<span class="dv">5</span>, <span class="at">units =</span> <span class="st">"mm"</span>), <span class="at">barheight =</span> <span class="fu">unit</span>(<span class="dv">30</span>, <span class="at">units =</span> <span class="st">"mm"</span>)))</span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-43"><a href="#cb50-43" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure to pdf</span></span>
<span id="cb50-44"><a href="#cb50-44" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(here("plots", "plot.map_author_counts.pdf"), width = 16, height = 6, units = "cm",  dpi = "retina", scale = 2)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="articles---journals-names." class="level4">
<h4 class="anchored" data-anchor-id="articles---journals-names.">Articles - Journals names.</h4>
<p>Summarise journal-level information - journal names.</p>
<p>Clean journal names.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unique((dat_articles$Publication.Title))) #774 - some names duplicated due to different capitalisation</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unique(str_to_upper(dat_articles$Publication.Title))) #759 when all uppercase</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>dat_articles<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">str_to_upper</span>(dat_articles<span class="sc">$</span>Publication.Title) <span class="co">#create a new column with uppercase journal names</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="co"># View(table(dat_articles$Journal_name)) #visual check</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>dat_articles<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"&amp;"</span>, <span class="st">"AND"</span>, dat_articles<span class="sc">$</span>Journal_name, <span class="at">fixed =</span> <span class="cn">TRUE</span>) <span class="co">#replace all &amp; with AND</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>dat_articles<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"EMU - AUSTRAL ORNITHOLOGY"</span>, <span class="st">"EMU"</span>, dat_articles<span class="sc">$</span>Journal_name, <span class="at">fixed =</span> <span class="cn">TRUE</span>) <span class="co">#replace EMU - AUSTRAL ORNITHOLOGY with EMU</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co">#length(unique(dat_articles$Journal_name)) #753</span></span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat_articles$Journal_name == "") #15 missing Journal_name</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="co">#View(dat_articles[dat_articles$Journal_name == "", ]) #see records missing journal name - some are from ResearchGate records, local journals or conference publications</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles[dat_articles$Journal_name == "", "Title"] #titles of papers missing journal name</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The articles were published in 753 different journals.</p>
<p>Plot top 10 most frequent publication journals.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>dat_articles <span class="sc">%&gt;%</span> </span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Journal_name) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(<span class="at">order_by =</span> n, <span class="at">n =</span> <span class="dv">10</span>) <span class="ot">-&gt;</span>  count_Journal_name <span class="co">#get top10 most common journal names</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_Journal_name, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Journal_name, n), <span class="at">y =</span> n)) <span class="sc">+</span> </span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Plot journals with at least 25 articles.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>dat_articles <span class="sc">%&gt;%</span> </span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Journal_name) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(n <span class="sc">&gt;=</span> <span class="dv">25</span>) <span class="ot">-&gt;</span> count_Journal_name <span class="co">#get top10 most common journal names</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_Journal_name, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Journal_name, n), <span class="at">y =</span> n)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="articles---journal-if-from-jcr." class="level4">
<h4 class="anchored" data-anchor-id="articles---journal-if-from-jcr.">Articles - Journal IF from JCR.</h4>
<p>Do not run fetching - this data can be loaded from a saved file.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: Skip this code chunk and use saved data in the next chunk.</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Based on JCR 2022 Impact Factors: https://impactfactorforjournal.com/jcr-impact-factor-2022/. The website data is not usable - the link changes every year - and only top 100 are shown.</span></span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Data for 2022 are saved as a pdf: JCR-2021-Impact-Factor-PDF.pdf (password protected for exports) with IF for 10593 journals.</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Package tabulizer is not longer supported and requires JVR for R, so use pdftools package instead.</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a><span class="co"># library(pdftools)</span></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the table</span></span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>pdf_IF_table <span class="ot">&lt;-</span> <span class="fu">pdf_text</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"JCR-2021-Impact-Factor-PDF.pdf"</span>))</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a><span class="co">#str(pdf_IF_table) #a character vector</span></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a><span class="co">#length(pdf_IF_table) #222</span></span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a><span class="co">#head(pdf_IF_table)</span></span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>pdf_IF_table2 <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(pdf_IF_table, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>) <span class="co">#split on every newline mark</span></span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>pdf_IF_table2 <span class="ot">&lt;-</span> <span class="fu">unlist</span>(pdf_IF_table2) <span class="co">#change list into a single vector of stings</span></span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a><span class="co">#pdf_IF_table2[1:10]</span></span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>pdf_IF_table2 <span class="ot">&lt;-</span> pdf_IF_table2[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)] <span class="co">#remove first 6 elements with table headers and description</span></span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a>pdf_IF_table2 <span class="ot">&lt;-</span> <span class="fu">trimws</span>(pdf_IF_table2,  <span class="at">which =</span> <span class="st">"left"</span>) <span class="co">#remove unnecessary leading white spaces</span></span>
<span id="cb54-17"><a href="#cb54-17" aria-hidden="true" tabindex="-1"></a>pdf_IF_table2 <span class="ot">&lt;-</span> <span class="fu">str_split</span>(pdf_IF_table2, <span class="st">" {2,}"</span>) <span class="co">#split at points with at least two white spaces</span></span>
<span id="cb54-18"><a href="#cb54-18" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">lengths</span>(pdf_IF_table2)) <span class="co">#there are few hundreds of misaligned records (1 or 2 strings only) - need to be removed</span></span>
<span id="cb54-19"><a href="#cb54-19" aria-hidden="true" tabindex="-1"></a>pdf_IF_table2 <span class="ot">&lt;-</span> pdf_IF_table2[<span class="fu">as.logical</span>(<span class="fu">lengths</span>(pdf_IF_table2) <span class="sc">==</span> <span class="dv">3</span>)] <span class="co">#keep only the list elements with 3 sub-elements</span></span>
<span id="cb54-20"><a href="#cb54-20" aria-hidden="true" tabindex="-1"></a>pdf_IF_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind.data.frame, pdf_IF_table2) <span class="co">#convert list to data frame</span></span>
<span id="cb54-21"><a href="#cb54-21" aria-hidden="true" tabindex="-1"></a><span class="co">#str(pdf_IF_df)</span></span>
<span id="cb54-22"><a href="#cb54-22" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pdf_IF_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb54-23"><a href="#cb54-23" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Nr"</span>,</span>
<span id="cb54-24"><a href="#cb54-24" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Journal_name"</span>,</span>
<span id="cb54-25"><a href="#cb54-25" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Journal_IF"</span></span>
<span id="cb54-26"><a href="#cb54-26" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb54-27"><a href="#cb54-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-28"><a href="#cb54-28" aria-hidden="true" tabindex="-1"></a>pdf_IF_df[<span class="fu">grepl</span>(<span class="st">"[0-9].[0-9]"</span>, pdf_IF_df<span class="sc">$</span>Journal_name), ] <span class="co">#journal titles with at least 2 numbers</span></span>
<span id="cb54-29"><a href="#cb54-29" aria-hidden="true" tabindex="-1"></a>pdf_IF_df <span class="ot">&lt;-</span> pdf_IF_df[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"[0-9].[0-9]"</span>, pdf_IF_df<span class="sc">$</span>Journal_name), ] <span class="co">#remover rows where journal titles have at least 2 numbers</span></span>
<span id="cb54-30"><a href="#cb54-30" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(pdf_IF_df) <span class="co">#10316</span></span>
<span id="cb54-31"><a href="#cb54-31" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(pdf_IF_df, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"pdf_IF_df.csv"</span>))</span>
<span id="cb54-32"><a href="#cb54-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-33"><a href="#cb54-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-34"><a href="#cb54-34" aria-hidden="true" tabindex="-1"></a><span class="do">##Merge and summarise information on IF of journals </span></span>
<span id="cb54-35"><a href="#cb54-35" aria-hidden="true" tabindex="-1"></a><span class="co">#pdf_IF_df &lt;- readr::read_csv(file = here("data", "pdf_IF_df.csv"), show_col_types = FALSE)</span></span>
<span id="cb54-36"><a href="#cb54-36" aria-hidden="true" tabindex="-1"></a><span class="co">#names(pdf_IF_df)</span></span>
<span id="cb54-37"><a href="#cb54-37" aria-hidden="true" tabindex="-1"></a><span class="co">#glimpse(pdf_IF_df) #Journal_name are a mix of title and upper case </span></span>
<span id="cb54-38"><a href="#cb54-38" aria-hidden="true" tabindex="-1"></a>pdf_IF_df<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">str_to_upper</span>(pdf_IF_df<span class="sc">$</span>Journal_name) <span class="co">#change to all capitals</span></span>
<span id="cb54-39"><a href="#cb54-39" aria-hidden="true" tabindex="-1"></a>pdf_IF_df<span class="sc">$</span>Journal_IF <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(pdf_IF_df<span class="sc">$</span>Journal_IF) <span class="co">#change to numeric values</span></span>
<span id="cb54-40"><a href="#cb54-40" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(pdf_IF_df$Journal_IF) #strongly right-skewed</span></span>
<span id="cb54-41"><a href="#cb54-41" aria-hidden="true" tabindex="-1"></a><span class="co">#pdf_IF_df[pdf_IF_df$Journal_IF &gt; 100, ]</span></span>
<span id="cb54-42"><a href="#cb54-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-43"><a href="#cb54-43" aria-hidden="true" tabindex="-1"></a><span class="co">#check overlap of journal names</span></span>
<span id="cb54-44"><a href="#cb54-44" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(dat_articles<span class="sc">$</span>Journal_name)) <span class="co">#total</span></span>
<span id="cb54-45"><a href="#cb54-45" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">intersect</span>(pdf_IF_df<span class="sc">$</span>Journal_name, dat_articles<span class="sc">$</span>Journal_name)) <span class="co">#number  matching</span></span>
<span id="cb54-46"><a href="#cb54-46" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">setdiff</span>(dat_articles<span class="sc">$</span>Journal_name, pdf_IF_df<span class="sc">$</span>Journal_name)) <span class="co">#number not matching</span></span>
<span id="cb54-47"><a href="#cb54-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-48"><a href="#cb54-48" aria-hidden="true" tabindex="-1"></a><span class="co">#check for &amp; symbol in journal names and do some cleaning in IF data frame before merging</span></span>
<span id="cb54-49"><a href="#cb54-49" aria-hidden="true" tabindex="-1"></a><span class="co">#grep("&amp;", pdf_IF_df$Journal_name, value = TRUE) #lots!</span></span>
<span id="cb54-50"><a href="#cb54-50" aria-hidden="true" tabindex="-1"></a>pdf_IF_df<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" &amp; "</span>, <span class="st">" AND "</span>, pdf_IF_df<span class="sc">$</span>Journal_name, <span class="at">fixed =</span> <span class="cn">TRUE</span>)</span>
<span id="cb54-51"><a href="#cb54-51" aria-hidden="true" tabindex="-1"></a><span class="co">#check overlap of journal names</span></span>
<span id="cb54-52"><a href="#cb54-52" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">intersect</span>(pdf_IF_df<span class="sc">$</span>Journal_name, dat_articles<span class="sc">$</span>Journal_name)) <span class="co">#413 matching</span></span>
<span id="cb54-53"><a href="#cb54-53" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">setdiff</span>(dat_articles<span class="sc">$</span>Journal_name, pdf_IF_df<span class="sc">$</span>Journal_name)) <span class="co">#340 not matching</span></span>
<span id="cb54-54"><a href="#cb54-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-55"><a href="#cb54-55" aria-hidden="true" tabindex="-1"></a><span class="co">#match and merge data to dat_articles by uppercase journal name</span></span>
<span id="cb54-56"><a href="#cb54-56" aria-hidden="true" tabindex="-1"></a>dat_articles_IF <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dat_articles, pdf_IF_df, <span class="at">by =</span> <span class="st">"Journal_name"</span>)</span>
<span id="cb54-57"><a href="#cb54-57" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles_IF)</span></span>
<span id="cb54-58"><a href="#cb54-58" aria-hidden="true" tabindex="-1"></a><span class="co">#str(dat_articles_IF)</span></span>
<span id="cb54-59"><a href="#cb54-59" aria-hidden="true" tabindex="-1"></a><span class="co">#names(dat_articles_IF)</span></span>
<span id="cb54-60"><a href="#cb54-60" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat_articles_IF, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_IF.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Summarise journal Impact Factor (IF) data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>dat_articles_IF <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_IF.csv"</span>))</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="co">#check for missing data on IF </span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">is.na</span>(dat_articles_IF<span class="sc">$</span>Journal_IF)) <span class="co">#1543 articles have IF, 863 no</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
 1543   863 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles_IF %&gt;% drop_na(journal_IF) -&gt; dat_articles_IF #only keep records with IF data</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_articles_IF)[<span class="dv">1</span>] <span class="co">#2406 records</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2406</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">n_distinct</span>(dat_articles_IF<span class="sc">$</span>Journal_name) <span class="co">#753 different journals</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 753</code></pre>
</div>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">n_distinct</span>(dat_articles_IF<span class="sc">$</span>Journal_name[<span class="sc">!</span><span class="fu">is.na</span>(dat_articles_IF<span class="sc">$</span>Journal_IF)]) <span class="co">#413 different journals with IF</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 413</code></pre>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(<span class="fu">na.omit</span>(dat_articles_IF<span class="sc">$</span>Journal_IF)) <span class="co">#63.714 highest IF</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 63.714</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(dat_articles_IF$Journal_IF, na.rm = TRUE) #strongly right-skewed distribution</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(dat_articles_IF<span class="sc">$</span>Journal_IF, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3.282</code></pre>
</div>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">IQR</span>(dat_articles_IF<span class="sc">$</span>Journal_IF, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3.225</code></pre>
</div>
</div>
<p>There are 1543 articles in journals with IF, 863 without. Hghest IF is 63.714 (Science, 2 articles).</p>
<p>Plot journals with IF &gt;15</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>dat_articles_IF <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_IF.csv"</span>)) <span class="co">#load saved fetched data</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="co">#get records for journals with IF &gt; 15</span></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>dat_articles_IF <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Journal_IF)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(Journal_name, Journal_IF, Manual.Tags, Title) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Journal_IF <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="ot">-&gt;</span> dat_articles_IF_over15</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles_IF_over15)[1] #20 in journals with IF &gt;15</span></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles_IF_over15)[1]/dim(dat_articles_IF)[1]*100 # 20 is 1% of all articles</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_over15 <span class="sc">%&gt;%</span> <span class="fu">add_count</span>(Journal_name, <span class="at">name =</span> <span class="st">"Count"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) <span class="ot">-&gt;</span> dat_articles_IF_over15 <span class="co">#counts added as a column to records</span></span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles_IF_over15$Manual.Tags #see manual tags</span></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a><span class="co">#see the journals</span></span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_over15 <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(Journal_name, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) <span class="ot">-&gt;</span> dat_articles_IF_over15_journals <span class="co">#counts as new data frame of journals</span></span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles_IF_over15_journals)[1] #10 journals with &gt;15 IF</span></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a><span class="co">#plot as counts with gradient fill for IF</span></span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(dat_articles_IF_over15_journals, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Journal_name, Count), <span class="at">y =</span> Count, <span class="at">fill =</span> Journal_IF)) <span class="sc">+</span> <span class="co">#, fill = "grey"</span></span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) <span class="sc">+</span></span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">name =</span> <span class="st">'journal impact factor &gt;15:'</span>, <span class="at">low =</span> <span class="st">"lightblue"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="articles---scimago-journal-ranking-sjr" class="level4">
<h4 class="anchored" data-anchor-id="articles---scimago-journal-ranking-sjr">Articles - Scimago Journal Ranking (SJR)</h4>
<p>Using data from Scimago website https://www.scimagojr.com/journalrank.php - downloaded as “scimagojr 2021.csv”.<br>
This data is used to get journals Quantile, Rank, Country, Region and Subject Categories.</p>
<p>(Do not run - load saved data object).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>dat_scimago <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_delim</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"scimagojr 2021.csv"</span>), <span class="at">delim =</span> <span class="st">";"</span>)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="fu">attr</span>(dat_scimago,<span class="st">'problems'</span>) <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co">#remove problems attribute from the IF column</span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="co">#clean-up Scimago journal names: </span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>dat_scimago<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(dat_scimago<span class="sc">$</span>Title, <span class="st">"[</span><span class="sc">\r\n</span><span class="st">]"</span> , <span class="st">" "</span>) <span class="sc">%&gt;%</span> <span class="fu">str_to_upper</span>() <span class="co">#remove newline symbols and change all to uppercase</span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>dat_scimago<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"&amp;"</span>, <span class="st">"AND"</span>, dat_scimago<span class="sc">$</span>Journal_name, <span class="at">fixed =</span> <span class="cn">TRUE</span>) <span class="co">#replace all &amp; with AND</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a><span class="co">#names(dat_scimago)</span></span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a><span class="co"># glimpse(dat_scimago) #all capitals</span></span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_scimago$Title</span></span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_scimago$"SJR Best Quartile"</span></span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_scimago$Country</span></span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_scimago$Region)</span></span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_scimago$Categories #needs to be strsplit</span></span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a><span class="do">##Merge SJR data with articles dataframe.  </span></span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a><span class="co">#check the overlap of journal names in two data frames</span></span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>dat_articles_IF<span class="sc">$</span>Journal_name <span class="sc">%in%</span> dat_scimago<span class="sc">$</span>Journal_name <span class="ot">-&gt;</span> fit</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(fit) <span class="co">#1880 found</span></span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span>fit) <span class="co">#526 not found</span></span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat_articles_IF$Journal_name[!fit]) #journals not found in Scimago (mostly local ones, some preprint repos, etc.)</span></span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a><span class="do">### testing - find "EMU": "EMU" %in% dat_scimago$Journal_name #TRUE</span></span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_scimago[dat_scimago$Journal_name=="EMU", ]</span></span>
<span id="cb70-25"><a href="#cb70-25" aria-hidden="true" tabindex="-1"></a><span class="co">#table(str_detect(dat_scimago$Journal_name, "PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES")) #6 partial matches</span></span>
<span id="cb70-26"><a href="#cb70-26" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_scimago$Journal_name[str_detect(dat_scimago$Journal_name, "PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES")] #showing matching names - "PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES OF THE UNITED STATES OF AMERICA"  can be fixed</span></span>
<span id="cb70-27"><a href="#cb70-27" aria-hidden="true" tabindex="-1"></a>dat_scimago<span class="sc">$</span>Journal_name <span class="ot">&lt;-</span> <span class="fu">str_replace</span>(dat_scimago<span class="sc">$</span>Journal_name, <span class="st">"^PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES OF THE UNITED STATES OF AMERICA$"</span>, <span class="st">"PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES"</span>)</span>
<span id="cb70-28"><a href="#cb70-28" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_scimago[dat_scimago$Title=="PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES", ]</span></span>
<span id="cb70-29"><a href="#cb70-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-30"><a href="#cb70-30" aria-hidden="true" tabindex="-1"></a><span class="co">#remove columns that are not essential</span></span>
<span id="cb70-31"><a href="#cb70-31" aria-hidden="true" tabindex="-1"></a>dat_scimago <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">"Title"</span>, <span class="st">"Sourceid"</span>, <span class="st">"Type"</span>, <span class="st">"Issn"</span>, <span class="st">"H index"</span>, <span class="st">"Total Docs. (2021)"</span>, <span class="st">"Total Docs. (3years)"</span>, <span class="st">"Total Refs."</span>, <span class="st">"Total Cites (3years)"</span>, <span class="st">"Citable Docs. (3years)"</span>, <span class="st">"Cites / Doc. (2years)"</span>,  <span class="st">"Ref. / Doc."</span>, <span class="st">"Publisher"</span>, <span class="st">"Coverage"</span>)) <span class="ot">-&gt;</span> dat_scimago</span>
<span id="cb70-32"><a href="#cb70-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-33"><a href="#cb70-33" aria-hidden="true" tabindex="-1"></a><span class="co">#merge data frames to append journal IF to my publications table</span></span>
<span id="cb70-34"><a href="#cb70-34" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_SJR <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dat_articles_IF, dat_scimago, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Journal_name"</span>))</span>
<span id="cb70-35"><a href="#cb70-35" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_SJR<span class="sc">$</span>SJR <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">","</span>, <span class="st">"."</span>, dat_articles_IF_SJR<span class="sc">$</span>SJR)) <span class="co">#change to numeric scores</span></span>
<span id="cb70-36"><a href="#cb70-36" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat_articles_IF_SJR, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_IF_SJR.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Summarise Scimago journal country.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_SJR <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_IF_SJR.csv"</span>))</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="co">#table(is.na(dat_articles_IF_SJR$SJR)) #1886 available</span></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="co">#counts per journal Country</span></span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">table</span>(dat_articles_IF_SJR<span class="sc">$</span>Country)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Country =</span> Var1, <span class="at">Count =</span> Freq) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>Count) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Country, Count), <span class="at">y =</span> Count)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Summarise Best Quartile data from Scimago.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co">#names(dat_articles_IF_SJR)</span></span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat_articles_IF_SJR$"SJR Best Quartile", useNA = "always")</span></span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a><span class="co">#table(is.na(dat_articles_IF_SJR$"SJR Best Quartile"), useNA = "always") #available for 1886 articles</span></span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">table</span>(dat_articles_IF_SJR<span class="sc">$</span><span class="st">"SJR Best Quartile"</span>)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Category =</span> Var1, <span class="at">Count =</span> Freq) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>Count) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Category <span class="sc">!=</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Category <span class="sc">==</span> <span class="st">"Q1"</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(Count) <span class="co">#number of articles in Q1, 1162 is 62% out of 1886 with quartile info available</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Count
1  1162</code></pre>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co">#counts per best Q category "SJR Best Quartile"</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">table</span>(dat_articles_IF_SJR<span class="sc">$</span><span class="st">"SJR Best Quartile"</span>)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Category =</span> Var1, <span class="at">Count =</span> Freq) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>Count) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Category <span class="sc">!=</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Category, Count), <span class="at">y =</span> Count)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Summarise Journal Region data from Scimago.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co">#counts per journal Region</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">table</span>(dat_articles_IF_SJR<span class="sc">$</span>Region)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Category =</span> Var1, <span class="at">Count =</span> Freq) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>Count) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Category, Count), <span class="at">y =</span> Count)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid" width="768"></p>
</div>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># #Mosaic plot for counts per journal Q and Region</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_articles_IF_SJR$SJR_Best_Quartile &lt;- dat_articles_IF_SJR$"SJR Best Quartile"</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_articles_IF_SJR$SJR_Best_Quartile[dat_articles_IF_SJR$SJR_Best_Quartile == "-"] &lt;- NA</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_IF_SJR$SJR_Best_Quartile, useNA = "always")</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_IF_SJR$Region, useNA = "always")</span></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_articles_IF_SJR$Regions &lt;- as.factor(dat_articles_IF_SJR$Region)</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a><span class="co"># levels(dat_articles_IF_SJR$Regions) &lt;- c("Other", "Other", "Other", "Other", "Other", "Northern America", "Pacific Region", "Western Europe")</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a><span class="co"># #using package ggmosaic</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a><span class="co"># dat_articles_IF_SJR %&gt;% filter(!is.na(SJR_Best_Quartile)) %&gt;% ggplot() +</span></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_mosaic(aes(x = product(Regions, SJR_Best_Quartile), fill = Regions)) +</span></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme_mosaic() + </span></span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme(legend.position = "none") </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Summarise journal category data from Scimago.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co">#counts by main use type (some articles have more than one, needs cleaning)</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>dat_articles_jcat <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat_articles_IF_SJR<span class="sc">$</span>Categories, <span class="st">"; "</span>) <span class="sc">%&gt;%</span> <span class="fu">unlist</span>() <span class="co">#separate the categories</span></span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>dat_articles_jcat <span class="ot">&lt;-</span> <span class="fu">substring</span>(dat_articles_jcat, <span class="dv">1</span>, <span class="fu">nchar</span>(dat_articles_jcat)<span class="sc">-</span><span class="dv">5</span>) <span class="co">#remove last 4 characters</span></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a><span class="co">#get top 20 most common journal categories and plot</span></span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">table</span>(dat_articles_jcat)) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">Category =</span> dat_articles_jcat, <span class="at">Count =</span> Freq) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="sc">-</span>Count) <span class="sc">%&gt;%</span> <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Category, Count), <span class="at">y =</span> Count)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Summarise Scimago journal Subject Categories.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co">#table(dat_articles_IF_SJR$Categories) #one or more Subject Categories per journal - need to separate</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>categories_list <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat_articles_IF_SJR<span class="sc">$</span>Categories, <span class="st">"; "</span>) <span class="co">#split column with author names to find most common names</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a><span class="do">##How many categories per journal? - interdisciplinarity</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a><span class="co">#count number of sub-elements in a list, for each journal</span></span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>count_categories <span class="ot">&lt;-</span> <span class="fu">lengths</span>(categories_list) <span class="co">#NA is counted as 1!</span></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>count_categories[<span class="fu">is.na</span>(categories_list)] <span class="ot">&lt;-</span> <span class="st">"no information"</span> <span class="co">#change NA to text value and all to strings</span></span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>categories_list_counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(count_categories)) <span class="do">##count each value</span></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(categories_list_counts) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"subject_categories"</span>, <span class="st">"article_count"</span>) <span class="co">#rename variables </span></span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(categories_list_counts<span class="sc">$</span>subject_categories) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"1 subject category"</span>, <span class="st">"2 subject categories"</span>, <span class="st">"3 subject categories"</span>, <span class="st">"4 subject categories"</span>, <span class="st">"5 subject categories"</span>, <span class="st">"6 subject categories"</span>, <span class="st">"7 subject categories"</span>, <span class="st">"8 subject categories"</span>, <span class="st">"9 subject categories"</span>, <span class="st">"no information"</span>) <span class="co">#rename levels</span></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a><span class="co"># #plot</span></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Colors_JCatL &lt;-  c("no information" = "#bdbdbd", rep.int("#2b2b2b", 9) ) #custom colors</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot(categories_list_counts, aes(x = subject_categories, y = article_count, fill = subject_categories)) +</span></span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   geom_bar(stat = "identity") +</span></span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   coord_flip() +</span></span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   xlab("")  +</span></span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_fill_manual(values = Colors_JCatL) +</span></span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   #geom_text(aes(label = article_count), color="white", size = 3, position = position_stack(vjust = 0.5)) +</span></span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   theme(legend.position = "none") +</span></span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   scale_x_discrete(limits = rev(levels(categories_list_counts$subject_categories))) + #reverse order of the bars (factor)</span></span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   ylab("article count") </span></span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-24"><a href="#cb78-24" aria-hidden="true" tabindex="-1"></a><span class="do">##Most common journal categories? - dominating research subjects</span></span>
<span id="cb78-25"><a href="#cb78-25" aria-hidden="true" tabindex="-1"></a><span class="co">#unlist and clean categories by removing last 5 characters (Q-values) from each sub-element</span></span>
<span id="cb78-26"><a href="#cb78-26" aria-hidden="true" tabindex="-1"></a>categories_unlisted <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">'.{5}$'</span>, <span class="st">''</span>, <span class="fu">unlist</span>(categories_list))</span>
<span id="cb78-27"><a href="#cb78-27" aria-hidden="true" tabindex="-1"></a>count_categories_unlisted <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(categories_unlisted)) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(<span class="at">order_by =</span> Freq, <span class="at">n =</span> <span class="dv">10</span>) <span class="co">#get top10 most common subject categories</span></span>
<span id="cb78-28"><a href="#cb78-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-29"><a href="#cb78-29" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb78-30"><a href="#cb78-30" aria-hidden="true" tabindex="-1"></a>Colors_JCatUL <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">"Multidisciplinary"</span> <span class="ot">=</span> <span class="st">"#8f2e2e"</span>,  <span class="st">"Management, Monitoring, Policy and Law"</span> <span class="ot">=</span> <span class="st">"#8f2e2e"</span>, <span class="st">"Medicine (miscellaneous)"</span> <span class="ot">=</span> <span class="st">"#8f2e2e"</span>, <span class="fu">rep.int</span>(<span class="st">"#2b2b2b"</span>, <span class="dv">17</span>) ) <span class="co">#custom colors</span></span>
<span id="cb78-31"><a href="#cb78-31" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(count_categories_unlisted, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(categories_unlisted, Freq), <span class="at">y =</span> Freq, <span class="at">fill =</span> categories_unlisted)) <span class="sc">+</span> </span>
<span id="cb78-32"><a href="#cb78-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb78-33"><a href="#cb78-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb78-34"><a href="#cb78-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">''</span>, <span class="at">values =</span> Colors_JCatUL, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">'Multidisciplinary'</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"journal subject categories other than Biology/Ecology/Environment"</span>)) <span class="sc">+</span></span>
<span id="cb78-35"><a href="#cb78-35" aria-hidden="true" tabindex="-1"></a>    <span class="co">#geom_text(aes(label = article_count), color = "white", size = 3, position = position_stack(vjust = 0.5)) + </span></span>
<span id="cb78-36"><a href="#cb78-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>, <span class="at">legend.direction =</span> <span class="st">"horizontal"</span>) <span class="sc">+</span></span>
<span id="cb78-37"><a href="#cb78-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb78-38"><a href="#cb78-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-49-1.png" class="img-fluid" width="768"></p>
</div>
</div>
</section>
<section id="article-open-access-status" class="level4">
<h4 class="anchored" data-anchor-id="article-open-access-status">Article open access status</h4>
<p>Get open access article status information from Unpaywall - use DOI to integrate with additional data available from Unpaywall via <a href="https://docs.ropensci.org/roadoi/reference/roadoi-package.html">roadoi R package</a>.</p>
<p>(Do not run this code chunk and use the saved data loaded in the next chunk).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co">#prepare</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>dat_articles<span class="sc">$</span>doi <span class="ot">&lt;-</span> dat_articles<span class="sc">$</span>DOI <span class="co">#copy column contents</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles$doi[dat_articles$doi == ""] &lt;- NA #replace empty DOI with "NA"</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a><span class="co">#table(is.na(dat_articles$doi), useNA = "always") #2406 not NA, 334 are NA</span></span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>dois <span class="ot">&lt;-</span> <span class="fu">unique</span>(dat_articles<span class="sc">$</span>doi) <span class="co">#get unique DOI (not empty strings)</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a><span class="co">#length(dois) # 2073</span></span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a><span class="co">#connect and fetch the data</span></span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>OA_df <span class="ot">&lt;-</span> roadoi<span class="sc">::</span><span class="fu">oadoi_fetch</span>(<span class="at">dois =</span> dois, <span class="at">email =</span> <span class="st">"losialagisz@gmail.com"</span>) <span class="co">#get refs data based on doi</span></span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a><span class="co">#There were 33 warnings (use warnings() to see them)</span></span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(OA_df, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"OA_article_data_Unpaywall.csv"</span>)) <span class="co">#save fetched data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Merge and summarise information on Open Access status of journal articles from Unpaywall.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>OA_df <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"OA_article_data_Unpaywall.csv"</span>)) <span class="co">#load saved data object</span></span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="co"># names(OA_df)</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="co"># class(OA_df) #list of data frames with lists of data frames</span></span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="co"># str(warnings()) #List of warnings - mainly that an article is not in Unpaywall</span></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a><span class="co"># failed_DOIs &lt;- warnings() #get failed DOIs</span></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(OA_df) #2040</span></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(OA_df$doi)) #2040 - successfully retrieved</span></span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a><span class="co"># length(anyDuplicated(OA_df$doi)) #1</span></span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-10"><a href="#cb80-10" aria-hidden="true" tabindex="-1"></a>dat_articles<span class="sc">$</span>doi <span class="ot">&lt;-</span> dat_articles<span class="sc">$</span>DOI <span class="co">#copy column contents for easier merging of the two dataframes</span></span>
<span id="cb80-11"><a href="#cb80-11" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(dat_articles$doi)) #2073 </span></span>
<span id="cb80-12"><a href="#cb80-12" aria-hidden="true" tabindex="-1"></a><span class="co"># length(anyDuplicated(dat_articles$doi)) #1</span></span>
<span id="cb80-13"><a href="#cb80-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-14"><a href="#cb80-14" aria-hidden="true" tabindex="-1"></a><span class="do">##check overlap</span></span>
<span id="cb80-15"><a href="#cb80-15" aria-hidden="true" tabindex="-1"></a><span class="co"># length(setdiff(dat_articles$doi, OA_df$doi)) #262 not matching (ignores missing doi)</span></span>
<span id="cb80-16"><a href="#cb80-16" aria-hidden="true" tabindex="-1"></a><span class="co"># length(setdiff(OA_df$doi, dat_articles$doi)) #NA</span></span>
<span id="cb80-17"><a href="#cb80-17" aria-hidden="true" tabindex="-1"></a><span class="co"># length(intersect(OA_df$doi, dat_articles$doi)) #1811 matching</span></span>
<span id="cb80-18"><a href="#cb80-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-19"><a href="#cb80-19" aria-hidden="true" tabindex="-1"></a><span class="do">##match and merge data to dat_articles by DOI</span></span>
<span id="cb80-20"><a href="#cb80-20" aria-hidden="true" tabindex="-1"></a>dat_articles_OA <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dat_articles, OA_df, <span class="at">by =</span> <span class="st">"doi"</span>)</span>
<span id="cb80-21"><a href="#cb80-21" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles_OA) #2406</span></span>
<span id="cb80-22"><a href="#cb80-22" aria-hidden="true" tabindex="-1"></a><span class="co">#str(dat_articles_OA) #note this also includes author info as nested data frames</span></span>
<span id="cb80-23"><a href="#cb80-23" aria-hidden="true" tabindex="-1"></a><span class="co">#names(dat_articles_OA)</span></span>
<span id="cb80-24"><a href="#cb80-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-25"><a href="#cb80-25" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat_articles_OA, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_OA.csv"</span>)) <span class="co">#save merged data frame for later - this removes nested data frames with author information</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Processing article open access status information.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>dat_articles_OA <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_OA.csv"</span>))</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="co">#summarise - note that &lt;NA&gt; is for 595 studies that had doi not matched with info on Unpaywall, resulting in NA values</span></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a><span class="co"># table(!is.na(dat_articles_OA$is_oa)) #data for 1811, missing for 595</span></span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_OA$is_oa, useNA = "always")</span></span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_OA$oa_status, useNA = "always")</span></span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_OA$has_repository_copy, useNA = "always")</span></span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_OA$journal_is_oa, useNA = "always")</span></span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_OA$is_oa, dat_articles_OA$oa_status)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Open Access status data available for 1811 articles.<br>
Among those, 1096 articles have some type of Open Access version available:<br>
1019 have repository copy,<br>
595 journal is Open Access.</p>
</section>
<section id="article-citations-counts." class="level4">
<h4 class="anchored" data-anchor-id="article-citations-counts.">Article citations counts.</h4>
<p>Get article academic citation data from Crossref using rcrossref package.</p>
<p>(Do not run this code chunk and use saved data in the next chunk).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Using rcrossref package: Register for the polite pool. Open file: file.edit("~/.Renviron"). Add email address to be shared with Crossref crossref_email= "name@example.com". Save the file and restart your R session</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="do">##Load  a processed version list of "impact" literature related to ALA - use the .csv file</span></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="co">#dat &lt;- read.csv(here("data", "dat_all_processed.csv")) #loading all data</span></span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles &lt;- dat %&gt;% filter(Item.Type == "journalArticle") #%&gt;% dplyr::select(Key:Manual.Tags)</span></span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a><span class="co"># OR</span></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles &lt;- read.csv(here("data", "dat_articles_processed.csv"))</span></span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles) #2406</span></span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="co">#names(dat_articles)</span></span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a><span class="co"># dois &lt;- dat$DOI #article doi</span></span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(dois)) #2171 DOI (but see below)</span></span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a><span class="co"># dois &lt;- unique(dois)</span></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a><span class="co"># dois &lt;- dois[dois != ""]</span></span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a><span class="co"># length(dois) #2170</span></span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a><span class="co">#fetch only citation counts</span></span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>citation_counts <span class="ot">&lt;-</span> <span class="fu">cr_citation_count</span>(<span class="at">doi =</span> dois, <span class="at">async =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a><span class="co">#str(citation_counts) #a data frame with just doi and count columns</span></span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">is.na</span>(citation_counts<span class="sc">$</span>count)) <span class="co">#2131 ok, 39 NA</span></span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a><span class="co">#crosreff_works &lt;- cr_works(dois = c(dois[1:10])) #using high level API function to get more information</span></span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a><span class="co">#str(crosreff_works) #messy output format, often no affiliations</span></span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a><span class="co">#readr::write_csv(citation_counts, file = here("data","citation_counts.csv")) #save fetched data for later</span></span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a>citation_counts <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"citation_counts.csv"</span>))</span>
<span id="cb82-26"><a href="#cb82-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-27"><a href="#cb82-27" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles &lt;- readr::read_csv(file = here("data","dat_articles.csv"))</span></span>
<span id="cb82-28"><a href="#cb82-28" aria-hidden="true" tabindex="-1"></a>dat_articles<span class="sc">$</span>doi <span class="ot">&lt;-</span> dat_articles<span class="sc">$</span>DOI <span class="co">#copy column contents</span></span>
<span id="cb82-29"><a href="#cb82-29" aria-hidden="true" tabindex="-1"></a>dat_articles<span class="sc">$</span>doi[dat_articles<span class="sc">$</span>doi <span class="sc">==</span> <span class="st">""</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span> <span class="co">#replace empty DOI with "NA"</span></span>
<span id="cb82-30"><a href="#cb82-30" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">is.na</span>(dat_articles<span class="sc">$</span>doi), <span class="at">useNA =</span> <span class="st">"always"</span>) <span class="co">#2740 not NA, 353 are NA</span></span>
<span id="cb82-31"><a href="#cb82-31" aria-hidden="true" tabindex="-1"></a>dat_articles_CR <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dat_articles, citation_counts, <span class="at">by =</span> <span class="st">"doi"</span>) <span class="co">#match and merge data to dat_articles by doi</span></span>
<span id="cb82-32"><a href="#cb82-32" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">is.na</span>(dat_articles_CR<span class="sc">$</span>count)) <span class="co">#2039 counts available</span></span>
<span id="cb82-33"><a href="#cb82-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-34"><a href="#cb82-34" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat_articles_CR, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_CR.csv"</span>)) <span class="co">#save merged data frame for plotting later</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Summarise and match citation data to the main data frame by doi.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>dat_articles_CR <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_CR.csv"</span>))  <span class="co">#load saved merged data frame</span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="co"># median(dat_articles_CR$count, na.rm = TRUE) #6</span></span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a><span class="co"># mean(dat_articles_CR$count, na.rm = TRUE) #19.3</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="co"># sd(dat_articles_CR$count, na.rm = TRUE) #96.4</span></span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a><span class="co"># IQR(dat_articles_CR$count, na.rm = TRUE) #14</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="co"># min(dat_articles_CR$count, na.rm = TRUE) #0</span></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="co"># max(dat_articles_CR$count, na.rm = TRUE) #3732</span></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a><span class="co">#subset highly cited (&gt;100) publications</span></span>
<span id="cb83-10"><a href="#cb83-10" aria-hidden="true" tabindex="-1"></a>dat_articles_CR_highlycited100 <span class="ot">&lt;-</span> dat_articles_CR <span class="sc">%&gt;%</span></span>
<span id="cb83-11"><a href="#cb83-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">100</span>)</span>
<span id="cb83-12"><a href="#cb83-12" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles_CR_highlycited100) #45 articles with &gt;100 citations</span></span>
<span id="cb83-13"><a href="#cb83-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb83-14"><a href="#cb83-14" aria-hidden="true" tabindex="-1"></a><span class="co">#subset highly cited (&gt;500) publications</span></span>
<span id="cb83-15"><a href="#cb83-15" aria-hidden="true" tabindex="-1"></a>dat_articles_CR_highlycited500 <span class="ot">&lt;-</span> dat_articles_CR <span class="sc">%&gt;%</span></span>
<span id="cb83-16"><a href="#cb83-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">500</span>)</span>
<span id="cb83-17"><a href="#cb83-17" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_articles_CR_highlycited500) <span class="co">#4 articles with &gt;500 citations</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  4 43</code></pre>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles_CR_highlycited[ , 3:6]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>There are 45, 43 articles with &gt;100 citations.<br>
There are 4, 43 articles with &gt;500 citations.</p>
<p>Plot citation counts across years:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="fl">0.1</span>, <span class="at">seed =</span> <span class="dv">1</span>) <span class="co">#used below to add jitter</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>dat_articles_CR <span class="sc">%&gt;%</span></span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(count)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have no available citation counts</span></span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Publication.Year, <span class="at">y =</span> count)) <span class="sc">+</span> </span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>, <span class="at">position =</span> pos, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span></span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_smooth(method = "lm", color = "grey") +</span></span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()  <span class="sc">+</span></span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"total citation count"</span>) <span class="sc">+</span></span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"year"</span>) <span class="sc">+</span></span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> dat_articles_CR_highlycited500, <span class="fu">aes</span>(<span class="at">label =</span> Publication.Title,), <span class="at">size =</span> <span class="dv">3</span>, <span class="at">max.overlaps =</span> <span class="dv">25</span>, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span> <span class="co">#add journal names as labels for highly cited publications only</span></span>
<span id="cb86-13"><a href="#cb86-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="co">#point labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid" width="768"></p>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure to pdf</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave(here("plots", "plot.citations_years.pdf"), width = 16, height = 6, units = "cm",  dpi = "retina", scale = 2)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="altmetrics" class="level4">
<h4 class="anchored" data-anchor-id="altmetrics">Altmetrics</h4>
<p>Gather Altmetrics data using a customised function picked up from discontinued rAltmetric R package.<br>
(Do not run this code chunk and use saved data in the next chunk).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: skip this chunk and use saved query output in the next chunk. </span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a><span class="co"># https://cran.r-project.org/web/packages/rAltmetric/index.html  </span></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Package manual: http://cran.nexr.com/web/packages/rAltmetric/rAltmetric.pdf and example: https://www.r-project.org/nosvn/pandoc/rAltmetric.html</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: rAltmetric package was "archived on 2022-06-02 at the maintainer's request".    </span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a><span class="co"># API Altmetric info: https://api.altmetric.com/docs/call_fetch.html</span></span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles &lt;- readr::read_csv(file = here("data","dat_articles.csv")) #or use:</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles_OA_IF_SJR &lt;- readr::read_csv(file = here("data","dat_articles_OA_IF_SJR.csv"))</span></span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a><span class="co">#names(dat_articles_OA_IF_SJR)</span></span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a><span class="co">#define getAltmetrics() function </span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>getAltmetrics <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">doi =</span> <span class="cn">NULL</span>,</span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>                          <span class="at">foptions =</span> <span class="fu">list</span>(),</span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>                           ...) {</span>
<span id="cb88-15"><a href="#cb88-15" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(doi)) doi <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_c</span>(<span class="st">"doi/"</span>, doi)</span>
<span id="cb88-16"><a href="#cb88-16" aria-hidden="true" tabindex="-1"></a>    identifiers <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">compact</span>(<span class="fu">list</span>(doi))</span>
<span id="cb88-17"><a href="#cb88-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(identifiers)) {</span>
<span id="cb88-18"><a href="#cb88-18" aria-hidden="true" tabindex="-1"></a>      ids <span class="ot">&lt;-</span> identifiers[[<span class="dv">1</span>]]</span>
<span id="cb88-19"><a href="#cb88-19" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb88-20"><a href="#cb88-20" aria-hidden="true" tabindex="-1"></a>    base_url <span class="ot">&lt;-</span> <span class="st">"http://api.altmetric.com/v1/"</span></span>
<span id="cb88-21"><a href="#cb88-21" aria-hidden="true" tabindex="-1"></a>    request <span class="ot">&lt;-</span> httr<span class="sc">::</span><span class="fu">GET</span>(<span class="fu">paste0</span>(base_url, ids))</span>
<span id="cb88-22"><a href="#cb88-22" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span></span>
<span id="cb88-23"><a href="#cb88-23" aria-hidden="true" tabindex="-1"></a>      jsonlite<span class="sc">::</span><span class="fu">fromJSON</span>(httr<span class="sc">::</span><span class="fu">content</span>(request, <span class="at">as =</span> <span class="st">"text"</span>), <span class="at">flatten =</span> <span class="cn">TRUE</span>)</span>
<span id="cb88-24"><a href="#cb88-24" aria-hidden="true" tabindex="-1"></a>    results <span class="ot">&lt;-</span> rlist<span class="sc">::</span><span class="fu">list.flatten</span>(results)</span>
<span id="cb88-25"><a href="#cb88-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">class</span>(results) <span class="ot">&lt;-</span> <span class="st">"altmetric"</span></span>
<span id="cb88-26"><a href="#cb88-26" aria-hidden="true" tabindex="-1"></a>    results</span>
<span id="cb88-27"><a href="#cb88-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb88-28"><a href="#cb88-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-29"><a href="#cb88-29" aria-hidden="true" tabindex="-1"></a><span class="co">#define helper function</span></span>
<span id="cb88-30"><a href="#cb88-30" aria-hidden="true" tabindex="-1"></a>altmetric_df <span class="ot">&lt;-</span> <span class="cf">function</span>(altmetric.object) {</span>
<span id="cb88-31"><a href="#cb88-31" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">unlist</span>(altmetric.object)), </span>
<span id="cb88-32"><a href="#cb88-32" aria-hidden="true" tabindex="-1"></a>                   <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb88-33"><a href="#cb88-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb88-34"><a href="#cb88-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-35"><a href="#cb88-35" aria-hidden="true" tabindex="-1"></a><span class="co">#run for a single doi:</span></span>
<span id="cb88-36"><a href="#cb88-36" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results &lt;- getAltmetrics(doi = "10.3389/fagro.2021.657785") %&gt;% altmetric_df()</span></span>
<span id="cb88-37"><a href="#cb88-37" aria-hidden="true" tabindex="-1"></a><span class="co">#tests for multiple dois</span></span>
<span id="cb88-38"><a href="#cb88-38" aria-hidden="true" tabindex="-1"></a><span class="co"># dois &lt;- list(c("10.3389/fagro.2021.657785", "10.1016/j.biocon.2022.109470"))</span></span>
<span id="cb88-39"><a href="#cb88-39" aria-hidden="true" tabindex="-1"></a><span class="co"># dois &lt;- list(c("10.3316/informit.240266496810807", "10.1089/bio.2021.0127", "10.1016/j.biocon.2022.109470"))</span></span>
<span id="cb88-40"><a href="#cb88-40" aria-hidden="true" tabindex="-1"></a><span class="co"># dois &lt;- dat_articles_OA_IF_SJR$DOI[!is.na(dat_articles_OA_IF_SJR$DOI)][1:25] @removing empty doi and subsetting</span></span>
<span id="cb88-41"><a href="#cb88-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-42"><a href="#cb88-42" aria-hidden="true" tabindex="-1"></a>dois <span class="ot">&lt;-</span> dat_articles<span class="sc">$</span>DOI <span class="co">#get all dois, or dois &lt;- dat_articles_OA_IF_SJR$DOI #all dois</span></span>
<span id="cb88-43"><a href="#cb88-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-44"><a href="#cb88-44" aria-hidden="true" tabindex="-1"></a><span class="co">#run custom function and try to catch errors if not found (articles which have 0 Altmetric score are not in the database)</span></span>
<span id="cb88-45"><a href="#cb88-45" aria-hidden="true" tabindex="-1"></a>Altmetrics.results <span class="ot">&lt;-</span> <span class="fu">list</span>(dois) <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">pmap</span>( <span class="sc">~</span> <span class="fu">tryCatch</span>( {<span class="fu">getAltmetrics</span>(<span class="at">doi =</span> .x) <span class="sc">%&gt;%</span> <span class="fu">altmetric_df</span>()},</span>
<span id="cb88-46"><a href="#cb88-46" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">error =</span> <span class="cf">function</span>(e) {</span>
<span id="cb88-47"><a href="#cb88-47" aria-hidden="true" tabindex="-1"></a>                                             <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">"•NOT_FOUND• "</span>, .x, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>))</span>
<span id="cb88-48"><a href="#cb88-48" aria-hidden="true" tabindex="-1"></a>                                             }))</span>
<span id="cb88-49"><a href="#cb88-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-50"><a href="#cb88-50" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(Altmetrics.results) <span class="co">#2073</span></span>
<span id="cb88-51"><a href="#cb88-51" aria-hidden="true" tabindex="-1"></a><span class="co">#str(Altmetrics.results) #list of data frames of 1 observation and different number of variables</span></span>
<span id="cb88-52"><a href="#cb88-52" aria-hidden="true" tabindex="-1"></a><span class="co">#table(sapply(Altmetrics.results, is.null)) #554 are NULL - no altmetrics data</span></span>
<span id="cb88-53"><a href="#cb88-53" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results[sapply(Altmetrics.results, is.null)] &lt;- NULL #remove empty list elements</span></span>
<span id="cb88-54"><a href="#cb88-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-55"><a href="#cb88-55" aria-hidden="true" tabindex="-1"></a><span class="co">#convert list of altmetrics results to cleaner data frame - it drops empty rows</span></span>
<span id="cb88-56"><a href="#cb88-56" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">rbindlist</span>(Altmetrics.results, <span class="at">fill =</span> <span class="cn">TRUE</span>)</span>
<span id="cb88-57"><a href="#cb88-57" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Altmetrics.results_df) <span class="co">#1519</span></span>
<span id="cb88-58"><a href="#cb88-58" aria-hidden="true" tabindex="-1"></a><span class="co">#str(Altmetrics.results_df)</span></span>
<span id="cb88-59"><a href="#cb88-59" aria-hidden="true" tabindex="-1"></a><span class="co">#names(Altmetrics.results_df) #345 columns - extra author names columns after 110!</span></span>
<span id="cb88-60"><a href="#cb88-60" aria-hidden="true" tabindex="-1"></a><span class="co">#View(Altmetrics.results_df[, 70:125]) #see some columns at the end</span></span>
<span id="cb88-61"><a href="#cb88-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-62"><a href="#cb88-62" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(Altmetrics.results_df, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"Altmetrics.results_df.csv"</span>)) <span class="co">#save</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Interpreting Altmetrics key output statistics:<br>
- score - “Altmetric score” = the main Altmetric attention score<br>
- cited_by_policies_count - “Number of policy sources that have mentioned the publication”<br>
- cited_by_msm_count - “Number of the news sources that have mentioned the publication”<br>
- cited_by_feeds_count - “Number of blogs that have mentioned the publication”<br>
- cited_by_tweeters_count - “Number of the twitter accounts that have tweeted this publication”<br>
- cited_by_fbwalls_count - “Number of the pages that have shared on Facebook”<br>
- cited_by_rdts_count - “Number of Reddit threads posted about this publication”<br>
- cited_by_videos_count - “Number of the Youtube/Vimeo channels”<br>
- cited_by_gplus_count - “Number of the accounts that have shared on Google+”</p>
<p>See more on https://docs.google.com/spreadsheets/d/1ndVY8Q2LOaZO_P_HDmSQulagjeUrS250mAL2N5V8GvY/edit#gid=0</p>
<p>Summarize Altmetrics data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"Altmetrics.results_df.csv"</span>)) <span class="co">#load saved data file</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="co">#plot a histogram of the Altmetric scores (code from https://www.r-bloggers.com/2015/09/altmetrics-on-citeulike-entries-in-r/):</span></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(as.numeric(Altmetrics.results_df$score), main="", xlab="Altmetric score", breaks=50)</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Altmetrics.results_df) <span class="co">#1519 articles with Altmetrics data</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1519  343</code></pre>
</div>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="do">##names(Altmetrics.results_df)</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking and processing to more useful formats</span></span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="co">#max(as.numeric(Altmetrics.results_df$score)) </span></span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results_df$authors1 #name of first author (subsequent authors names are stored in separate numbered author columns)</span></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a><span class="co">#table(is.na(Altmetrics.results_df$images.small)) #no NA</span></span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a><span class="co">#table(Altmetrics.results_df$is_oa, useNA = "always") #all FALSE - not useful</span></span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a><span class="co">#table(Altmetrics.results_df$subjects, useNA = "always") #with NA - not useful</span></span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results_df$history.1y</span></span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a><span class="co">#table(is.na(Altmetrics.results_df$score)) #no NA - overall composite altmetric score</span></span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Altmetrics.results_df<span class="sc">$</span>score) <span class="co">#make numeric</span></span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results_df$published_on #in POSIXct format - the number of seconds from 1 January 1970 (or 1960?)</span></span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>published_date <span class="ot">&lt;-</span> Altmetrics.results_df<span class="sc">$</span>published_on <span class="co">#add a new column </span></span>
<span id="cb91-14"><a href="#cb91-14" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>published_date <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">"NA        "</span>, <span class="cn">NA</span>, Altmetrics.results_df<span class="sc">$</span>published_date)) <span class="co">#change "NA        " to NA and convert to numeric</span></span>
<span id="cb91-15"><a href="#cb91-15" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>published_date <span class="ot">&lt;-</span> <span class="fu">as.POSIXct</span>(Altmetrics.results_df<span class="sc">$</span>published_date, <span class="at">origin =</span> <span class="st">"1970-01-01"</span>) <span class="co">#convert POSIXct</span></span>
<span id="cb91-16"><a href="#cb91-16" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>year <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">format</span>(Altmetrics.results_df<span class="sc">$</span>published_date, <span class="at">format=</span><span class="st">"%Y"</span>)) <span class="co">#extract year</span></span>
<span id="cb91-17"><a href="#cb91-17" aria-hidden="true" tabindex="-1"></a><span class="co">#table(is.na(Altmetrics.results_df$year)) #4 values missing</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>There are 1519 articles with Altmetrics data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Make tables of article subsets with specific types of attention.    </span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="do">##Patents</span></span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_patents_count, <span class="at">useNA =</span> <span class="st">"always"</span>) <span class="co"># 3 articles cited once each, 1 more than once</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    3 &lt;NA&gt; 
   3    1 1515 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with patent citations</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_patents_count) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_patents_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_patents_count)) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles cited by patents"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-f6540f2e19c7492ef971" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-f6540f2e19c7492ef971">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles cited by patents<\/caption>","data":[["The Antibacterial Activity of Australian Leptospermum Honey Correlates with Methylglyoxal Levels","Capturing Natural-Colour 3D Models of Insects for Species Discovery and Diagnostics","Toward an ecoregion scale evaluation of eDNA metabarcoding primers: A case study for the freshwater fish biodiversity of the Murray-Darling Basin (Australia)","Plant salt tolerance: adaptations in halophytes"],["PLOS ONE","PLOS ONE","Ecology and Evolution","Annals of Botany"],["Nural N. Cokcetin","Chuong V. Nguyen","Jonas Bylemans","Timothy J. Flowers"],[2016,2014,2018,2015],["10.1371/journal.pone.0167780","10.1371/journal.pone.0094346","10.1002/ece3.4387","10.1093/aob/mcu267"],[3,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_patents_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Policies</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_policies_count, <span class="at">useNA =</span> <span class="st">"always"</span>) <span class="co"># 66 articles cited once each, 10 more than once</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    2    3 &lt;NA&gt; 
  66    7    3 1443 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with patent citations</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_policies_count) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_policies_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_policies_count))  <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles cited by patents"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-69c039f41c819bcf55b2" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-69c039f41c819bcf55b2">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles cited by patents<\/caption>","data":[["Climate change influences on global distributions of dengue and chikungunya virus vectors","A statistical explanation of MaxEnt for ecologists","Overview of distribution patterns of zooxanthellate Scleractinia","A decision tree for assessing the risks and benefits of publishing biodiversity data","Dealing with Cumulative Biodiversity Impacts in Strategic Environmental Assessment: A New Frontier for Conservation Planning","Opening the black box: an open-source release of Maxent","Introducing the Global Register of Introduced and Invasive Species","Predicting species distributions for conservation decisions","Smartphones in ecology and evolution: a guide for the app-rehensive","Darwin Core: An Evolving Community-Developed Biodiversity Data Standard","Green Tongues into the Arid Zone: River Floodplains Extend the Distribution of Terrestrial Bird Species","Ecological drivers of soil microbial diversity and soil biological networks in the Southern Hemisphere","Black fungus gnats (Diptera: Sciaridae) found in association with cultivated plants and mushrooms in Australia, with notes on cosmopolitan pest species and biosecurity interceptions","Changing Climate Suitability for Dominant Eucalyptus Species May Affect Future Fuel Loads and Flammability in Tasmania","The IQ of the Crowd: Understanding and Improving Information Quality in Structured User-Generated Content","bioclim: the first species distribution modelling package, its early applications and relevance to most currentMaxEntstudies","A biography of an invasive terrestrial slug: the spread, distribution and habitat of Deroceras invadens","Transdisciplinary synthesis for ecosystem science, policy and management: The Australian experience","GlobalTreeSearch: The first complete global database of tree species and country distributions","A DNA ‘Barcode Blitz’: Rapid Digitization and Sequencing of a Natural History Collection","Developing biodiverse plantings suitable for changing climatic conditions 2: Using the Atlas of Living Australia","Reversal of competitive dominance between invasive and native freshwater crayfish species under near-future elevated water temperature","Occupancy models for citizen‐science data","Invasive weed species’ threats to global biodiversity: Future scenarios of changes in the number of invasive species in a changing climate","Assessing the vulnerability of Australia’s urban forests to climate extremes","Historical perspectives and new opportunities for Australian collections of microorganisms in the microbiome era","Spatial and temporal patterns of land clearing during policy change","Crop niche modeling projects major shifts in common bean growing areas","Comparative phylogeography of the western Indian Ocean reef fauna","Improving the Design of a Conservation Reserve for a Critically Endangered Species","Impact of Austropuccinia psidii (myrtle rust) on Myrtaceae-rich wet sclerophyll forests in south east Queensland","Responses of invasive predators and native prey to a prescribed forest fire","Sink or swim? Potential for high faunal turnover in Australian rivers under climate change","Plants show more flesh in the tropics: variation in fruit type along latitudinal and climatic gradients","Evaluating 318 continental-scale species distribution models over a 60-year prediction horizon: what factors influence the reliability of predictions?","National standards for the practice of ecological restoration in Australia","Disentangling the four demographic dimensions of species invasiveness","Climatic niche models and their consensus projections for future climates for four major forest tree species in the Asia–Pacific region","Which species distribution models are more (or less) likely to project broad-scale, climate-induced shifts in species ranges?","Maximizing Seed Resources for Restoration in an Uncertain Future","Noah’s Ark Conservation Will Not Preserve Threatened Ecological Communities under Climate Change","Absence of MERS-CoV antibodies in feral camels in Australia: Implications for the pathogen's origin and spread","Bias correction in species distribution models: pooling survey and collection data for multiple species","Priority threat management of non-native plants to maintain ecosystem integrity across heterogeneous landscapes","How will the ‘molecular revolution’ contribute to biological recording?","Emerging technologies for biological recording","Towards global interoperability for supporting biodiversity research on essential biodiversity variables (EBVs)","The Red-whiskered Bulbul Pycnonotus jocosus in Australia – a global perspective, history of introduction, current status and potential impacts","Ecosystem risk assessment of Georgina gidgee woodlands in central Australia","High‐productivity vegetation is important for lessening bird declines during prolonged drought","Suitable areas of Phakopsora pachyrhizi, Spodoptera exigua, and their host plant Phaseolus vulgaris are projected to reduce and shift due to climate change","Innovation in Citizen Science – Perspectives on Science-Policy Advances","Rapid shifts in distribution and high‐latitude persistence of oceanographic habitat revealed using citizen science data from a climate change hotspot","The Bari Manifesto: An interoperability framework for essential biodiversity variables","Contribution of citizen science towards international biodiversity monitoring","The use of replacement cost method to assess and manage the impacts of water resource development on Australian indigenous customary economies","Plant and fungal collections: Current status, future perspectives","Alpha e-taxonomy: responses from the systematics community to the biodiversity crisis","Marine Biodiversity in the Australian Region","Integrating biodiversity distribution knowledge: toward a global map of life","Science, biodiversity and Australian management of marine ecosystems","Improving biodiversity monitoring","Risk Levels of Invasive Fusarium oxysporum f. sp. in Areas Suitable for Date Palm (Phoenix dactylifera) Cultivation under Various Climate Change Projections","Risks of Biological Invasion on the Belt and Road","Big data of tree species distributions: how big and how good?","Aligning citizen science with best practice: Threatened species conservation in Australia","Relationships of the New Guinean subspecies, Gymnorhina tibicen papuana, of the Australian Magpie: an assessment from DNA sequence data","On the Biogeography of Centipeda: A Species-Tree Diffusion Approach","Regional and seasonal variation in airborne grass pollen levels between cities of Australia and New Zealand","Passive eDNA collection enhances aquatic biodiversity analysis","The Geographic Distribution, Venom Components, Pathology and Treatments of Stonefish (Synanceia spp.) Venom","Limited understanding of bushfire impacts on Australian invertebrates","Preempting the Arrival of the Brown Marmorated Stink Bug, Halyomorpha halys: Biological Control Options for Australia","A Comprehensive Overview of Technologies for Species and Habitat Monitoring and Conservation","The number of tree species on Earth","Thirty-eight years of CO&amp;lt;sub&amp;gt;2&amp;lt;/sub&amp;gt; fertilization has outpaced growing aridity to drive greening of Australian woody ecosystems"],["Philosophical Transactions of the Royal Society B: Biological Sciences","Diversity &amp; Distributions","Frontiers in Marine Science","Nature Ecology &amp; Evolution","Conservation Letters","Ecography","Scientific Data","Ecology Letters","Ecology and Evolution","PLOS ONE","Ecosystems","Ecology","ZOOTAXA","Fire","Information Systems Research","Diversity &amp; Distributions","NeoBiota","Science of the Total Environment","Journal of Sustainable Forestry","PLOS ONE","Ecological Management &amp; Restoration","Aquatic Invasions","Methods in Ecology and Evolution","Ecological Indicators","PLANTS, PEOPLE, PLANET","Microbiology Australia","Land Use Policy","Agricultural &amp; Forest Meteorology","Acta Oecologica","PLOS ONE","PLOS ONE","Journal of Mammalogy","Journal of Biogeography","Ecography","Global Ecology &amp; Biogeography","Restoration Ecology","Journal of Ecology","Forest Ecology &amp; Management","Ecological Modelling","BioScience","PLOS ONE","One Health","Methods in Ecology and Evolution","Journal of Applied Ecology","Biological Journal of the Linnean Society","Biological Journal of the Linnean Society","Biodiversity","Australian Zoologist","Austral Ecology","Journal of Applied Ecology","Theoretical &amp; Applied Climatology","Citizen Science: Theory and Practice","Global Change Biology","Ecological Informatics","Biological Conservation","Journal of Environmental Management","PLANTS, PEOPLE, PLANET","Kew Bulletin","PLOS ONE","Trends in Ecology &amp; Evolution","Ocean &amp; Coastal Management","Austral Ecology","PLOS ONE","Current Biology","Forest Ecosystems","Conservation Science and Practice","Emu","Systematic Biology","Aerobiologia","Communications Biology","Marine Drugs","Insect Conservation &amp; Diversity","Insects","BioScience","Proceedings of the National Academy of Sciences of the United States of America","Biogeosciences"],["Lindsay P. Campbell","Jane Elith","John Veron","Ayesha I. T. Tulloch","Amy L. Whitehead","Steven J. Phillips","Shyama Pagad","Antoine Guisan","Amber G. F. Teacher","John Wieczorek","Katherine E. Selwood","Manuel Delgado‐Baquerizo","ADAM BROADLEY","Jessica Lucas","Roman Lukyanenko","Trevor H. Booth","John Hutchinson","A.J.J. Lynch","E. Beech","Paul D. N. Hebert","Trevor H. Booth",null,"Res Altwegg","Farzin Shabani","Manuel Esperon‐Rodriguez",null,"B. Alexander Simmons","Nadiezhda Yakovleva Zitz Ramirez-Cabral","Philippe Borsa","Chris Taylor","Geoff Pegg","Bronwyn A. Hradsky","Cassandra S. James","Si-Chong Chen","Alejandra Morán-Ordóñez","Tein McDonald","Jane A. Catford","Tongli Wang","Linda J. Beaumont","Linda M. Broadhurst","Rebecca Mary Bernadette Harris","Gary Crameri","William Fithian","Jennifer Firn",null,"Tom August","W. Daniel Kissling",null,"Glenda M. Wardle","Katherine E. Selwood","Nadiezhda Yakovleva Zitz Ramirez-Cabral","Susanne Hecker","Curtis Champion","Alex R. Hardisty","Mark Chandler","Sue Jackson","Alan Paton","S. J. Mayo","Alan J. Butler","Walter Jetz","Richard Kenchington","DAVID B. LINDENMAYER","Farzin Shabani","Xuan Liu","Josep M. Serra-Diaz","Rochelle Steven","Alicia Toon","S. Nylinder","Danielle E. Medek","Cindy Bessey","Silvia L. Saggiomo","Manu E. Saunders","Valerie Caron","José J Lahoz-Monfort","Roberto Cazzolla Gatti","Sami W. Rifai"],[2015,2010,2015,2018,2016,2017,2018,2013,2013,2012,2016,2018,2018,2021,2014,2013,2014,2015,2017,2013,2012,2019,2019,2020,2019,2019,2018,2016,2016,2017,2017,2017,2017,2016,2016,2016,2016,2016,2016,2015,2015,2015,2014,2015,2015,2015,2015,2015,2015,2017,2018,2018,2018,2019,2017,2014,2020,2008,2010,2012,2012,2011,2013,2019,2018,2019,2017,2013,2015,2021,2021,2021,2021,2021,2022,2022],["10.1098/rstb.2014.0135","10.1111/j.1472-4642.2010.00725.x","10.3389/fmars.2014.00081","10.1038/s41559-018-0608-1","10.1111/conl.12260","10.1111/ecog.03049","10.1038/sdata.2017.202","10.1111/ele.12189","10.1002/ece3.888","10.1371/journal.pone.0029715","10.1007/s10021-016-0059-y","10.1002/ecy.2137","10.11646/zootaxa.4415.2.1","10.3390/fire4010001","10.1287/isre.2014.0537","10.1111/ddi.12144","10.3897/neobiota.23.7745","10.1016/j.scitotenv.2015.04.100","10.1080/10549811.2017.1310049","10.1371/journal.pone.0068535","10.1111/emr.12000","10.3391/ai.2019.14.3.10","10.1111/2041-210x.13090","10.1016/j.ecolind.2020.106436","10.1002/ppp3.10064","10.1071/ma19038","10.1016/j.landusepol.2018.03.049","10.1016/j.agrformet.2015.12.002","10.1016/j.actao.2015.10.009","10.1371/journal.pone.0169629","10.1371/journal.pone.0188058","10.1093/jmammal/gyx010","10.1111/jbi.12926","10.1111/ecog.02010","10.1111/geb.12545","10.1111/rec.12359","10.1111/1365-2745.12627","10.1016/j.foreco.2015.08.004","10.1016/j.ecolmodel.2016.10.004","10.1093/biosci/biv155","10.1371/journal.pone.0124014","10.1016/j.onehlt.2015.10.003","10.1111/2041-210x.12242","10.1111/1365-2664.12500","10.1111/bij.12516","10.1111/bij.12534","10.1080/14888386.2015.1068709","10.7882/az.2015.014","10.1111/aec.12265","10.1111/1365-2664.13052","10.1007/s00704-018-2385-9","10.5334/cstp.114","10.1111/gcb.14398","10.1016/j.ecoinf.2018.11.003","10.1016/j.biocon.2016.09.004","10.1016/j.jenvman.2014.01.018","10.1002/ppp3.10141","10.1007/s12225-008-9014-1","10.1371/journal.pone.0011831","10.1016/j.tree.2011.09.007","10.1016/j.ocecoaman.2012.08.009","10.1111/j.1442-9993.2011.02314.x","10.1371/journal.pone.0083404","10.1016/j.cub.2018.12.036","10.1186/s40663-017-0120-0","10.1111/csp2.100","10.1080/01584197.2017.1324249","10.1093/sysbio/syt102","10.1007/s10453-015-9399-x","10.1038/s42003-021-01760-8","10.3390/md19060302","10.1111/icad.12493","10.3390/insects12070581","10.1093/biosci/biab073","10.1073/pnas.2115329119","10.5194/bg-19-491-2022"],[3,3,3,2,2,2,2,2,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_policies_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Twitter (some high numbers here)</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_tweeters_count, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
 208  132  101   90   68   40   47   40   41   42   31   22   32   30   25   18 
  17   18   19   20   21   22   23   24   25   26   27   28   29   30   31   32 
  20   19   19   14   17   11    8   16   11    8   10   14   12   10    8    7 
  33   34   35   36   37   38   39   40   41   42   43   44   45   46   47   48 
  11    6    3    3    4   10    5    4   10    2    5    2    7    2    4    6 
  49   50   51   52   53   55   56   57   58   59   62   63   64   65   66   67 
   7    3    3    3    6    5    6    4    4    2    3    2    2    1    2    1 
  68   69   71   72   73   74   75   76   77   79   80   82   83   85   86   87 
   4    1    2    3    2    3    1    2    1    1    1    2    1    3    1    3 
  90   93   95   98  100  101  102  103  107  111  112  113  114  117  123  124 
   1    1    2    1    1    1    1    1    1    2    1    1    1    2    1    1 
 131  133  134  137  141  146  148  150  152  156  174  181  186  190  198  204 
   1    1    2    1    1    1    1    1    1    1    1    1    1    1    1    1 
 219  243  276  322  323  364  390  453  598  618 1456 &lt;NA&gt; 
   1    1    1    1    1    1    1    1    1    1    1  132 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_tweeters_count)) <span class="co"># 1387 articles mentioned at least once</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1387</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_tweeters_count, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="co">#28730 total numbers of mentions</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 28730</code></pre>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with cited_by_tweeters_count &gt;500</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_tweeters_count) <span class="sc">&gt;</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_tweeters_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_tweeters_count)) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles with Tweeter counts &gt; 100"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-49463c5815f529e11bd0" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-49463c5815f529e11bd0">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles with Tweeter counts &gt; 100<\/caption>","data":[["The number of tree species on Earth","Birds, butterflies and flowers in the tropics are not more colourful than those at higher latitudes","Striped bodypainting protects against horseflies","A new species of bandy-bandy (Vermicella: Serpentes: Elapidae) from the Weipa region, Cape York, Australia","Global Patterns and Drivers of Bee Distribution","Loss of vocal culture and fitness costs in a critically endangered songbird","Opening the black box: an open-source release of Maxent","A decision tree for assessing the risks and benefits of publishing biodiversity data","Rights, interests and expectations: Indigenous perspectives on unrestricted access to genomic data","Museum genomics reveals the rapid decline and extinction of Australian rodents since European settlement","Phylogenomics of Monitor Lizards and the Role of Competition in Dictating Body Size Disparity","Data integration enables global biodiversity synthesis","Dispersal of the introduced red fox (Vulpes vulpes) across Australia","Towards global data products of Essential Biodiversity Variables on species traits","When cities are the last chance for saving species","Widespread vulnerability of flowering plant seed production to pollinator declines","GlobalTreeSearch: The first complete global database of tree species and country distributions","Botanical Monography in the Anthropocene","Global priorities for an effective information basis of biodiversity distributions","Counting the bodies: Estimating the numbers and spatial variation of Australian reptiles, birds and mammals killed by two invasive mesopredators","Capturing Natural-Colour 3D Models of Insects for Species Discovery and Diagnostics","Australia’s continental-scale acoustic tracking database and its automated quality control process","Reintroduction of Tasmanian devils to mainland Australia can restore top-down control in ecosystems where dingoes have been extirpated","Introducing the Global Register of Introduced and Invasive Species","A lack of genetically compatible mates constrains the spread of an invasive weed","A New Genus and Species of Pygmy Pipehorse from Taitokerau Northland, Aotearoa New Zealand, with a Redescription of Acentronura Kaup, 1853 and Idiotropiscis Whitley, 1947 (Teleostei, Syngnathidae)","Red hot frogs: identifying the Australian frogs most at risk of extinction","Contribution of citizen science towards international biodiversity monitoring","How to publish a new fungal species, or name, version 3.0","Collections-based science in the 21st Century","World Flora Online: Placing taxonomists at the heart of a definitive and comprehensive global resource on the world's plants","Reflection of near-infrared light confers thermal protection in birds","Animal-Borne Telemetry: An Integral Component of the Ocean Observing Toolkit","From matte banded to glossy black: structures underlying colour change in the caudal lures of southern death adders (Acanthophis antarcticus, Reptilia: Elapidae)","A high-throughput multispectral imaging system for museum specimens","A Guide to Public Engagement for Entomological Collections and Natural History Museums in the Age of Social Media","A Comprehensive Overview of Technologies for Species and Habitat Monitoring and Conservation","Contribution to the knowledge of the endemic Australian genus Binburrum Pollock, 1995 (Coleoptera: Pyrochroidae: Pilipalpinae), with description of three new species","Mystery pufferfish create elaborate circular nests at mesophotic depths in Australia","Targeted gene flow and rapid adaptation in an endangered marsupial","A review of the heterogeneous landscape of biodiversity databases: Opportunities and challenges for a synthesized biodiversity knowledge base"],["Proceedings of the National Academy of Sciences of the United States of America","Global Ecology &amp; Biogeography","Royal Society Open Science","ZOOTAXA","Current Biology","Proceedings of the Royal Society B: Biological Sciences","Ecography","Nature Ecology &amp; Evolution","Nature Reviews Genetics","Proceedings of the National Academy of Sciences of the United States of America","Systematic Biology","Proceedings of the National Academy of Sciences of the United States of America","Biological Invasions","Nature Ecology &amp; Evolution","Frontiers in Ecology &amp; the Environment","Science Advances","Journal of Sustainable Forestry","Trends in Plant Science","Nature Communications","Diversity &amp; Distributions","PLOS ONE","Scientific Data","Biological Conservation","Scientific Data","New Phytologist","Ichthyology &amp; Herpetology","Pacific Conservation Biology","Biological Conservation","IMA Fungus","Journal of Systematics &amp; Evolution","Taxon","Nature Communications","Frontiers in Marine Science","Biological Journal of the Linnean Society","Communications Biology","Annals of the Entomological Society of America","BioScience","Canadian Entomologist","Journal of Fish Biology","Conservation Biology","Global Ecology &amp; Biogeography"],["Roberto Cazzolla Gatti","Rhiannon L. Dalrymple","Gábor Horváth","CHANTELLE M. DEREZ","Michael C. Orr","Ross Crates","Steven J. Phillips","Ayesha I. T. Tulloch","Maui Hudson","Emily Roycroft","Ian G Brennan","J. Mason Heberling",null,"W. Daniel Kissling","Kylie Soanes","James G. Rodger","E. Beech","Olwen M. Grace","Carsten Meyer","Alyson M. Stobo‐Wilson","Chuong V. Nguyen","Xavier Hoenner","Daniel O. Hunter","Shyama Pagad","Akane Uesugi","Graham A. Short","Hayley M. Geyle","Mark Chandler","Aime, M. Catherine",null,"Thomas Borsch","Iliana Medina","Rob Harcourt","Jenna M Crowe-Riddell","Wei-Ping Chan","Bryan D. Lessard","José J Lahoz-Monfort","Yun Hsiao","Todd Bond","Ella Kelly","Xiao Feng"],[2022,2015,2019,2018,2021,2021,2017,2018,2020,2021,2020,2021,2018,2018,2019,2021,2017,2021,2015,2022,2014,2018,2015,2018,2020,2021,2021,2017,2021,2018,2020,2018,2019,2021,2022,2017,2021,2020,2020,2018,2022],["10.1073/pnas.2115329119","10.1111/geb.12368","10.1098/rsos.181325","10.11646/zootaxa.4446.1.1","10.1016/j.cub.2020.10.053","10.1098/rspb.2021.0225","10.1111/ecog.03049","10.1038/s41559-018-0608-1","10.1038/s41576-020-0228-x","10.1073/pnas.2021390118","10.1093/sysbio/syaa046","10.1073/pnas.2018093118","10.1007/s10530-018-1897-7","10.1038/s41559-018-0667-3","10.1002/fee.2032","10.1126/sciadv.abd3524","10.1080/10549811.2017.1310049","10.1016/j.tplants.2020.12.018","10.1038/ncomms9221","10.1111/ddi.13497","10.1371/journal.pone.0094346","10.1038/sdata.2017.206","10.1016/j.biocon.2015.07.030","10.1038/sdata.2017.202","10.1111/nph.16496","10.1643/i2020136","10.1071/pc21019","10.1016/j.biocon.2016.09.004","10.1186/s43008-021-00063-1","10.1111/jse.12315","10.1002/tax.12373","10.1038/s41467-018-05898-8","10.3389/fmars.2019.00326","10.1093/biolinnean/blaa218","10.1038/s42003-022-04282-z","10.1093/aesa/sax058","10.1093/biosci/biab073","10.4039/tce.2020.74","10.1111/jfb.14506","10.1111/cobi.13149","10.1111/geb.13497"],[1456,618,598,453,390,364,323,322,276,243,219,204,198,190,186,181,174,156,152,150,148,146,141,137,134,134,133,131,124,123,117,117,114,113,112,111,111,107,103,102,101]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_tweeters_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Facebook</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_fbwalls_count, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   14   16 &lt;NA&gt; 
 276   96   30   22   11    3    5    3    1    2    1    1    1 1067 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_fbwalls_count)) <span class="co">#452 articles mentioned at least once</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 452</code></pre>
</div>
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_fbwalls_count, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="co">#848 total numbers of mentions</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 848</code></pre>
</div>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with cited_by_fbwalls_count &gt;10</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_fbwalls_count) <span class="sc">&gt;</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_fbwalls_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_fbwalls_count)) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles with Facebook counts &gt; 10"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-a8648bd48fa4bd544445" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-a8648bd48fa4bd544445">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles with Facebook counts &gt; 10<\/caption>","data":[["GlobalTreeSearch: The first complete global database of tree species and country distributions","Introducing the Global Register of Introduced and Invasive Species","Contribution of citizen science towards international biodiversity monitoring"],["Journal of Sustainable Forestry","Scientific Data","Biological Conservation"],["E. Beech","Shyama Pagad","Mark Chandler"],[2017,2018,2017],["10.1080/10549811.2017.1310049","10.1038/sdata.2017.202","10.1016/j.biocon.2016.09.004"],[16,14,11]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_fbwalls_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Number of blogs that have mentioned the publication</span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_feeds_count, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    2    3    4    5    6    7    8    9   11   12   16   17 &lt;NA&gt; 
 223   69   17   11   11    6    1    1    1    4    1    1    1 1172 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_feeds_count)) <span class="co">#347 mentioned at least once</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 347</code></pre>
</div>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_feeds_count, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="co">#660 total numbers of mentions</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 660</code></pre>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with cited_by_feeds_count &gt;10</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_feeds_count) <span class="sc">&gt;</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_feeds_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_feeds_count)) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles with blog feeds counts &gt; 10"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-f587906992c14f9f80b3" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-f587906992c14f9f80b3">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles with blog feeds counts &gt; 10<\/caption>","data":[["The number of tree species on Earth","Global Patterns and Drivers of Bee Distribution","Urbanisation at Multiple Scales Is Associated with Larger Size and Higher Fecundity of an Orb-Weaving Spider","Striped bodypainting protects against horseflies","A stitch in time – Synergistic impacts to platypus metapopulation extinction risk","Creative Commons licenses and the non-commercial condition: Implications for the re-use of biodiversity information","Missing for almost 100 years: the rare and potentially threatened bee, Pharohylaeus lactiferus (Hymenoptera, Colletidae)"],["Proceedings of the National Academy of Sciences of the United States of America","Current Biology","PLOS ONE","Royal Society Open Science","Biological Conservation","Zookeys","Journal of Hymenoptera Research"],["Roberto Cazzolla Gatti","Michael C. Orr","Elizabeth C. Lowe","Gábor Horváth","Gilad Bino","Gregor Hagedorn",null],[2022,2021,2014,2019,2020,2011,2021],["10.1073/pnas.2115329119","10.1016/j.cub.2020.10.053","10.1371/journal.pone.0105480","10.1098/rsos.181325","10.1016/j.biocon.2019.108399","10.3897/zookeys.150.2189","10.3897/jhr.81.59365"],[17,16,12,11,11,11,11]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_feeds_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="do">##News sources that have mentioned the publication</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_msm_count, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
  76   36   25   21   21   22   10    9    5    7    7    8    2    4    1    3 
  17   18   21   22   23   24   26   27   29   39   40   42   43   46   47   58 
   1    1    3    1    1    2    1    2    2    1    1    2    1    1    1    1 
  74   91   92   99  111  113  126  130  164  177  261 &lt;NA&gt; 
   1    2    1    1    1    1    1    1    1    1    1 1228 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_msm_count)) <span class="co">#291 mentioned at least once</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 291</code></pre>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_msm_count, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="co">#3334 total numbers of mentions</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3334</code></pre>
</div>
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with cited_by_msm_count &gt;100</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_msm_count) <span class="sc">&gt;</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_msm_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_msm_count)) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles with news sources counts &gt; 10"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-a7e10073793ebe74773e" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-a7e10073793ebe74773e">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles with news sources counts &gt; 10<\/caption>","data":[["The number of tree species on Earth","Loss of vocal culture and fitness costs in a critically endangered songbird","Museum genomics reveals the rapid decline and extinction of Australian rodents since European settlement","The platypus: evolutionary history, biology, and an uncertain future","Evidence for a complex evolutionary history of mound building in the Australian nasute termites (Nasutitermitinae)","Morphometric comparisons and novel observations of diurnal and low-light-foraging bees","Counting the bodies: Estimating the numbers and spatial variation of Australian reptiles, birds and mammals killed by two invasive mesopredators"],["Proceedings of the National Academy of Sciences of the United States of America","Proceedings of the Royal Society B: Biological Sciences","Proceedings of the National Academy of Sciences of the United States of America","Journal of Mammalogy","Biological Journal of the Linnean Society","Journal of Hymenoptera Research","Diversity &amp; Distributions"],["Roberto Cazzolla Gatti","Ross Crates","Emily Roycroft","Gilad Bino","Perry G Beasley-Hall","James B. Dorey","Alyson M. Stobo‐Wilson"],[2022,2021,2021,2019,2018,2020,2022],["10.1073/pnas.2115329119","10.1098/rspb.2021.0225","10.1073/pnas.2021390118","10.1093/jmammal/gyz058","10.1093/biolinnean/bly187","10.3897/jhr.79.57308","10.1111/ddi.13497"],[261,177,164,130,126,113,111]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_msm_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Wikipedia</span></span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_wikipedia_count, <span class="at">useNA =</span> <span class="st">"always"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   14   19   41   77 
 164   52   22   13    9    4    1    2    2    2    2    1    1    1    1    1 
  96  131 &lt;NA&gt; 
   1    1 1239 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_wikipedia_count)) <span class="co">#280 cited at least once</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 280</code></pre>
</div>
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_wikipedia_count, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="co">#928 total numbers of mentions</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 928</code></pre>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with cited_by_wikipedia_count &gt;10</span></span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_wikipedia_count) <span class="sc">&gt;</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_wikipedia_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_wikipedia_count)) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles with Wikipedia counts &gt; 10"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-7dc6285b0c7563b655bc" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-7dc6285b0c7563b655bc">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles with Wikipedia counts &gt; 10<\/caption>","data":[["A monograph of Daviesia (Mirbelieae, Faboideae, Fabaceae)","Muelleria","Systematic dismantlement ofLichenostomusimproves the basis for understanding relationships within the honeyeaters (Meliphagidae) and the historical development of Australo-Papuan bird communities","The Global Urban Tree Inventory: A database of the diverse tree flora that inhabits the world’s cities","Global Diversity of Sponges (Porifera)","Deep-reef fish assemblages of the Great Barrier Reef shelf-break (Australia)","Museum genomics reveals the rapid decline and extinction of Australian rodents since European settlement","Evolutionary history of birds across southern Australia: structure, history and taxonomic implications of mitochondrial DNA diversity in an ecologically diverse suite of species","Twentieth century occurrence of the Long-Beaked Echidna Zaglossus bruijnii in the Kimberley region of Australia"],["Phytotaxa","Biodiversity Heritage Library","Emu","Global Ecology &amp; Biogeography","PLOS ONE","Scientific Reports","Proceedings of the National Academy of Sciences of the United States of America","Emu","Zookeys"],["MICHAEL D. CRISP","Royal Botanic Gardens (Vic.)","Árpád S. Nyári","Alessandro Ossola","Rob W. M. Van Soest","Tiffany L. Sih","Emily Roycroft","Gaynor Dolman","Kristofer M. Helgen"],[2017,1989,2016,2020,2012,2017,2021,2016,2012],["10.11646/phytotaxa.300.1.1","10.5962/bhl.title.112965","10.1071/mu10047","10.1111/geb.13169","10.1371/journal.pone.0035105","10.1038/s41598-017-11452-1","10.1073/pnas.2021390118","10.1071/mu14047","10.3897/zookeys.255.3774"],[131,96,77,41,19,14,12,11,11]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_wikipedia_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="do">##The sum of all "cited_by" entries (profiles per data source)</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Altmetrics.results_df<span class="sc">$</span>cited_by_accounts_count, <span class="at">useNA =</span> <span class="st">"always"</span>) <span class="co">#across all above+</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
 229  140   98   98   65   56   52   47   42   38   30   32   35   29   24   18 
  17   18   19   20   21   22   23   24   25   26   27   28   29   30   31   32 
  26   23   19   18   18   14   11   12   17   12    8   15   10   12    7   10 
  33   34   35   36   37   38   39   40   41   42   43   44   45   46   47   48 
   7    6    4    3    8    4    7    6    6    4    3    5    7    4    2    7 
  49   50   51   52   53   54   55   56   57   58   59   60   61   62   63   64 
   7    5    5    2    6    5    4    3    6    6    3    5    1    1    4    1 
  65   66   67   68   69   70   71   72   73   74   75   76   77   78   81   84 
   1    3    1    2    2    3    2    1    3    2    3    1    5    3    1    1 
  85   86   87   89   90   91   93   94   96   98   99  100  102  103  108  109 
   1    2    1    1    2    1    1    1    1    1    2    1    1    2    1    2 
 110  111  112  113  114  118  119  120  121  123  126  129  130  131  134  141 
   1    1    3    1    1    1    1    2    1    1    1    2    1    2    1    1 
 143  144  145  156  157  158  159  163  165  168  170  179  198  200  202  211 
   1    1    1    1    1    1    1    1    1    2    1    1    1    1    1    1 
 224  227  264  280  301  338  431  491  506  551  628  663 1752 &lt;NA&gt; 
   1    1    1    1    1    2    1    1    1    1    1    1    1    0 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="do">##table of articles with cited_by_accounts_count &gt;10</span></span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">as.numeric</span>(cited_by_accounts_count) <span class="sc">&gt;</span> <span class="dv">100</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(title, journal, authors1, year, doi, cited_by_accounts_count) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(cited_by_accounts_count)) <span class="sc">%&gt;%</span> DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="at">rownames =</span> <span class="cn">FALSE</span>, <span class="at">width =</span> <span class="st">"100%"</span>, <span class="at">options =</span> <span class="fu">list</span>(<span class="at">dom =</span> <span class="st">'t'</span>, <span class="at">scrollY =</span> <span class="st">'800px'</span>, <span class="at">pageLength =</span> <span class="dv">20</span>), <span class="at">caption =</span> <span class="st">"Table of Altmetrics for articles with all mentions counts &gt; 10"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<div id="htmlwidget-e560352a4cf3578c8da4" style="width:100%;height:auto;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-e560352a4cf3578c8da4">{"x":{"filter":"none","vertical":false,"caption":"<caption>Table of Altmetrics for articles with all mentions counts &gt; 10<\/caption>","data":[["The number of tree species on Earth","Striped bodypainting protects against horseflies","Birds, butterflies and flowers in the tropics are not more colourful than those at higher latitudes","Loss of vocal culture and fitness costs in a critically endangered songbird","A new species of bandy-bandy (Vermicella: Serpentes: Elapidae) from the Weipa region, Cape York, Australia","Global Patterns and Drivers of Bee Distribution","Museum genomics reveals the rapid decline and extinction of Australian rodents since European settlement","A decision tree for assessing the risks and benefits of publishing biodiversity data","Opening the black box: an open-source release of Maxent","GlobalTreeSearch: The first complete global database of tree species and country distributions","Rights, interests and expectations: Indigenous perspectives on unrestricted access to genomic data","Counting the bodies: Estimating the numbers and spatial variation of Australian reptiles, birds and mammals killed by two invasive mesopredators","Phylogenomics of Monitor Lizards and the Role of Competition in Dictating Body Size Disparity","Data integration enables global biodiversity synthesis","Widespread vulnerability of flowering plant seed production to pollinator declines","Towards global data products of Essential Biodiversity Variables on species traits","When cities are the last chance for saving species","Dispersal of the introduced red fox (Vulpes vulpes) across Australia","The platypus: evolutionary history, biology, and an uncertain future","Introducing the Global Register of Introduced and Invasive Species","Reintroduction of Tasmanian devils to mainland Australia can restore top-down control in ecosystems where dingoes have been extirpated","Global priorities for an effective information basis of biodiversity distributions","Capturing Natural-Colour 3D Models of Insects for Species Discovery and Diagnostics","A stitch in time – Synergistic impacts to platypus metapopulation extinction risk","Contribution of citizen science towards international biodiversity monitoring","Red hot frogs: identifying the Australian frogs most at risk of extinction","Botanical Monography in the Anthropocene","Australia’s continental-scale acoustic tracking database and its automated quality control process","A New Genus and Species of Pygmy Pipehorse from Taitokerau Northland, Aotearoa New Zealand, with a Redescription of Acentronura Kaup, 1853 and Idiotropiscis Whitley, 1947 (Teleostei, Syngnathidae)","Morphometric comparisons and novel observations of diurnal and low-light-foraging bees","Urbanisation at Multiple Scales Is Associated with Larger Size and Higher Fecundity of an Orb-Weaving Spider","A monograph of Daviesia (Mirbelieae, Faboideae, Fabaceae)","A lack of genetically compatible mates constrains the spread of an invasive weed","Few changes in native Australian alpine plant morphology, despite substantial local climate change","Integrating Biodiversity Infrastructure into Pathogen Discovery and Mitigation of Emerging Infectious Diseases","Collections-based science in the 21st Century","Evidence for a complex evolutionary history of mound building in the Australian nasute termites (Nasutitermitinae)","Contribution to the knowledge of the endemic Australian genus Binburrum Pollock, 1995 (Coleoptera: Pyrochroidae: Pilipalpinae), with description of three new species","How to publish a new fungal species, or name, version 3.0","Species distribution models derived from citizen science data predict the fine scale movements of owls in an urbanizing landscape","World Flora Online: Placing taxonomists at the heart of a definitive and comprehensive global resource on the world's plants","Risks of Biological Invasion on the Belt and Road","Reflection of near-infrared light confers thermal protection in birds","Targeted gene flow and rapid adaptation in an endangered marsupial","Animal-Borne Telemetry: An Integral Component of the Ocean Observing Toolkit","A Guide to Public Engagement for Entomological Collections and Natural History Museums in the Age of Social Media","From matte banded to glossy black: structures underlying colour change in the caudal lures of southern death adders (Acanthophis antarcticus, Reptilia: Elapidae)","Implications of the 2019–2020 megafires for the biogeography and conservation of Australian vegetation","A Comprehensive Overview of Technologies for Species and Habitat Monitoring and Conservation","A high-throughput multispectral imaging system for museum specimens","Active acoustic tracking suggests that soft sediment fishes can show site attachment: a preliminary assessment of the movement patterns of the blue-spotted flathead (Platycephalus caeruleopunctatus)","Mystery pufferfish create elaborate circular nests at mesophotic depths in Australia","A continental-scale analysis of feral cat diet in Australia","High exposure of global tree diversity to human pressure","Multiple trans-Torres Strait colonisations by tree frogs in the","Topographic ruggedness and rainfall mediate geographic range contraction of a threatened marsupial predator","Creative Commons licenses and the non-commercial condition: Implications for the re-use of biodiversity information","A review of the heterogeneous landscape of biodiversity databases: Opportunities and challenges for a synthesized biodiversity knowledge base"],["Proceedings of the National Academy of Sciences of the United States of America","Royal Society Open Science","Global Ecology &amp; Biogeography","Proceedings of the Royal Society B: Biological Sciences","ZOOTAXA","Current Biology","Proceedings of the National Academy of Sciences of the United States of America","Nature Ecology &amp; Evolution","Ecography","Journal of Sustainable Forestry","Nature Reviews Genetics","Diversity &amp; Distributions","Systematic Biology","Proceedings of the National Academy of Sciences of the United States of America","Science Advances","Nature Ecology &amp; Evolution","Frontiers in Ecology &amp; the Environment","Biological Invasions","Journal of Mammalogy","Scientific Data","Biological Conservation","Nature Communications","PLOS ONE","Biological Conservation","Biological Conservation","Pacific Conservation Biology","Trends in Plant Science","Scientific Data","Ichthyology &amp; Herpetology","Journal of Hymenoptera Research","PLOS ONE","Phytotaxa","New Phytologist","Ecology and Evolution","BioScience","Journal of Systematics &amp; Evolution","Biological Journal of the Linnean Society","Canadian Entomologist","IMA Fungus","Biological Conservation","Taxon","Current Biology","Nature Communications","Conservation Biology","Frontiers in Marine Science","Annals of the Entomological Society of America","Biological Journal of the Linnean Society","Nature Communications","BioScience","Communications Biology","Animal Biotelemetry","Journal of Fish Biology","Journal of Biogeography","Proceedings of the National Academy of Sciences of the United States of America","Australian Journal of Zoology","Diversity &amp; Distributions","Zookeys","Global Ecology &amp; Biogeography"],["Roberto Cazzolla Gatti","Gábor Horváth","Rhiannon L. Dalrymple","Ross Crates","CHANTELLE M. DEREZ","Michael C. Orr","Emily Roycroft","Ayesha I. T. Tulloch","Steven J. Phillips","E. Beech","Maui Hudson","Alyson M. Stobo‐Wilson","Ian G Brennan","J. Mason Heberling","James G. Rodger","W. Daniel Kissling","Kylie Soanes",null,"Gilad Bino","Shyama Pagad","Daniel O. Hunter","Carsten Meyer","Chuong V. Nguyen","Gilad Bino","Mark Chandler","Hayley M. Geyle","Olwen M. Grace","Xavier Hoenner","Graham A. Short","James B. Dorey","Elizabeth C. Lowe","MICHAEL D. CRISP","Akane Uesugi","Meena S. Sritharan","Joseph A Cook",null,"Perry G Beasley-Hall","Yun Hsiao","Aime, M. Catherine","Nick Bradsworth","Thomas Borsch","Xuan Liu","Iliana Medina","Ella Kelly","Rob Harcourt","Bryan D. Lessard","Jenna M Crowe-Riddell","Robert C. Godfree","José J Lahoz-Monfort","Wei-Ping Chan","Lachlan C. Fetterplace","Todd Bond","Tim S. Doherty","Wen-Yong Guo","Paul M. Oliver","Harry A. Moore","Gregor Hagedorn","Xiao Feng"],[2022,2019,2015,2021,2018,2021,2021,2018,2017,2017,2020,2022,2020,2021,2021,2018,2019,2018,2019,2018,2015,2015,2014,2020,2017,2021,2021,2018,2021,2020,2014,2017,2020,2021,2020,2018,2018,2020,2021,2017,2020,2019,2018,2018,2019,2017,2021,2021,2021,2022,2016,2020,2015,2022,2021,2019,2011,2022],["10.1073/pnas.2115329119","10.1098/rsos.181325","10.1111/geb.12368","10.1098/rspb.2021.0225","10.11646/zootaxa.4446.1.1","10.1016/j.cub.2020.10.053","10.1073/pnas.2021390118","10.1038/s41559-018-0608-1","10.1111/ecog.03049","10.1080/10549811.2017.1310049","10.1038/s41576-020-0228-x","10.1111/ddi.13497","10.1093/sysbio/syaa046","10.1073/pnas.2018093118","10.1126/sciadv.abd3524","10.1038/s41559-018-0667-3","10.1002/fee.2032","10.1007/s10530-018-1897-7","10.1093/jmammal/gyz058","10.1038/sdata.2017.202","10.1016/j.biocon.2015.07.030","10.1038/ncomms9221","10.1371/journal.pone.0094346","10.1016/j.biocon.2019.108399","10.1016/j.biocon.2016.09.004","10.1071/pc21019","10.1016/j.tplants.2020.12.018","10.1038/sdata.2017.206","10.1643/i2020136","10.3897/jhr.79.57308","10.1371/journal.pone.0105480","10.11646/phytotaxa.300.1.1","10.1111/nph.16496","10.1002/ece3.7392","10.1093/biosci/biaa064","10.1111/jse.12315","10.1093/biolinnean/bly187","10.4039/tce.2020.74","10.1186/s43008-021-00063-1","10.1016/j.biocon.2017.06.039","10.1002/tax.12373","10.1016/j.cub.2018.12.036","10.1038/s41467-018-05898-8","10.1111/cobi.13149","10.3389/fmars.2019.00326","10.1093/aesa/sax058","10.1093/biolinnean/blaa218","10.1038/s41467-021-21266-5","10.1093/biosci/biab073","10.1038/s42003-022-04282-z","10.1186/s40317-016-0107-6","10.1111/jfb.14506","10.1111/jbi.12469","10.1073/pnas.2026733119","10.1071/zo20071","10.1111/ddi.12982","10.3897/zookeys.150.2189","10.1111/geb.13497"],[1752,663,628,551,506,491,431,338,338,301,280,264,227,224,211,202,200,198,179,170,168,168,165,163,159,158,157,156,145,144,143,141,134,131,131,130,129,129,126,123,121,120,120,119,118,114,113,112,112,112,111,110,109,109,108,103,103,102]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>title<\/th>\n      <th>journal<\/th>\n      <th>authors1<\/th>\n      <th>year<\/th>\n      <th>doi<\/th>\n      <th>cited_by_accounts_count<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"t","scrollY":"800px","pageLength":20,"columnDefs":[{"className":"dt-right","targets":[3,5]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="co">#hist(as.numeric(Altmetrics.results_df$context.journal.pct)) #which was in top 95% of the journal</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a><span class="co">#View(Altmetrics.results_df[Altmetrics.results_df$context.journal.pct &gt;= 95, 1:20] ) #highest score per given journal</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Match altmetrics data to main data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a><span class="co">#match and merge data to dat_articles by DOI</span></span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>dat_articles_A <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dat_articles, Altmetrics.results_df, <span class="at">by =</span> <span class="st">"doi"</span>)</span>
<span id="cb137-3"><a href="#cb137-3" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_csv</span>(dat_articles_A, <span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_A.csv"</span>)) <span class="co">#save merged data frame for plotting later</span></span>
<span id="cb137-4"><a href="#cb137-4" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles_A &lt;- readr::read_csv(file = here("data","dat_articles_A.csv"))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Find articles with highest Altmetric scores.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a>dat_articles_A <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_A.csv"</span>)) <span class="co">#load processed and merged Altmetric data</span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="ot">&lt;-</span> dat_articles_A</span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a><span class="co">#subset highly scoring (&gt;100) publications</span></span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df_highlyscored100 <span class="ot">&lt;-</span> Altmetrics.results_df <span class="sc">%&gt;%</span></span>
<span id="cb138-6"><a href="#cb138-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(score <span class="sc">&gt;</span> <span class="dv">100</span>)</span>
<span id="cb138-7"><a href="#cb138-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Altmetrics.results_df_highlyscored100) <span class="co">#79 articles with &gt;100 score</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  76 385</code></pre>
</div>
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="co">#subset highly scoring (&gt;1000) publications</span></span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df_highlyscored1000 <span class="ot">&lt;-</span> Altmetrics.results_df <span class="sc">%&gt;%</span></span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(score <span class="sc">&gt;</span> <span class="dv">1000</span>)</span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Altmetrics.results_df_highlyscored1000) <span class="co">#4 articles with &gt;1000 score</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]   4 385</code></pre>
</div>
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results_df_highlyscored$journal</span></span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results_df_highlyscored$authors1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>There are 76 articles with &gt;100 Altmetric score.<br>
There are 4 articles with &gt;1000 Altmetric score.</p>
<p>Plot Altmetric scores across article publication years:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">height =</span> <span class="fl">0.1</span>, <span class="at">seed =</span> <span class="dv">1</span>) <span class="co">#used below to add jitter</span></span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> </span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(score)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have no scores</span></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(year)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have no year</span></span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> year, <span class="at">y =</span> score)) <span class="sc">+</span> </span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"#66c2a5"</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>, <span class="at">position =</span> pos) <span class="sc">+</span></span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_smooth(method = "lm", color = "grey") + </span></span>
<span id="cb143-10"><a href="#cb143-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()  <span class="sc">+</span></span>
<span id="cb143-11"><a href="#cb143-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"total Altmetric score"</span>) <span class="sc">+</span></span>
<span id="cb143-12"><a href="#cb143-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"year"</span>) <span class="sc">+</span></span>
<span id="cb143-13"><a href="#cb143-13" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> Altmetrics.results_df_highlyscored1000, <span class="fu">aes</span>(<span class="at">label =</span> journal), <span class="at">size =</span> <span class="dv">3</span>, <span class="at">max.overlaps =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="co">#point labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<p>Plot Altmetric journal context percentile across years as a scatterplot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Altmetrics.results_df$context.journal.pct</span></span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="sc">%&gt;%</span> </span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(year)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have no year</span></span>
<span id="cb144-4"><a href="#cb144-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> year, <span class="at">y =</span> <span class="fu">as.numeric</span>(context.journal.pct))) <span class="sc">+</span></span>
<span id="cb144-5"><a href="#cb144-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.2</span>, <span class="at">position =</span> pos, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span></span>
<span id="cb144-6"><a href="#cb144-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_smooth(method = "lm", color = "grey") + </span></span>
<span id="cb144-7"><a href="#cb144-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb144-8"><a href="#cb144-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"percentile within journal context"</span>) <span class="sc">+</span></span>
<span id="cb144-9"><a href="#cb144-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"year"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="SI_methods_analyses_figures_files/figure-html/unnamed-chunk-62-1.png" class="img-fluid" width="768"></p>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co">#  ggrepel::geom_text_repel(aes(label = journal, color = "grey"), size = 3, max.overlaps = 20) + theme(legend.position = "none") #point labels</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="final-plots" class="level3">
<h3 class="anchored" data-anchor-id="final-plots">Final plots</h3>
<p>The final set of plots for the main manuscript text - the plots themselves are not part of the Supplementary Information, so the code is not automatically evaluated and the figures are not displayed in the html document.<br>
All plots here can be saved as files separate pdf files. Each plot is created primarily from the saved pre-processed data files, so no fetching from other remote sources or external data files is required. Thus, the code in this section can be run independently of all the earlier sections. Data wrangling is only done to produce the final plots in Figures 2 - 9 for the manuscript and data for Figure 1.</p>
<section id="figure-1---counts-of-records-in-living-atlases." class="level4">
<h4 class="anchored" data-anchor-id="figure-1---counts-of-records-in-living-atlases.">Figure 1 - counts of records in Living Atlases.</h4>
<p>Using package galah to gather data used in panel B of the figure.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb146"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="do">##get an overview of the Living Atlases </span></span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="fu">atlas_counts</span>()</span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="fu">show_all</span>(atlases) <span class="co">#11</span></span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a>atlases <span class="ot">&lt;-</span> <span class="fu">show_all</span>(atlases)</span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a><span class="co">#atlases$region</span></span>
<span id="cb146-7"><a href="#cb146-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-8"><a href="#cb146-8" aria-hidden="true" tabindex="-1"></a><span class="co">#galah_config(atlas = "ALA")  #just for Australia</span></span>
<span id="cb146-9"><a href="#cb146-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-10"><a href="#cb146-10" aria-hidden="true" tabindex="-1"></a><span class="do">## print out cutrrent record counts for each Living Atlas, by region</span></span>
<span id="cb146-11"><a href="#cb146-11" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(atlases<span class="sc">$</span>region)) {     </span>
<span id="cb146-12"><a href="#cb146-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">galah_config</span>(<span class="at">atlas =</span> atlases<span class="sc">$</span>region[i])</span>
<span id="cb146-13"><a href="#cb146-13" aria-hidden="true" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">paste0</span>(atlases<span class="sc">$</span>region[i], <span class="st">" = "</span>, <span class="fu">atlas_counts</span>())  <span class="co"># Some output</span></span>
<span id="cb146-14"><a href="#cb146-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(out)                                                <span class="co"># Using print function</span></span>
<span id="cb146-15"><a href="#cb146-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb146-16"><a href="#cb146-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-17"><a href="#cb146-17" aria-hidden="true" tabindex="-1"></a><span class="co">#citation("galah")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-2---item-types-counts-across-publication-years." class="level4">
<h4 class="anchored" data-anchor-id="figure-2---item-types-counts-across-publication-years.">Figure 2 - item types counts across publication years.</h4>
<p>Plot year and literature type.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load processed version list of "impact" literature related to ALA - use the .csv file.   </span></span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a>dat_all <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"dat_all_processed.csv"</span>))</span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_all) #3764   96</span></span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-5"><a href="#cb147-5" aria-hidden="true" tabindex="-1"></a>dat_all <span class="sc">%&gt;%</span> </span>
<span id="cb147-6"><a href="#cb147-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Publication.Year, Item.Type) <span class="ot">-&gt;</span> count_year</span>
<span id="cb147-7"><a href="#cb147-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-8"><a href="#cb147-8" aria-hidden="true" tabindex="-1"></a>nb.cols <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(dat_all<span class="sc">$</span>Item.Type))</span>
<span id="cb147-9"><a href="#cb147-9" aria-hidden="true" tabindex="-1"></a>mycolors <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">8</span>, <span class="st">"Set2"</span>))(nb.cols)</span>
<span id="cb147-10"><a href="#cb147-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-11"><a href="#cb147-11" aria-hidden="true" tabindex="-1"></a>figure2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_year, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.character</span>(Publication.Year), <span class="at">y =</span> n, <span class="at">fill =</span> <span class="fu">reorder</span>(Item.Type, n))) <span class="sc">+</span> </span>
<span id="cb147-12"><a href="#cb147-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span> <span class="co">#, position = "dodge"</span></span>
<span id="cb147-13"><a href="#cb147-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb147-14"><a href="#cb147-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">rev</span>(mycolors), <span class="at">name =</span> <span class="st">'ALA-linked output type:'</span>) <span class="sc">+</span></span>
<span id="cb147-15"><a href="#cb147-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"item count"</span>)  <span class="sc">+</span> </span>
<span id="cb147-16"><a href="#cb147-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span>
<span id="cb147-17"><a href="#cb147-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-18"><a href="#cb147-18" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure to pdf</span></span>
<span id="cb147-19"><a href="#cb147-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> figure2, <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.type_years_counts.pdf"</span>), <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>,  <span class="at">dpi =</span> <span class="st">"retina"</span>, <span class="at">scale =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-3---comparing-scopes-of-articles-vs-grey-publications." class="level4">
<h4 class="anchored" data-anchor-id="figure-3---comparing-scopes-of-articles-vs-grey-publications.">Figure 3 - comparing scopes of articles vs grey publications.</h4>
<p>Panels with article scope categories using manual tags.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>dat_all <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"dat_all_processed.csv"</span>))</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_all) <span class="co">#3764   96</span></span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Subset data into peer-reviewed vs. grey literature.</span></span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a>dat_all_articles <span class="ot">&lt;-</span> dat_all <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Item.Type <span class="sc">==</span> <span class="st">"journalArticle"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Key<span class="sc">:</span>Manual.Tags)</span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_all_articles) <span class="co">#2406</span></span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a>dat_all_grey <span class="ot">&lt;-</span> dat_all <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Item.Type <span class="sc">!=</span> <span class="st">"journalArticle"</span>) <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Key<span class="sc">:</span>Manual.Tags)</span>
<span id="cb148-7"><a href="#cb148-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_all_grey) <span class="co">#1358</span></span>
<span id="cb148-8"><a href="#cb148-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-9"><a href="#cb148-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot as counts with Custom Color Pallettes (color-blind friendly) and labels</span></span>
<span id="cb148-10"><a href="#cb148-10" aria-hidden="true" tabindex="-1"></a>Colors_TF <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">"TRUE"</span> <span class="ot">=</span> <span class="st">"#E69F00"</span>, <span class="st">"FALSE"</span> <span class="ot">=</span> <span class="st">"#56B4E9"</span>)</span>
<span id="cb148-11"><a href="#cb148-11" aria-hidden="true" tabindex="-1"></a>Colors_7light <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'#99DDFF'</span>, <span class="st">'#AAAA00'</span>, <span class="st">'#EE8866'</span>, <span class="st">'#EEDD88'</span>, <span class="st">'#FFAABB'</span>, <span class="st">'#44BB99'</span>, <span class="st">'#77AADD'</span>)</span>
<span id="cb148-12"><a href="#cb148-12" aria-hidden="true" tabindex="-1"></a>Colors_6bright <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">'#EE6677'</span>, <span class="st">'#228833'</span>, <span class="st">'#4477AA'</span>, <span class="st">'#CCBB44'</span>, <span class="st">'#66CCEE'</span>, <span class="st">'#AA3377'</span>)</span>
<span id="cb148-13"><a href="#cb148-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-14"><a href="#cb148-14" aria-hidden="true" tabindex="-1"></a><span class="do">##  Main use of ALA - based on manual tags - by publication type</span></span>
<span id="cb148-15"><a href="#cb148-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-16"><a href="#cb148-16" aria-hidden="true" tabindex="-1"></a><span class="co">#for articles</span></span>
<span id="cb148-17"><a href="#cb148-17" aria-hidden="true" tabindex="-1"></a>mtags_articles <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat_all_articles<span class="sc">$</span>Manual.Tags, <span class="st">"; "</span>) </span>
<span id="cb148-18"><a href="#cb148-18" aria-hidden="true" tabindex="-1"></a>mtags_articles_num <span class="ot">&lt;-</span> <span class="fu">unlist</span>(mtags_articles)[<span class="fu">str_detect</span>(<span class="fu">unlist</span>(mtags_articles), <span class="st">"^[0-9]"</span>)] <span class="co">#keep only the 6 tags starting with numerics (main use)</span></span>
<span id="cb148-19"><a href="#cb148-19" aria-hidden="true" tabindex="-1"></a>count_mtags_articles_num <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(mtags_articles_num)) <span class="co">#739 used</span></span>
<span id="cb148-20"><a href="#cb148-20" aria-hidden="true" tabindex="-1"></a>count_mtags_articles_num <span class="ot">&lt;-</span> <span class="fu">rename</span>(count_mtags_articles_num, <span class="at">Tag =</span> mtags_articles_num)</span>
<span id="cb148-21"><a href="#cb148-21" aria-hidden="true" tabindex="-1"></a>count_mtags_articles_num<span class="sc">$</span>Subset <span class="ot">&lt;-</span> <span class="st">"articles"</span></span>
<span id="cb148-22"><a href="#cb148-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-23"><a href="#cb148-23" aria-hidden="true" tabindex="-1"></a><span class="do">##for grey</span></span>
<span id="cb148-24"><a href="#cb148-24" aria-hidden="true" tabindex="-1"></a>mtags_grey <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat_all_grey<span class="sc">$</span>Manual.Tags, <span class="st">"; "</span>) </span>
<span id="cb148-25"><a href="#cb148-25" aria-hidden="true" tabindex="-1"></a>mtags_grey_num <span class="ot">&lt;-</span> <span class="fu">unlist</span>(mtags_grey)[<span class="fu">str_detect</span>(<span class="fu">unlist</span>(mtags_grey), <span class="st">"^[0-9]"</span>)] <span class="co">#keep only the 6 tags starting with numerics (main use)</span></span>
<span id="cb148-26"><a href="#cb148-26" aria-hidden="true" tabindex="-1"></a>count_mtags_grey_num <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(mtags_grey_num)) <span class="co">#only 181 used ALA resources</span></span>
<span id="cb148-27"><a href="#cb148-27" aria-hidden="true" tabindex="-1"></a>count_mtags_grey_num <span class="ot">&lt;-</span> <span class="fu">rename</span>(count_mtags_grey_num, <span class="at">Tag =</span> mtags_grey_num)</span>
<span id="cb148-28"><a href="#cb148-28" aria-hidden="true" tabindex="-1"></a>count_mtags_grey_num<span class="sc">$</span>Subset <span class="ot">&lt;-</span> <span class="st">"grey"</span></span>
<span id="cb148-29"><a href="#cb148-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-30"><a href="#cb148-30" aria-hidden="true" tabindex="-1"></a><span class="co">#merge into a single dataframe</span></span>
<span id="cb148-31"><a href="#cb148-31" aria-hidden="true" tabindex="-1"></a>count_mtags_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(count_mtags_grey_num, count_mtags_articles_num)</span>
<span id="cb148-32"><a href="#cb148-32" aria-hidden="true" tabindex="-1"></a>count_mtags_df<span class="sc">$</span>Subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(count_mtags_df<span class="sc">$</span>Subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"grey"</span>,<span class="st">"articles"</span>)) <span class="co">#reorder levels for plotting</span></span>
<span id="cb148-33"><a href="#cb148-33" aria-hidden="true" tabindex="-1"></a>count_mtags_df<span class="sc">$</span>Tag <span class="ot">&lt;-</span> <span class="fu">factor</span>(count_mtags_df<span class="sc">$</span>Tag, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"6 - ALA published"</span>, <span class="st">"4 - ALA acknowledged"</span>, <span class="st">"3 - ALA discussed"</span>, <span class="st">"5 - ALA mentioned"</span>, <span class="st">"2 - ALA cited"</span>, <span class="st">"1 - ALA used"</span>))</span>
<span id="cb148-34"><a href="#cb148-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-35"><a href="#cb148-35" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb148-36"><a href="#cb148-36" aria-hidden="true" tabindex="-1"></a>figure3A <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_mtags_df, <span class="fu">aes</span>(<span class="at">x =</span> Subset, <span class="at">y =</span> Freq, <span class="at">fill =</span> Tag)) <span class="sc">+</span> </span>
<span id="cb148-37"><a href="#cb148-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb148-38"><a href="#cb148-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb148-39"><a href="#cb148-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb148-40"><a href="#cb148-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"proportion"</span>) <span class="sc">+</span></span>
<span id="cb148-41"><a href="#cb148-41" aria-hidden="true" tabindex="-1"></a>  <span class="co">#scale_x_discrete(labels = c("articles", "grey")) +</span></span>
<span id="cb148-42"><a href="#cb148-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"ALA use type"</span>, <span class="at">values =</span> Colors_6bright, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">"1 - ALA used"</span>, <span class="st">"2 - ALA cited"</span>, <span class="st">"3 - ALA discussed"</span>, <span class="st">"4 - ALA acknowledged"</span>, <span class="st">"5 - ALA mentioned"</span>, <span class="st">"6 - ALA published"</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"data/service"</span>, <span class="st">"cited"</span>, <span class="st">"discussed"</span>, <span class="st">"acknowledged"</span>, <span class="st">"mentioned"</span>, <span class="st">"published by"</span>))</span>
<span id="cb148-43"><a href="#cb148-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-44"><a href="#cb148-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-45"><a href="#cb148-45" aria-hidden="true" tabindex="-1"></a><span class="do">##  Main ALA data type used  - based on manual tags - by publication type</span></span>
<span id="cb148-46"><a href="#cb148-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-47"><a href="#cb148-47" aria-hidden="true" tabindex="-1"></a><span class="co">#Note when tags used: "Species occurrence records" or "Map" (in parallel to numerical tags - details of use)</span></span>
<span id="cb148-48"><a href="#cb148-48" aria-hidden="true" tabindex="-1"></a>utags_articles <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat_all_articles<span class="sc">$</span>Manual.Tags, <span class="st">"; "</span>) </span>
<span id="cb148-49"><a href="#cb148-49" aria-hidden="true" tabindex="-1"></a><span class="do">##for articles</span></span>
<span id="cb148-50"><a href="#cb148-50" aria-hidden="true" tabindex="-1"></a>utags_articles_det <span class="ot">&lt;-</span> <span class="fu">unlist</span>(utags_articles)[<span class="fu">unlist</span>(utags_articles) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Species occurrence records"</span>, <span class="st">"Map"</span> , <span class="st">"Modelling"</span>, <span class="st">"Climate Data"</span>, <span class="st">"Spatial Portal"</span>, <span class="st">"Species lists"</span>, <span class="st">"Profiles"</span>)]</span>
<span id="cb148-51"><a href="#cb148-51" aria-hidden="true" tabindex="-1"></a>count_utags_articles_det <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(utags_articles_det))</span>
<span id="cb148-52"><a href="#cb148-52" aria-hidden="true" tabindex="-1"></a>count_utags_articles_det <span class="ot">&lt;-</span> <span class="fu">rename</span>(count_utags_articles_det, <span class="at">Tag =</span> utags_articles_det)</span>
<span id="cb148-53"><a href="#cb148-53" aria-hidden="true" tabindex="-1"></a>count_utags_articles_det<span class="sc">$</span>Subset <span class="ot">&lt;-</span> <span class="st">"articles"</span></span>
<span id="cb148-54"><a href="#cb148-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-55"><a href="#cb148-55" aria-hidden="true" tabindex="-1"></a><span class="do">##for grey</span></span>
<span id="cb148-56"><a href="#cb148-56" aria-hidden="true" tabindex="-1"></a>utags_grey <span class="ot">&lt;-</span> <span class="fu">str_split</span>(dat_all_grey<span class="sc">$</span>Manual.Tags, <span class="st">"; "</span>) </span>
<span id="cb148-57"><a href="#cb148-57" aria-hidden="true" tabindex="-1"></a>utags_grey_det <span class="ot">&lt;-</span> <span class="fu">unlist</span>(utags_grey)[<span class="fu">unlist</span>(utags_grey) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Species occurrence records"</span>, <span class="st">"Map"</span> , <span class="st">"Modelling"</span>, <span class="st">"Climate Data"</span>, <span class="st">"Spatial Portal"</span>, <span class="st">"Species lists"</span>, <span class="st">"Profiles"</span>)]</span>
<span id="cb148-58"><a href="#cb148-58" aria-hidden="true" tabindex="-1"></a>count_utags_grey_det <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(utags_grey_det))</span>
<span id="cb148-59"><a href="#cb148-59" aria-hidden="true" tabindex="-1"></a>count_utags_grey_det <span class="ot">&lt;-</span> <span class="fu">rename</span>(count_utags_grey_det, <span class="at">Tag =</span> utags_grey_det)</span>
<span id="cb148-60"><a href="#cb148-60" aria-hidden="true" tabindex="-1"></a>count_utags_grey_det<span class="sc">$</span>Subset <span class="ot">&lt;-</span> <span class="st">"grey"</span></span>
<span id="cb148-61"><a href="#cb148-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-62"><a href="#cb148-62" aria-hidden="true" tabindex="-1"></a><span class="co">#merge into a single dataframe</span></span>
<span id="cb148-63"><a href="#cb148-63" aria-hidden="true" tabindex="-1"></a>count_utags_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(count_utags_grey_det, count_utags_articles_det)</span>
<span id="cb148-64"><a href="#cb148-64" aria-hidden="true" tabindex="-1"></a>count_utags_df<span class="sc">$</span>Subset <span class="ot">&lt;-</span> <span class="fu">factor</span>(count_utags_df<span class="sc">$</span>Subset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"grey"</span>,<span class="st">"articles"</span>)) <span class="co">#reorder levels for plotting</span></span>
<span id="cb148-65"><a href="#cb148-65" aria-hidden="true" tabindex="-1"></a>count_utags_df<span class="sc">$</span>Tag <span class="ot">&lt;-</span> <span class="fu">factor</span>(count_utags_df<span class="sc">$</span>Tag, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Species lists"</span>, <span class="st">"Profiles"</span>, <span class="st">"Spatial Portal"</span>, <span class="st">"Modelling"</span>, <span class="st">"Climate Data"</span>, <span class="st">"Map"</span>, <span class="st">"Species occurrence records"</span>))</span>
<span id="cb148-66"><a href="#cb148-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-67"><a href="#cb148-67" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb148-68"><a href="#cb148-68" aria-hidden="true" tabindex="-1"></a>figure3B <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_utags_df, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Subset, Freq), <span class="at">y =</span> Freq, <span class="at">fill =</span> Tag)) <span class="sc">+</span> </span>
<span id="cb148-69"><a href="#cb148-69" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb148-70"><a href="#cb148-70" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb148-71"><a href="#cb148-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb148-72"><a href="#cb148-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"proportion"</span>) <span class="sc">+</span></span>
<span id="cb148-73"><a href="#cb148-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"used ALA data/service"</span>, <span class="at">values =</span> Colors_7light, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">"Species lists"</span>, <span class="st">"Profiles"</span>, <span class="st">"Spatial Portal"</span>, <span class="st">"Modelling"</span>, <span class="st">"Climate Data"</span>, <span class="st">"Map"</span>, <span class="st">"Species occurrence records"</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Species lists"</span>, <span class="st">"Profiles"</span>, <span class="st">"Spatial Portal"</span>, <span class="st">"Modelling"</span>, <span class="st">"Climate Data"</span>, <span class="st">"Map"</span>, <span class="st">"Species occurrence records"</span>))</span>
<span id="cb148-74"><a href="#cb148-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-75"><a href="#cb148-75" aria-hidden="true" tabindex="-1"></a><span class="do">## Australan focus overall - by publication type </span></span>
<span id="cb148-76"><a href="#cb148-76" aria-hidden="true" tabindex="-1"></a><span class="co">#as.data.frame(table(dat$article_grey, dat$article_grey))</span></span>
<span id="cb148-77"><a href="#cb148-77" aria-hidden="true" tabindex="-1"></a>count_AU_article_grey <span class="ot">&lt;-</span> <span class="fu">count</span>(dat_all, article_grey, AU_title_abs, <span class="at">name =</span> <span class="st">"Count"</span>)</span>
<span id="cb148-78"><a href="#cb148-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-79"><a href="#cb148-79" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb148-80"><a href="#cb148-80" aria-hidden="true" tabindex="-1"></a>figure3C <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_AU_article_grey, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(article_grey, Count), <span class="at">y =</span> Count, <span class="at">fill =</span> AU_title_abs)) <span class="sc">+</span> </span>
<span id="cb148-81"><a href="#cb148-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb148-82"><a href="#cb148-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb148-83"><a href="#cb148-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb148-84"><a href="#cb148-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"proportion"</span>) <span class="sc">+</span> </span>
<span id="cb148-85"><a href="#cb148-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-86"><a href="#cb148-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"Australia mentioned"</span>, <span class="at">values =</span> Colors_TF)</span>
<span id="cb148-87"><a href="#cb148-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-88"><a href="#cb148-88" aria-hidden="true" tabindex="-1"></a><span class="do">## Community/Citizen Science focus/mentioned - by publication type</span></span>
<span id="cb148-89"><a href="#cb148-89" aria-hidden="true" tabindex="-1"></a><span class="co">#as.data.frame(table(dat$article_grey, dat$article_grey))</span></span>
<span id="cb148-90"><a href="#cb148-90" aria-hidden="true" tabindex="-1"></a>count_CS_article_grey <span class="ot">&lt;-</span> <span class="fu">count</span>(dat_all, article_grey, CS_title_abs, <span class="at">name =</span> <span class="st">"Count"</span>)</span>
<span id="cb148-91"><a href="#cb148-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-92"><a href="#cb148-92" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb148-93"><a href="#cb148-93" aria-hidden="true" tabindex="-1"></a>figure3D <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_CS_article_grey, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(article_grey, Count), <span class="at">y =</span> Count, <span class="at">fill =</span> CS_title_abs)) <span class="sc">+</span> </span>
<span id="cb148-94"><a href="#cb148-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb148-95"><a href="#cb148-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb148-96"><a href="#cb148-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb148-97"><a href="#cb148-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"proportion"</span>) <span class="sc">+</span> </span>
<span id="cb148-98"><a href="#cb148-98" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="st">"articles"</span>)) <span class="sc">+</span></span>
<span id="cb148-99"><a href="#cb148-99" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">name =</span> <span class="st">"community mentioned"</span>, <span class="at">values =</span> Colors_TF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Assemble into a multipanel figure and save as pdf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assemble the panels</span></span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a>figure3 <span class="ot">&lt;-</span> figure3A <span class="sc">/</span> figure3B <span class="sc">/</span> figure3C <span class="sc">/</span> figure3D <span class="sc">+</span></span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb149-4"><a href="#cb149-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb149-5"><a href="#cb149-5" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure </span></span>
<span id="cb149-6"><a href="#cb149-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> figure3, <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.scopes_articles_grey.pdf"</span>), <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">scale =</span> <span class="fl">1.6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-4---comparing-authorship-of-articles-vs-grey-publications." class="level4">
<h4 class="anchored" data-anchor-id="figure-4---comparing-authorship-of-articles-vs-grey-publications.">Figure 4 - comparing authorship of articles vs grey publications.</h4>
<p>Panel with author counts articles vs grey.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb150"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dat &lt;- read.csv(here("data", "dat_all_processed.csv")) #loading all processed data</span></span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Plots of  author numbers per paper across years</span></span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a><span class="co">#create a dataframe of author counts and literature type</span></span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a>dat <span class="sc">%&gt;%</span> </span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(article_grey) <span class="sc">%&gt;%</span></span>
<span id="cb150-7"><a href="#cb150-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">reframe</span>(Publication.Year, Author_counts) <span class="sc">%&gt;%</span> </span>
<span id="cb150-8"><a href="#cb150-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="ot">-&gt;</span> count_year_authors</span>
<span id="cb150-9"><a href="#cb150-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-10"><a href="#cb150-10" aria-hidden="true" tabindex="-1"></a><span class="co"># plot of author counts</span></span>
<span id="cb150-11"><a href="#cb150-11" aria-hidden="true" tabindex="-1"></a>figure4A <span class="ot">&lt;-</span> count_year_authors <span class="sc">%&gt;%</span> </span>
<span id="cb150-12"><a href="#cb150-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Publication.Year, <span class="at">y =</span> Author_counts, <span class="at">color =</span> article_grey)) <span class="sc">+</span></span>
<span id="cb150-13"><a href="#cb150-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb150-14"><a href="#cb150-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">220</span>) <span class="sc">+</span></span>
<span id="cb150-15"><a href="#cb150-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> article_grey) <span class="sc">+</span></span>
<span id="cb150-16"><a href="#cb150-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#66c2a5"</span>, <span class="st">"#808080"</span>)) <span class="sc">+</span></span>
<span id="cb150-17"><a href="#cb150-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of authors per output"</span>) <span class="sc">+</span></span>
<span id="cb150-18"><a href="#cb150-18" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_smooth(method = lm, formula = y ~ x) +</span></span>
<span id="cb150-19"><a href="#cb150-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span> </span>
<span id="cb150-20"><a href="#cb150-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb150-21"><a href="#cb150-21" aria-hidden="true" tabindex="-1"></a>  <span class="co">#theme(strip.text.x = element_blank()) + #Remove labels from facet plot</span></span>
<span id="cb150-22"><a href="#cb150-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">""</span>, <span class="at">x =</span> <span class="st">"Publication years"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Panel with authorship network of articles and grey.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dat &lt;- read.csv(here("data", "dat_all_processed.csv")) #loading all processed data</span></span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Plots of overall outcome authorship network graph</span></span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Separate author names for overall summary per paper (list of vectors format) for article</span></span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a>Author_split <span class="ot">&lt;-</span> <span class="fu">str_split</span>(<span class="at">string =</span> dat[ , <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>) <span class="co">#list of vectors of authors</span></span>
<span id="cb151-7"><a href="#cb151-7" aria-hidden="true" tabindex="-1"></a>df_auth_collab <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ref.id =</span> <span class="fu">rep</span>(dat[, <span class="st">"Key"</span>], <span class="fu">lengths</span>(Author_split)), <span class="at">auth_names =</span> <span class="fu">unlist</span>(Author_split))</span>
<span id="cb151-8"><a href="#cb151-8" aria-hidden="true" tabindex="-1"></a>df_auth_collab <span class="ot">&lt;-</span> df_auth_collab[<span class="sc">!</span><span class="fu">is.na</span>(df_auth_collab<span class="sc">$</span>auth_names), ] <span class="co">#remove empty nodes </span></span>
<span id="cb151-9"><a href="#cb151-9" aria-hidden="true" tabindex="-1"></a>df_auth_collab <span class="ot">&lt;-</span> df_auth_collab[df_auth_collab<span class="sc">$</span>auth_names <span class="sc">!=</span> <span class="st">""</span>, ] <span class="co">#remove empty nodes </span></span>
<span id="cb151-10"><a href="#cb151-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-11"><a href="#cb151-11" aria-hidden="true" tabindex="-1"></a>df_auth_collab <span class="sc">%&gt;%</span> <span class="co">#data frame of connections between outputs by shared authors</span></span>
<span id="cb151-12"><a href="#cb151-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_auth_collab, <span class="at">by =</span> <span class="st">"auth_names"</span>, <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb151-13"><a href="#cb151-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="co">#remove many-to-many duplicates)</span></span>
<span id="cb151-14"><a href="#cb151-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ref.id.x <span class="sc">!=</span> ref.id.y) <span class="sc">%&gt;%</span> <span class="co">#use &lt;= to include unconnected outcomes (self-connections counted as 1)</span></span>
<span id="cb151-15"><a href="#cb151-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(ref.id.x, ref.id.y) <span class="sc">%&gt;%</span> </span>
<span id="cb151-16"><a href="#cb151-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">graph_from_data_frame</span>(<span class="at">directed =</span> <span class="cn">FALSE</span>) <span class="ot">-&gt;</span> all_outputs_graph <span class="co">#with output ID (Key) names as vertices</span></span>
<span id="cb151-17"><a href="#cb151-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-18"><a href="#cb151-18" aria-hidden="true" tabindex="-1"></a><span class="co">#add color attribute showing if  article or grey</span></span>
<span id="cb151-19"><a href="#cb151-19" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>name <span class="sc">%in%</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Key"</span>], <span class="st">"article"</span>, <span class="st">"grey"</span>)</span>
<span id="cb151-20"><a href="#cb151-20" aria-hidden="true" tabindex="-1"></a><span class="co">#table(V(all_outputs_graph)$color)</span></span>
<span id="cb151-21"><a href="#cb151-21" aria-hidden="true" tabindex="-1"></a><span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">V</span>(all_outputs_graph)<span class="sc">$</span>name <span class="sc">%in%</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Key"</span>], <span class="st">"#0FD19D"</span>, <span class="st">"#808080"</span>)</span>
<span id="cb151-22"><a href="#cb151-22" aria-hidden="true" tabindex="-1"></a><span class="co">#table(V(all_outputs_graph)$color)</span></span>
<span id="cb151-23"><a href="#cb151-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-24"><a href="#cb151-24" aria-hidden="true" tabindex="-1"></a><span class="co">#plot graph with outcomes as nodes (single-author articles shown)</span></span>
<span id="cb151-25"><a href="#cb151-25" aria-hidden="true" tabindex="-1"></a>figure4B <span class="ot">&lt;-</span> <span class="fu">ggraph</span>(all_outputs_graph, <span class="st">'igraph'</span>, <span class="at">algorithm =</span> <span class="st">'kk'</span>) <span class="sc">+</span>   <span class="co">#, colour = 'forestgreen'</span></span>
<span id="cb151-26"><a href="#cb151-26" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_edge_link0(aes(width = 0.01), edge_alpha = 0.1) + </span></span>
<span id="cb151-27"><a href="#cb151-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_edge_fan</span>(<span class="at">color =</span> <span class="st">"gray50"</span>, <span class="at">width =</span> <span class="fl">0.5</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb151-28"><a href="#cb151-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_node_point</span>(<span class="fu">aes</span>(<span class="at">size =</span> <span class="fl">0.001</span>, <span class="at">fill =</span> type, <span class="at">color =</span> type, <span class="at">alpha =</span> <span class="fl">0.01</span>)) <span class="sc">+</span></span>
<span id="cb151-29"><a href="#cb151-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">"article"</span>, <span class="st">"grey"</span>), <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#66c2a5"</span>, <span class="st">"#808080"</span>)) <span class="sc">+</span> <span class="co"># using directly $color for geom_node_point does not work</span></span>
<span id="cb151-30"><a href="#cb151-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">alpha =</span> <span class="st">"none"</span>, <span class="at">size =</span> <span class="st">"none"</span>, <span class="at">type =</span> <span class="st">"none"</span>) <span class="sc">+</span> <span class="co">#remove unnecessary legend elements</span></span>
<span id="cb151-31"><a href="#cb151-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">colour =</span> <span class="st">"type"</span>) <span class="sc">+</span> <span class="co">#change legend title</span></span>
<span id="cb151-32"><a href="#cb151-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb151-33"><a href="#cb151-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb151-34"><a href="#cb151-34" aria-hidden="true" tabindex="-1"></a>  <span class="co">#theme(strip.text.x = element_blank()) + #Remove labels from facet plot</span></span>
<span id="cb151-35"><a href="#cb151-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Outcome connectedness via shared authors "</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Panels of network degree centrality articles vs grey.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dat &lt;- read.csv(here("data", "dat_all_processed.csv")) #loading all processed data</span></span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a><span class="co">#create separate author networks based on co-authorship for article and grey</span></span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a><span class="do">### article literature subset</span></span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-7"><a href="#cb152-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Separate author names for overall summary per paper (list of vectors format) for article</span></span>
<span id="cb152-8"><a href="#cb152-8" aria-hidden="true" tabindex="-1"></a>Author_split_article <span class="ot">&lt;-</span> <span class="fu">str_split</span>(<span class="at">string =</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>) <span class="co">#list of vectors of authors</span></span>
<span id="cb152-9"><a href="#cb152-9" aria-hidden="true" tabindex="-1"></a><span class="co">#length(Author_split_article) #3093 elements (outputs) in the list</span></span>
<span id="cb152-10"><a href="#cb152-10" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ref.id =</span> <span class="fu">rep</span>(dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"article"</span>, <span class="st">"Key"</span>], <span class="fu">lengths</span>(Author_split_article)),</span>
<span id="cb152-11"><a href="#cb152-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">auth_names =</span> <span class="fu">unlist</span>(Author_split_article))</span>
<span id="cb152-12"><a href="#cb152-12" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="ot">&lt;-</span> df_auth_collab_article[<span class="sc">!</span><span class="fu">is.na</span>(df_auth_collab_article<span class="sc">$</span>auth_names), ] <span class="co">#remove empty</span></span>
<span id="cb152-13"><a href="#cb152-13" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="ot">&lt;-</span> df_auth_collab_article[df_auth_collab_article<span class="sc">$</span>auth_names <span class="sc">!=</span> <span class="st">""</span>, ] <span class="co">#remove empty</span></span>
<span id="cb152-14"><a href="#cb152-14" aria-hidden="true" tabindex="-1"></a><span class="co">#compute graph of coauthorships for all authors</span></span>
<span id="cb152-15"><a href="#cb152-15" aria-hidden="true" tabindex="-1"></a>df_auth_collab_article <span class="sc">%&gt;%</span></span>
<span id="cb152-16"><a href="#cb152-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_auth_collab_article, <span class="at">by =</span> <span class="st">"ref.id"</span>, <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-17"><a href="#cb152-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="co">#remove many-to-many duplicates)</span></span>
<span id="cb152-18"><a href="#cb152-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(auth_names.x <span class="sc">!=</span> auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb152-19"><a href="#cb152-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(auth_names.x, auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb152-20"><a href="#cb152-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">graph_from_data_frame</span>(<span class="at">directed =</span> <span class="cn">FALSE</span>) <span class="ot">-&gt;</span> a1 <span class="co">#with author names as vertices</span></span>
<span id="cb152-21"><a href="#cb152-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-22"><a href="#cb152-22" aria-hidden="true" tabindex="-1"></a><span class="do">### Grey literature subset</span></span>
<span id="cb152-23"><a href="#cb152-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Separate author names for overall summary per paper (list of vectors format) for grey</span></span>
<span id="cb152-24"><a href="#cb152-24" aria-hidden="true" tabindex="-1"></a>Author_split_grey <span class="ot">&lt;-</span> <span class="fu">str_split</span>(<span class="at">string =</span> dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"grey"</span>, <span class="st">"Author_abbr"</span>], <span class="at">pattern =</span> <span class="st">";"</span>) <span class="co">#list of vectors of authors</span></span>
<span id="cb152-25"><a href="#cb152-25" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ref.id =</span> <span class="fu">rep</span>(dat[dat<span class="sc">$</span>article_grey <span class="sc">==</span> <span class="st">"grey"</span>, <span class="st">"Key"</span>], <span class="fu">lengths</span>(Author_split_grey)),</span>
<span id="cb152-26"><a href="#cb152-26" aria-hidden="true" tabindex="-1"></a>                  <span class="at">auth_names =</span> <span class="fu">unlist</span>(Author_split_grey))</span>
<span id="cb152-27"><a href="#cb152-27" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="ot">&lt;-</span> df_auth_collab_grey[<span class="sc">!</span><span class="fu">is.na</span>(df_auth_collab_grey<span class="sc">$</span>auth_names), ] <span class="co">#remove empty</span></span>
<span id="cb152-28"><a href="#cb152-28" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="ot">&lt;-</span> df_auth_collab_grey[df_auth_collab_grey<span class="sc">$</span>auth_names <span class="sc">!=</span> <span class="st">""</span>, ] <span class="co">#remove empty</span></span>
<span id="cb152-29"><a href="#cb152-29" aria-hidden="true" tabindex="-1"></a><span class="co">#compute graph of coauthorships for all authors</span></span>
<span id="cb152-30"><a href="#cb152-30" aria-hidden="true" tabindex="-1"></a>df_auth_collab_grey <span class="sc">%&gt;%</span></span>
<span id="cb152-31"><a href="#cb152-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(df_auth_collab_grey, <span class="at">by =</span> <span class="st">"ref.id"</span>, <span class="at">relationship =</span> <span class="st">"many-to-many"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb152-32"><a href="#cb152-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="co">#remove many-to-many duplicates)</span></span>
<span id="cb152-33"><a href="#cb152-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(auth_names.x <span class="sc">&lt;=</span> auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb152-34"><a href="#cb152-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(auth_names.x, auth_names.y) <span class="sc">%&gt;%</span></span>
<span id="cb152-35"><a href="#cb152-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">graph_from_data_frame</span>(<span class="at">directed =</span> <span class="cn">FALSE</span>) <span class="ot">-&gt;</span> g1 <span class="co">#with author names as vertices</span></span>
<span id="cb152-36"><a href="#cb152-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-37"><a href="#cb152-37" aria-hidden="true" tabindex="-1"></a><span class="co">#collect node centrality characteristics as a data frame (each node is an author)</span></span>
<span id="cb152-38"><a href="#cb152-38" aria-hidden="true" tabindex="-1"></a>metrics_nodes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">data.frame</span>(<span class="at">type =</span> <span class="fu">rep</span>(<span class="st">"article"</span>, <span class="fu">length</span>(igraph<span class="sc">::</span><span class="fu">degree</span>(a1))), <span class="at">deg =</span> igraph<span class="sc">::</span><span class="fu">degree</span>(a1)),</span>
<span id="cb152-39"><a href="#cb152-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">type =</span> <span class="fu">rep</span>(<span class="st">"grey"</span>, <span class="fu">length</span>(igraph<span class="sc">::</span><span class="fu">degree</span>(g1))), <span class="at">deg =</span> igraph<span class="sc">::</span><span class="fu">degree</span>(g1)))</span>
<span id="cb152-40"><a href="#cb152-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-41"><a href="#cb152-41" aria-hidden="true" tabindex="-1"></a><span class="co">#str(metrics_nodes)</span></span>
<span id="cb152-42"><a href="#cb152-42" aria-hidden="true" tabindex="-1"></a>metrics_nodes<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(metrics_nodes<span class="sc">$</span>type) <span class="co">#Convert to a factor</span></span>
<span id="cb152-43"><a href="#cb152-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-44"><a href="#cb152-44" aria-hidden="true" tabindex="-1"></a>figure4C <span class="ot">&lt;-</span> metrics_nodes <span class="sc">%&gt;%</span>  <span class="co">#node degree centrality distribution (measure of immediate connection in a network = authors collaborated directly with a larger number of other academics)</span></span>
<span id="cb152-45"><a href="#cb152-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> deg, <span class="at">fill =</span> type,  <span class="at">color =</span> type)) <span class="sc">+</span></span>
<span id="cb152-46"><a href="#cb152-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> .<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb152-47"><a href="#cb152-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#66c2a5"</span>, <span class="st">"#808080"</span>)) <span class="sc">+</span></span>
<span id="cb152-48"><a href="#cb152-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#66c2a5"</span>, <span class="st">"#808080"</span>)) <span class="sc">+</span></span>
<span id="cb152-49"><a href="#cb152-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> type) <span class="sc">+</span></span>
<span id="cb152-50"><a href="#cb152-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb152-51"><a href="#cb152-51" aria-hidden="true" tabindex="-1"></a>  <span class="co">#theme(strip.text.x = element_blank()) + #Remove labels from facet plot</span></span>
<span id="cb152-52"><a href="#cb152-52" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">""</span>, <span class="at">x =</span> <span class="st">"Node degree centrality"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Assemble into a multipanel figure and save as pdf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assemble the panels</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>figure4 <span class="ot">=</span> figure4AA <span class="sc">/</span> figure4B <span class="sc">/</span> figure4C  <span class="sc">+</span></span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure </span></span>
<span id="cb153-7"><a href="#cb153-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> figure4, <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.authorship.pdf"</span>), <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">18</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">scale =</span> <span class="fl">1.6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-5---global-authorship." class="level4">
<h4 class="anchored" data-anchor-id="figure-5---global-authorship.">Figure 5 - global authorship.</h4>
<p>Global map of authorship by country affiliation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load OpenAlex query results which contain affiliation country information</span></span>
<span id="cb154-2"><a href="#cb154-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"results_OpenAlex_full.RData"</span>)) <span class="co">#loads block_test data object with full OpenAlex block query output</span></span>
<span id="cb154-3"><a href="#cb154-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(block_test) <span class="co">#2835 (number of articles with doi, but some of those have empty affiliations)</span></span>
<span id="cb154-4"><a href="#cb154-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-5"><a href="#cb154-5" aria-hidden="true" tabindex="-1"></a><span class="co">#extract two variables from the results</span></span>
<span id="cb154-6"><a href="#cb154-6" aria-hidden="true" tabindex="-1"></a>res_OpenA <span class="ot">&lt;-</span> <span class="fu">lapply</span>(block_test, <span class="st">"["</span>, <span class="fu">c</span>(<span class="st">"doi"</span>, <span class="st">"AU_CO"</span>))</span>
<span id="cb154-7"><a href="#cb154-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#data.frame(doi = sapply(block_test, "[", c("doi")), AU_CO = sapply(block_test, "[", c("AU_CO")) )</span></span>
<span id="cb154-8"><a href="#cb154-8" aria-hidden="true" tabindex="-1"></a><span class="co">#str(res_OpenA)</span></span>
<span id="cb154-9"><a href="#cb154-9" aria-hidden="true" tabindex="-1"></a>res_OpenA <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind, res_OpenA))</span>
<span id="cb154-10"><a href="#cb154-10" aria-hidden="true" tabindex="-1"></a><span class="co">#str(res_OpenA) #'data.frame':  2835 obs. of  2 variables: doi and AU_CO (affiliation countries of authors)</span></span>
<span id="cb154-11"><a href="#cb154-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-12"><a href="#cb154-12" aria-hidden="true" tabindex="-1"></a><span class="do">## Processing affiliation country information</span></span>
<span id="cb154-13"><a href="#cb154-13" aria-hidden="true" tabindex="-1"></a>doi_country_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">doi =</span> <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">"https://doi.org/"</span>, <span class="st">""</span>, res_OpenA<span class="sc">$</span>doi)), <span class="at">AU_CO =</span> res_OpenA<span class="sc">$</span>AU_CO)</span>
<span id="cb154-14"><a href="#cb154-14" aria-hidden="true" tabindex="-1"></a><span class="co">#str(doi_country_df) #some only have NA as a value</span></span>
<span id="cb154-15"><a href="#cb154-15" aria-hidden="true" tabindex="-1"></a>doi_country_all <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(doi_country_df<span class="sc">$</span>AU_CO, <span class="st">";"</span>)) <span class="co">#extract country names as a single vector</span></span>
<span id="cb154-16"><a href="#cb154-16" aria-hidden="true" tabindex="-1"></a><span class="co">#length(doi_country_all) #total 15295 country values</span></span>
<span id="cb154-17"><a href="#cb154-17" aria-hidden="true" tabindex="-1"></a><span class="co">#table(doi_country_all) #table of country frequencies (note many NA)</span></span>
<span id="cb154-18"><a href="#cb154-18" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(table(doi_country_all)) #111 different country names (including NA)</span></span>
<span id="cb154-19"><a href="#cb154-19" aria-hidden="true" tabindex="-1"></a>doi_country_all <span class="ot">&lt;-</span> doi_country_all[doi_country_all <span class="sc">!=</span> <span class="st">"NA"</span>] <span class="co">#remove "NA" values</span></span>
<span id="cb154-20"><a href="#cb154-20" aria-hidden="true" tabindex="-1"></a><span class="co">#length(doi_country_all) #total 12209 country values</span></span>
<span id="cb154-21"><a href="#cb154-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-22"><a href="#cb154-22" aria-hidden="true" tabindex="-1"></a><span class="do">##Count number of authors per country</span></span>
<span id="cb154-23"><a href="#cb154-23" aria-hidden="true" tabindex="-1"></a>country_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">table</span>(doi_country_all))</span>
<span id="cb154-24"><a href="#cb154-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-25"><a href="#cb154-25" aria-hidden="true" tabindex="-1"></a><span class="do">##load world map data</span></span>
<span id="cb154-26"><a href="#cb154-26" aria-hidden="true" tabindex="-1"></a>world_map <span class="ot">&lt;-</span> <span class="fu">map_data</span>(<span class="st">"world"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb154-27"><a href="#cb154-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span> long <span class="sc">&gt;</span> <span class="dv">180</span>) <span class="co">#remove countries with longitude &gt;180 to make equal projection-like map without artifacts</span></span>
<span id="cb154-28"><a href="#cb154-28" aria-hidden="true" tabindex="-1"></a><span class="co">#table(world_map$region)</span></span>
<span id="cb154-29"><a href="#cb154-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-30"><a href="#cb154-30" aria-hidden="true" tabindex="-1"></a><span class="do">##fix some country names to match regions on the world map</span></span>
<span id="cb154-31"><a href="#cb154-31" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region <span class="ot">&lt;-</span> <span class="fu">str_to_title</span>(<span class="fu">as.character</span>(country_count<span class="sc">$</span>doi_country_all)) <span class="co">#create simplified names for matching</span></span>
<span id="cb154-32"><a href="#cb154-32" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Usa"</span>] <span class="ot">&lt;-</span> <span class="st">"USA"</span> <span class="co">#fix</span></span>
<span id="cb154-33"><a href="#cb154-33" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"United Kingdom"</span>] <span class="ot">&lt;-</span> <span class="st">"UK"</span> <span class="co">#fix</span></span>
<span id="cb154-34"><a href="#cb154-34" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"United Kindom"</span>] <span class="ot">&lt;-</span> <span class="st">"UK"</span> <span class="co">#fix</span></span>
<span id="cb154-35"><a href="#cb154-35" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Russian Federation"</span>] <span class="ot">&lt;-</span> <span class="st">"Russia"</span></span>
<span id="cb154-36"><a href="#cb154-36" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Czechia"</span>] <span class="ot">&lt;-</span> <span class="st">"Czech Republic"</span></span>
<span id="cb154-37"><a href="#cb154-37" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Bolivia (Plurinational State Of)"</span>] <span class="ot">&lt;-</span> <span class="st">"Bolivia"</span></span>
<span id="cb154-38"><a href="#cb154-38" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Venezuela (Bolivarian Republic Of)"</span>] <span class="ot">&lt;-</span> <span class="st">"Venezuela"</span></span>
<span id="cb154-39"><a href="#cb154-39" aria-hidden="true" tabindex="-1"></a><span class="co">#country_count$region[country_count$region == "Hong Kong"] &lt;- "China" #this messess up the counts?</span></span>
<span id="cb154-40"><a href="#cb154-40" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Viet Nam"</span>] <span class="ot">&lt;-</span> <span class="st">"Vietnam"</span></span>
<span id="cb154-41"><a href="#cb154-41" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Brunei Darussalam"</span>] <span class="ot">&lt;-</span> <span class="st">"Brunei"</span></span>
<span id="cb154-42"><a href="#cb154-42" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Congo"</span>] <span class="ot">&lt;-</span> <span class="st">"Republic of Congo"</span></span>
<span id="cb154-43"><a href="#cb154-43" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Syrian Arab Republic"</span>] <span class="ot">&lt;-</span> <span class="st">"Syria"</span></span>
<span id="cb154-44"><a href="#cb154-44" aria-hidden="true" tabindex="-1"></a>country_count<span class="sc">$</span>region[country_count<span class="sc">$</span>region <span class="sc">==</span> <span class="st">"Tanzania, United Republic Of"</span>] <span class="ot">&lt;-</span> <span class="st">"Tanzania"</span></span>
<span id="cb154-45"><a href="#cb154-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-46"><a href="#cb154-46" aria-hidden="true" tabindex="-1"></a><span class="co">#country_count$region %in% world_map$region #check matching</span></span>
<span id="cb154-47"><a href="#cb154-47" aria-hidden="true" tabindex="-1"></a><span class="co">#(country_count$region)[!(country_count$region %in% world_map$region)] #check matching</span></span>
<span id="cb154-48"><a href="#cb154-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-49"><a href="#cb154-49" aria-hidden="true" tabindex="-1"></a><span class="do">## prepare tabele of counts for all countries (regions) on the world  map:</span></span>
<span id="cb154-50"><a href="#cb154-50" aria-hidden="true" tabindex="-1"></a>emptymap <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">region =</span> <span class="fu">unique</span>(world_map<span class="sc">$</span>region), <span class="at">n =</span> <span class="fu">rep</span>(<span class="dv">0</span>,<span class="fu">length</span>(<span class="fu">unique</span>(world_map<span class="sc">$</span>region)))) <span class="co">#create table with all counts as 0</span></span>
<span id="cb154-51"><a href="#cb154-51" aria-hidden="true" tabindex="-1"></a>fullmap <span class="ot">&lt;-</span> <span class="fu">left_join</span>(emptymap, country_count, <span class="at">by =</span> <span class="st">"region"</span>) <span class="co">#join with actual counts table</span></span>
<span id="cb154-52"><a href="#cb154-52" aria-hidden="true" tabindex="-1"></a>fullmap<span class="sc">$</span>count <span class="ot">&lt;-</span> fullmap<span class="sc">$</span>n <span class="sc">+</span> fullmap<span class="sc">$</span>Freq <span class="co"># make new column for fixed counts</span></span>
<span id="cb154-53"><a href="#cb154-53" aria-hidden="true" tabindex="-1"></a>fullmap<span class="sc">$</span>count[<span class="fu">is.na</span>(fullmap<span class="sc">$</span>count)] <span class="ot">&lt;-</span> <span class="dv">0</span> <span class="co">#change NA to 0 for regions with no counts</span></span>
<span id="cb154-54"><a href="#cb154-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-55"><a href="#cb154-55" aria-hidden="true" tabindex="-1"></a><span class="do">##plot</span></span>
<span id="cb154-56"><a href="#cb154-56" aria-hidden="true" tabindex="-1"></a>figure5 <span class="ot">&lt;-</span> fullmap <span class="sc">%&gt;%</span> </span>
<span id="cb154-57"><a href="#cb154-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> count, <span class="at">map_id =</span> region)) <span class="sc">+</span></span>
<span id="cb154-58"><a href="#cb154-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_map</span>(<span class="at">map =</span> world_map) <span class="sc">+</span></span>
<span id="cb154-59"><a href="#cb154-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">x =</span> world_map<span class="sc">$</span>long, <span class="at">y =</span> world_map<span class="sc">$</span>lat) <span class="sc">+</span></span>
<span id="cb154-60"><a href="#cb154-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_map</span>(<span class="st">"moll"</span>) <span class="sc">+</span></span>
<span id="cb154-61"><a href="#cb154-61" aria-hidden="true" tabindex="-1"></a>  ggthemes<span class="sc">::</span><span class="fu">theme_map</span>() <span class="sc">+</span> </span>
<span id="cb154-62"><a href="#cb154-62" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">"left"</span>) <span class="sc">+</span></span>
<span id="cb154-63"><a href="#cb154-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"lightgrey"</span>, <span class="at">high =</span> <span class="st">"darkgreen"</span>, <span class="at">name =</span> <span class="st">"author count"</span>,</span>
<span id="cb154-64"><a href="#cb154-64" aria-hidden="true" tabindex="-1"></a>      <span class="at">guide =</span> <span class="fu">guide_colorbar</span>(<span class="at">direction =</span> <span class="st">"vertical."</span>)) <span class="sc">+</span></span>
<span id="cb154-65"><a href="#cb154-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_colourbar</span>(<span class="at">barwidth =</span> <span class="fu">unit</span>(<span class="dv">3</span>, <span class="at">units =</span> <span class="st">"mm"</span>), <span class="at">barheight =</span> <span class="fu">unit</span>(<span class="dv">35</span>, <span class="at">units =</span> <span class="st">"mm"</span>)))</span>
<span id="cb154-66"><a href="#cb154-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-67"><a href="#cb154-67" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure to pdf</span></span>
<span id="cb154-68"><a href="#cb154-68" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> figure5, <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.map_author_counts.pdf"</span>), <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">units =</span> <span class="st">"cm"</span>,  <span class="at">dpi =</span> <span class="st">"retina"</span>, <span class="at">scale =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-6---global-country-collaboration." class="level4">
<h4 class="anchored" data-anchor-id="figure-6---global-country-collaboration.">Figure 6 - global country collaboration.</h4>
<p>Chord diagram of top country coauthorships.<br>
Note: this is a chord diagram is an atypical grid (base) graphics and is hard to combine in a single figure with a ggplot graphics.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"results_OpenAlex_full.RData"</span>)) <span class="co">#loads block_test data object with full OpenAlex block query output</span></span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a><span class="co">#length(block_test) #loaded data object length = 2835</span></span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a><span class="co">#extract two variables from the results</span></span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a>res_OpenA <span class="ot">&lt;-</span> <span class="fu">lapply</span>(block_test, <span class="st">"["</span>, <span class="fu">c</span>(<span class="st">"doi"</span>, <span class="st">"AU_CO"</span>))</span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a>res_OpenA <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">do.call</span>(rbind, res_OpenA))</span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Processing affiliation country information</span></span>
<span id="cb155-8"><a href="#cb155-8" aria-hidden="true" tabindex="-1"></a>doi_country_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">doi =</span> <span class="fu">tolower</span>(<span class="fu">gsub</span>(<span class="st">"https://doi.org/"</span>, <span class="st">""</span>, res_OpenA<span class="sc">$</span>doi)), <span class="at">AU_CO =</span> res_OpenA<span class="sc">$</span>AU_CO)</span>
<span id="cb155-9"><a href="#cb155-9" aria-hidden="true" tabindex="-1"></a>doi_country_all <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">strsplit</span>(doi_country_df<span class="sc">$</span>AU_CO, <span class="st">";"</span>)) <span class="co">#extract country names as a single vector</span></span>
<span id="cb155-10"><a href="#cb155-10" aria-hidden="true" tabindex="-1"></a><span class="co">#table(doi_country_all) #table of country frequencies (note many NA)</span></span>
<span id="cb155-11"><a href="#cb155-11" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(table(doi_country_all)) #111 different country names (including NA)</span></span>
<span id="cb155-12"><a href="#cb155-12" aria-hidden="true" tabindex="-1"></a>doi_country_all <span class="ot">&lt;-</span> doi_country_all[doi_country_all <span class="sc">!=</span> <span class="st">"NA"</span>] <span class="co">#remove "NA" affiliations</span></span>
<span id="cb155-13"><a href="#cb155-13" aria-hidden="true" tabindex="-1"></a><span class="co">#length(doi_country_all) #total 12209 country values</span></span>
<span id="cb155-14"><a href="#cb155-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-15"><a href="#cb155-15" aria-hidden="true" tabindex="-1"></a><span class="do">##Count number of authors per country</span></span>
<span id="cb155-16"><a href="#cb155-16" aria-hidden="true" tabindex="-1"></a>country_count <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">table</span>(doi_country_all))</span>
<span id="cb155-17"><a href="#cb155-17" aria-hidden="true" tabindex="-1"></a><span class="co">#str(country_count)</span></span>
<span id="cb155-18"><a href="#cb155-18" aria-hidden="true" tabindex="-1"></a>country_count <span class="ot">&lt;-</span> <span class="fu">arrange</span>(country_count, <span class="sc">-</span>Freq)</span>
<span id="cb155-19"><a href="#cb155-19" aria-hidden="true" tabindex="-1"></a>top10countries <span class="ot">&lt;-</span> country_count[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="st">'doi_country_all'</span>]</span>
<span id="cb155-20"><a href="#cb155-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-21"><a href="#cb155-21" aria-hidden="true" tabindex="-1"></a><span class="co">#create big matrix of coauthorship links among all countries (incl NA for now)</span></span>
<span id="cb155-22"><a href="#cb155-22" aria-hidden="true" tabindex="-1"></a>doi_country_df<span class="sc">$</span>DB <span class="ot">&lt;-</span> <span class="st">"SCOPUS"</span> <span class="co">#add extra column to enable processing with biblioNetwork function below</span></span>
<span id="cb155-23"><a href="#cb155-23" aria-hidden="true" tabindex="-1"></a>NetMatrix <span class="ot">&lt;-</span> <span class="fu">biblioNetwork</span>(doi_country_df, <span class="at">analysis =</span> <span class="st">"collaboration"</span>, <span class="at">network =</span> <span class="st">"countries"</span>, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb155-24"><a href="#cb155-24" aria-hidden="true" tabindex="-1"></a><span class="co">#str(NetMatrix)</span></span>
<span id="cb155-25"><a href="#cb155-25" aria-hidden="true" tabindex="-1"></a>net_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(NetMatrix)</span>
<span id="cb155-26"><a href="#cb155-26" aria-hidden="true" tabindex="-1"></a><span class="co">#country_sum &lt;- data.frame(countries = rownames(net_matrix), sum = rowSums(net_matrix) #sum of links per country</span></span>
<span id="cb155-27"><a href="#cb155-27" aria-hidden="true" tabindex="-1"></a>small_matrix <span class="ot">&lt;-</span> net_matrix[<span class="fu">rownames</span>(net_matrix) <span class="sc">%in%</span> top10countries, <span class="fu">colnames</span>(net_matrix) <span class="sc">%in%</span> top10countries] <span class="co">#reduce matrix to 10 most productive countries (with highest total number of authors)</span></span>
<span id="cb155-28"><a href="#cb155-28" aria-hidden="true" tabindex="-1"></a><span class="fu">diag</span>(small_matrix) <span class="ot">&lt;-</span> <span class="dv">0</span> <span class="co">#get rid of collaboration with same country</span></span>
<span id="cb155-29"><a href="#cb155-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-30"><a href="#cb155-30" aria-hidden="true" tabindex="-1"></a><span class="co"># getting rid of lower triangle (as this is duplication of info)</span></span>
<span id="cb155-31"><a href="#cb155-31" aria-hidden="true" tabindex="-1"></a>small_matrix[<span class="fu">lower.tri</span>(small_matrix)] <span class="ot">&lt;-</span> <span class="dv">0</span> </span>
<span id="cb155-32"><a href="#cb155-32" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(small_matrix) <span class="ot">&lt;-</span> <span class="fu">str_to_title</span>(<span class="fu">colnames</span>(small_matrix)) <span class="co">#change to tile case</span></span>
<span id="cb155-33"><a href="#cb155-33" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(small_matrix) <span class="ot">&lt;-</span> <span class="fu">str_to_title</span>(<span class="fu">rownames</span>(small_matrix)) <span class="co">#change to title case</span></span>
<span id="cb155-34"><a href="#cb155-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-35"><a href="#cb155-35" aria-hidden="true" tabindex="-1"></a><span class="co">#Fix abbreviated country names :</span></span>
<span id="cb155-36"><a href="#cb155-36" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(small_matrix)[<span class="fu">colnames</span>(small_matrix) <span class="sc">==</span> <span class="st">"Usa"</span>] <span class="ot">&lt;-</span> <span class="st">"USA"</span></span>
<span id="cb155-37"><a href="#cb155-37" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(small_matrix)[<span class="fu">rownames</span>(small_matrix) <span class="sc">==</span> <span class="st">"Usa"</span>] <span class="ot">&lt;-</span> <span class="st">"USA"</span></span>
<span id="cb155-38"><a href="#cb155-38" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(small_matrix)[<span class="fu">colnames</span>(small_matrix) <span class="sc">==</span> <span class="st">"United Kindom"</span>] <span class="ot">&lt;-</span> <span class="st">"United Kingdom"</span></span>
<span id="cb155-39"><a href="#cb155-39" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(small_matrix)[<span class="fu">rownames</span>(small_matrix) <span class="sc">==</span> <span class="st">"United Kindom"</span>] <span class="ot">&lt;-</span> <span class="st">"United Kingdom"</span></span>
<span id="cb155-40"><a href="#cb155-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-41"><a href="#cb155-41" aria-hidden="true" tabindex="-1"></a><span class="do">## plot country collaborations as a chord diagram for top 10 countries (using circlize package) (colors are generated randomly each time)</span></span>
<span id="cb155-42"><a href="#cb155-42" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.chord_country_coauthorship.pdf"</span>), <span class="at">width=</span><span class="dv">6</span>, <span class="at">height=</span><span class="dv">6</span>, <span class="at">pointsize=</span><span class="dv">14</span>)</span>
<span id="cb155-43"><a href="#cb155-43" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb155-44"><a href="#cb155-44" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">oma=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb155-45"><a href="#cb155-45" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.clear</span>()</span>
<span id="cb155-46"><a href="#cb155-46" aria-hidden="true" tabindex="-1"></a><span class="fu">circos.par</span>(<span class="at">canvas.xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.8</span>, <span class="fl">0.8</span>), <span class="at">canvas.ylim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">1.0</span>, <span class="fl">1.0</span>)) <span class="co">#set margins</span></span>
<span id="cb155-47"><a href="#cb155-47" aria-hidden="true" tabindex="-1"></a><span class="fu">chordDiagramFromMatrix</span>(small_matrix) <span class="co">#actual plotting</span></span>
<span id="cb155-48"><a href="#cb155-48" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-7---journal-impact-factor-popularity-regionality." class="level4">
<h4 class="anchored" data-anchor-id="figure-7---journal-impact-factor-popularity-regionality.">Figure 7 - journal impact factor, popularity, regionality.</h4>
<p>Plot of journal Impact Factors.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a>dat_articles_IF <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_IF.csv"</span>))</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a><span class="co">#get records for journals with IF &gt; 15</span></span>
<span id="cb156-4"><a href="#cb156-4" aria-hidden="true" tabindex="-1"></a>dat_articles_IF <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Journal_IF)) <span class="sc">%&gt;%</span> <span class="fu">select</span>(Journal_name, Journal_IF, Manual.Tags, Title) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Journal_IF <span class="sc">&gt;</span> <span class="dv">15</span>) <span class="ot">-&gt;</span> dat_articles_IF_over15</span>
<span id="cb156-5"><a href="#cb156-5" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_articles_IF_over15)[<span class="dv">1</span>] <span class="co">#162</span></span>
<span id="cb156-6"><a href="#cb156-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_articles_IF_over15)[<span class="dv">1</span>]<span class="sc">/</span><span class="fu">dim</span>(dat_articles_IF)[<span class="dv">1</span>]<span class="sc">*</span><span class="dv">100</span> <span class="co"># 162 is 5% of articles with IF are in journals with IF &gt;10</span></span>
<span id="cb156-7"><a href="#cb156-7" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_over15 <span class="sc">%&gt;%</span> <span class="fu">add_count</span>(Journal_name, <span class="at">name =</span> <span class="st">"Count"</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) <span class="ot">-&gt;</span> dat_articles_IF_over15 <span class="co">#counts added as a column to records</span></span>
<span id="cb156-8"><a href="#cb156-8" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_over15 <span class="sc">%&gt;%</span> <span class="fu">distinct</span>(Journal_name, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) <span class="ot">-&gt;</span> dat_articles_IF_over15_journals <span class="co">#counts as new data frame of journals</span></span>
<span id="cb156-9"><a href="#cb156-9" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_articles_IF_over15_journals)[<span class="dv">1</span>] <span class="co">#31 journals with &gt;15 IF</span></span>
<span id="cb156-10"><a href="#cb156-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-11"><a href="#cb156-11" aria-hidden="true" tabindex="-1"></a><span class="co">#plot as counts with gradient fill for IF</span></span>
<span id="cb156-12"><a href="#cb156-12" aria-hidden="true" tabindex="-1"></a>figure7A <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat_articles_IF_over15_journals, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Journal_name, Count), <span class="at">y =</span> Count, <span class="at">fill =</span> Journal_IF)) <span class="sc">+</span> <span class="co">#, fill = "grey"</span></span>
<span id="cb156-13"><a href="#cb156-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb156-14"><a href="#cb156-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>, <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb156-15"><a href="#cb156-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb156-16"><a href="#cb156-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb156-17"><a href="#cb156-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) <span class="sc">+</span></span>
<span id="cb156-18"><a href="#cb156-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">name =</span> <span class="st">'journal impact factor &gt;15:'</span>, <span class="at">low =</span> <span class="st">"lightblue"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot of most common journals.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>dat_articles_IF <span class="sc">%&gt;%</span> </span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Journal_name) <span class="sc">%&gt;%</span> <span class="fu">slice_max</span>(<span class="at">order_by =</span> n, <span class="at">n =</span> <span class="dv">15</span>) <span class="ot">-&gt;</span>  count_Journal_name <span class="co">#get top10 most common journal names</span></span>
<span id="cb157-3"><a href="#cb157-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-4"><a href="#cb157-4" aria-hidden="true" tabindex="-1"></a>figure7B <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_Journal_name, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Journal_name, n), <span class="at">y =</span> n)) <span class="sc">+</span> </span>
<span id="cb157-5"><a href="#cb157-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span></span>
<span id="cb157-6"><a href="#cb157-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb157-7"><a href="#cb157-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>))<span class="sc">+</span></span>
<span id="cb157-8"><a href="#cb157-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb157-9"><a href="#cb157-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) <span class="sc">+</span></span>
<span id="cb157-10"><a href="#cb157-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">name =</span> <span class="st">'journal impact factor:'</span>, <span class="at">low =</span> <span class="st">"lightblue"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot of most common journal regions/country and quartile.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_SJR <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_IF_SJR.csv"</span>))</span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_SJR<span class="sc">$</span>SJR_Best_Quartile <span class="ot">&lt;-</span> dat_articles_IF_SJR<span class="sc">$</span><span class="st">"SJR Best Quartile"</span></span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_SJR<span class="sc">$</span>SJR_Best_Quartile[dat_articles_IF_SJR<span class="sc">$</span>SJR_Best_Quartile <span class="sc">==</span> <span class="st">"-"</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb158-4"><a href="#cb158-4" aria-hidden="true" tabindex="-1"></a>dat_articles_IF_SJR<span class="sc">$</span>Regions <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(dat_articles_IF_SJR<span class="sc">$</span>Region)</span>
<span id="cb158-5"><a href="#cb158-5" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(dat_articles_IF_SJR<span class="sc">$</span>Regions) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Other"</span>, <span class="st">"Other"</span>, <span class="st">"Other"</span>, <span class="st">"Other"</span>, <span class="st">"Other"</span>, <span class="st">"Northern America"</span>, <span class="st">"Pacific Region"</span>, <span class="st">"Western Europe"</span>)</span>
<span id="cb158-6"><a href="#cb158-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-7"><a href="#cb158-7" aria-hidden="true" tabindex="-1"></a><span class="co">#using package ggmosaic</span></span>
<span id="cb158-8"><a href="#cb158-8" aria-hidden="true" tabindex="-1"></a>figure7C <span class="ot">&lt;-</span> dat_articles_IF_SJR <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(SJR_Best_Quartile)) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb158-9"><a href="#cb158-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_mosaic</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">product</span>(Regions, SJR_Best_Quartile), <span class="at">fill =</span> Regions)) <span class="sc">+</span></span>
<span id="cb158-10"><a href="#cb158-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_mosaic</span>() <span class="sc">+</span> </span>
<span id="cb158-11"><a href="#cb158-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"SJR journal best Quartile"</span>) <span class="sc">+</span></span>
<span id="cb158-12"><a href="#cb158-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"journal Region"</span>) <span class="sc">+</span></span>
<span id="cb158-13"><a href="#cb158-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Assemble into a multipanel figure and save as pdf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assemble the panels</span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>figure7 <span class="ot">&lt;-</span> figure7A <span class="sc">/</span> figure7B <span class="sc">/</span> figure7C <span class="sc">+</span></span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>) <span class="sc">+</span> </span>
<span id="cb159-5"><a href="#cb159-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">theme =</span> <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>()))</span>
<span id="cb159-6"><a href="#cb159-6" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure </span></span>
<span id="cb159-7"><a href="#cb159-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> figure7, <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.journal_characteristics.pdf"</span>), <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">18</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">scale =</span> <span class="fl">1.6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-8---open-access-status." class="level4">
<h4 class="anchored" data-anchor-id="figure-8---open-access-status.">Figure 8 - open access status.</h4>
<p>Plot article and journals Open Access status info.<br>
Note: Meta-data available on https://unpaywall.org/data-format.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles_OA &lt;- readr::read_csv(file = here("data","OA_dat_articles_refs.csv")) #not working</span></span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a><span class="co">#dat_articles_OA &lt;- readr::read_csv(file = here("data","dat_articles_OA_IF.csv"))</span></span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(dat_articles_OA)</span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dat_articles_OA)</span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">is.na</span>(dat_articles_OA<span class="sc">$</span>is_oa)) <span class="co">#no information for 640 articles</span></span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a><span class="co">#create new data frame of counts</span></span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a>count_dat_articles_OA <span class="ot">&lt;-</span> <span class="fu">count</span>(dat_articles_OA, is_oa, oa_status, <span class="at">name =</span> <span class="st">"Count"</span>, <span class="at">sort =</span> <span class="cn">TRUE</span>) <span class="co">#%&gt;% mutate(oa_status = str_replace(oa_status, "NA", "no information"))</span></span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a><span class="co">#count_dat_articles_OA$is_oa &lt;- factor(as.character(count_dat_articles_OA$is_oa))</span></span>
<span id="cb160-10"><a href="#cb160-10" aria-hidden="true" tabindex="-1"></a>count_dat_articles_OA<span class="sc">$</span>is_oa[<span class="fu">is.na</span>(count_dat_articles_OA<span class="sc">$</span>is_oa)] <span class="ot">&lt;-</span> <span class="st">"no information"</span></span>
<span id="cb160-11"><a href="#cb160-11" aria-hidden="true" tabindex="-1"></a>count_dat_articles_OA<span class="sc">$</span>oa_status[<span class="fu">is.na</span>(count_dat_articles_OA<span class="sc">$</span>oa_status)] <span class="ot">&lt;-</span> <span class="st">"no information"</span></span>
<span id="cb160-12"><a href="#cb160-12" aria-hidden="true" tabindex="-1"></a>count_dat_articles_OA<span class="sc">$</span>is_oa <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(count_dat_articles_OA<span class="sc">$</span>is_oa)</span>
<span id="cb160-13"><a href="#cb160-13" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(count_dat_articles_OA<span class="sc">$</span>is_oa) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"closed"</span>, <span class="st">"no information"</span>, <span class="st">"open access"</span>)</span>
<span id="cb160-14"><a href="#cb160-14" aria-hidden="true" tabindex="-1"></a>count_dat_articles_OA<span class="sc">$</span>oa_status <span class="ot">&lt;-</span> <span class="fu">factor</span>(count_dat_articles_OA<span class="sc">$</span>oa_status, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"no information"</span>, <span class="st">"closed"</span>, <span class="st">"hybrid"</span>, <span class="st">"gold"</span>, <span class="st">"bronze"</span>, <span class="st">"green"</span>))</span>
<span id="cb160-15"><a href="#cb160-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-16"><a href="#cb160-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot as counts with Custom Color Pallette (color-blind friendly) and labels</span></span>
<span id="cb160-17"><a href="#cb160-17" aria-hidden="true" tabindex="-1"></a>Colors_OA <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">"no information"</span> <span class="ot">=</span> <span class="st">"#999999"</span>, <span class="st">"closed"</span> <span class="ot">=</span> <span class="st">"#000000"</span>, <span class="st">"gold"</span> <span class="ot">=</span> <span class="st">"#E69F00"</span>, <span class="st">"green"</span> <span class="ot">=</span> <span class="st">"#009E73"</span>, <span class="st">"bronze"</span> <span class="ot">=</span> <span class="st">"#D55E00"</span>, <span class="st">"hybrid"</span> <span class="ot">=</span> <span class="st">"#56B4E9"</span>)</span>
<span id="cb160-18"><a href="#cb160-18" aria-hidden="true" tabindex="-1"></a>figure8A <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_dat_articles_OA, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(is_oa, Count), <span class="at">y =</span> Count, <span class="at">fill =</span> oa_status)) <span class="sc">+</span> </span>
<span id="cb160-19"><a href="#cb160-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb160-20"><a href="#cb160-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb160-21"><a href="#cb160-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>)  <span class="sc">+</span></span>
<span id="cb160-22"><a href="#cb160-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> Colors_OA, <span class="at">name =</span> <span class="st">'open access type:'</span>, <span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">'green'</span>, <span class="st">'bronze'</span>, <span class="st">'gold'</span>,  <span class="st">'hybrid'</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Green = unpublished version freely available via a repository"</span>, <span class="st">"Bronze = published version is free to read only on the publisher's website"</span>, <span class="st">"Gold = author pays for a free-to-read version to be available from the fully-OA journal"</span>, <span class="st">"Hybrid = author pays for a free-to-read version to be available from the subscription journal"</span>)) <span class="sc">+</span></span>
<span id="cb160-23"><a href="#cb160-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> Count), <span class="at">color=</span><span class="st">"white"</span>, <span class="at">size =</span> <span class="dv">3</span>, <span class="at">position =</span> <span class="fu">position_stack</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb160-24"><a href="#cb160-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>, <span class="at">legend.direction =</span> <span class="st">"vertical"</span>) <span class="sc">+</span></span>
<span id="cb160-25"><a href="#cb160-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"article count"</span>) </span>
<span id="cb160-26"><a href="#cb160-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-27"><a href="#cb160-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Note: this plot matches this table: </span></span>
<span id="cb160-28"><a href="#cb160-28" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_OA$oa_status, dat_articles_OA$is_oa, useNA = "always") #OA status of articles</span></span>
<span id="cb160-29"><a href="#cb160-29" aria-hidden="true" tabindex="-1"></a><span class="co"># table(dat_articles_OA$is_oa)/sum(table(dat_articles_OA$is_oa)) #percentages for OA TRUE or FALSE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot of OA status - counts across publication years.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>dat_articles_OA <span class="sc">%&gt;%</span> </span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(Publication.Year, is_oa) <span class="sc">%&gt;%</span> <span class="fu">drop_na</span>() <span class="ot">-&gt;</span> count_year_oa</span>
<span id="cb161-3"><a href="#cb161-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-4"><a href="#cb161-4" aria-hidden="true" tabindex="-1"></a><span class="co">#assign custom colors</span></span>
<span id="cb161-5"><a href="#cb161-5" aria-hidden="true" tabindex="-1"></a>Colors_OA2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#f5d4fc"</span>, <span class="st">"#000000"</span>)</span>
<span id="cb161-6"><a href="#cb161-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-7"><a href="#cb161-7" aria-hidden="true" tabindex="-1"></a><span class="co">#plot change of proportion oa across years</span></span>
<span id="cb161-8"><a href="#cb161-8" aria-hidden="true" tabindex="-1"></a>figure8B <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(count_year_oa, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.character</span>(Publication.Year), <span class="at">y =</span> n, <span class="at">fill =</span> <span class="fu">reorder</span>(is_oa, n))) <span class="sc">+</span> </span>
<span id="cb161-9"><a href="#cb161-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>) <span class="sc">+</span> <span class="co">#, position = "dodge"</span></span>
<span id="cb161-10"><a href="#cb161-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb161-11"><a href="#cb161-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">rev</span>(Colors_OA2), <span class="at">name =</span> <span class="st">""</span>, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"closed access"</span>, <span class="st">"open access"</span>)) <span class="sc">+</span> </span>
<span id="cb161-12"><a href="#cb161-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"proportion"</span>) <span class="sc">+</span></span>
<span id="cb161-13"><a href="#cb161-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">legend_bottom</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Assemble into a multipanel figure and save as pdf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assemble the panels</span></span>
<span id="cb162-2"><a href="#cb162-2" aria-hidden="true" tabindex="-1"></a>figure8 <span class="ot">&lt;-</span> figure8A <span class="sc">/</span> figure8B <span class="sc">+</span></span>
<span id="cb162-3"><a href="#cb162-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb162-4"><a href="#cb162-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb162-5"><a href="#cb162-5" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure </span></span>
<span id="cb162-6"><a href="#cb162-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> figure8, <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.oa_status.pdf"</span>), <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">12</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">scale =</span> <span class="fl">1.6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-9---article-impact-via-citations-and-altmetrics." class="level4">
<h4 class="anchored" data-anchor-id="figure-9---article-impact-via-citations-and-altmetrics.">Figure 9 - article impact via citations and altmetrics.</h4>
<p>Plot of academic citations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a>dat_articles_CR <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"dat_articles_CR.csv"</span>))  <span class="co">#load meged data frame</span></span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a>dat_articles_CR_highlycited <span class="ot">&lt;-</span> dat_articles_CR <span class="sc">%&gt;%</span> <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">500</span>)</span>
<span id="cb163-3"><a href="#cb163-3" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(dat_articles_CR_highlycited) #4 articles with &gt;500 citations</span></span>
<span id="cb163-4"><a href="#cb163-4" aria-hidden="true" tabindex="-1"></a>pos <span class="ot">&lt;-</span> <span class="fu">position_jitter</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">height =</span> <span class="fl">0.1</span>, <span class="at">seed =</span> <span class="dv">1</span>) <span class="co">#used below to add jitter</span></span>
<span id="cb163-5"><a href="#cb163-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-6"><a href="#cb163-6" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb163-7"><a href="#cb163-7" aria-hidden="true" tabindex="-1"></a>figure9A <span class="ot">&lt;-</span> dat_articles_CR <span class="sc">%&gt;%</span></span>
<span id="cb163-8"><a href="#cb163-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(count)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have no available citation counts</span></span>
<span id="cb163-9"><a href="#cb163-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Publication.Year <span class="sc">&gt;</span> <span class="dv">2007</span>) <span class="sc">%&gt;%</span> <span class="co">#remove data points from before 2008</span></span>
<span id="cb163-10"><a href="#cb163-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Publication.Year <span class="sc">&lt;</span> <span class="dv">2023</span>) <span class="sc">%&gt;%</span> <span class="co">#remove data points after 2022</span></span>
<span id="cb163-11"><a href="#cb163-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Publication.Year, <span class="at">y =</span> count)) <span class="sc">+</span> </span>
<span id="cb163-12"><a href="#cb163-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">position =</span> pos, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span></span>
<span id="cb163-13"><a href="#cb163-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_smooth(method = "lm", color = "grey") +</span></span>
<span id="cb163-14"><a href="#cb163-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()  <span class="sc">+</span></span>
<span id="cb163-15"><a href="#cb163-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"total citation count"</span>) <span class="sc">+</span></span>
<span id="cb163-16"><a href="#cb163-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"year"</span>) <span class="sc">+</span></span>
<span id="cb163-17"><a href="#cb163-17" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> dat_articles_CR_highlycited, <span class="fu">aes</span>(<span class="at">label =</span> Publication.Title,), <span class="at">size =</span> <span class="dv">4</span>, <span class="at">max.overlaps =</span> <span class="dv">25</span>, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span> <span class="co">#add journal names as labels for highly cited publications only </span></span>
<span id="cb163-18"><a href="#cb163-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot of altmetric scores.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_csv</span>(<span class="at">file =</span> <span class="fu">here</span>(<span class="st">"data"</span>,<span class="st">"Altmetrics.results_df.csv"</span>))</span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Checking and processing to ore useful formats</span></span>
<span id="cb164-3"><a href="#cb164-3" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Altmetrics.results_df<span class="sc">$</span>score) <span class="co">#make numeric</span></span>
<span id="cb164-4"><a href="#cb164-4" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>published_date <span class="ot">&lt;-</span> Altmetrics.results_df<span class="sc">$</span>published_on <span class="co">#add a new column </span></span>
<span id="cb164-5"><a href="#cb164-5" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>published_date <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">"NA        "</span>, <span class="cn">NA</span>, Altmetrics.results_df<span class="sc">$</span>published_date)) <span class="co">#change "NA        " to NA and convert to numeric</span></span>
<span id="cb164-6"><a href="#cb164-6" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>published_date <span class="ot">&lt;-</span> <span class="fu">as.POSIXct</span>(Altmetrics.results_df<span class="sc">$</span>published_date, <span class="at">origin =</span> <span class="st">"1970-01-01"</span>) <span class="co">#convert POSIXct</span></span>
<span id="cb164-7"><a href="#cb164-7" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df<span class="sc">$</span>year <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">format</span>(Altmetrics.results_df<span class="sc">$</span>published_date, <span class="at">format=</span><span class="st">"%Y"</span>)) </span>
<span id="cb164-8"><a href="#cb164-8" aria-hidden="true" tabindex="-1"></a><span class="co">#match and merge data to dat_articles by DOI</span></span>
<span id="cb164-9"><a href="#cb164-9" aria-hidden="true" tabindex="-1"></a>dat_articles_A <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dat_articles, Altmetrics.results_df, <span class="at">by =</span> <span class="st">"doi"</span>) </span>
<span id="cb164-10"><a href="#cb164-10" aria-hidden="true" tabindex="-1"></a><span class="co">#subset highly scoring (&gt;100) publications</span></span>
<span id="cb164-11"><a href="#cb164-11" aria-hidden="true" tabindex="-1"></a>Altmetrics.results_df_highlyscored <span class="ot">&lt;-</span> Altmetrics.results_df <span class="sc">%&gt;%</span></span>
<span id="cb164-12"><a href="#cb164-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(score <span class="sc">&gt;</span> <span class="dv">1000</span>)</span>
<span id="cb164-13"><a href="#cb164-13" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(Altmetrics.results_df_highlyscored) #8 articles with &gt;1000 score</span></span>
<span id="cb164-14"><a href="#cb164-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-15"><a href="#cb164-15" aria-hidden="true" tabindex="-1"></a><span class="co">#plot</span></span>
<span id="cb164-16"><a href="#cb164-16" aria-hidden="true" tabindex="-1"></a>figure9B <span class="ot">&lt;-</span> Altmetrics.results_df <span class="sc">%&gt;%</span> </span>
<span id="cb164-17"><a href="#cb164-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#replace(is.na(score), 0) %&gt;% #showing articles with no attention as 0</span></span>
<span id="cb164-18"><a href="#cb164-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(score)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have 0 scores</span></span>
<span id="cb164-19"><a href="#cb164-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(year)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have 0 scores</span></span>
<span id="cb164-20"><a href="#cb164-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(year <span class="sc">&gt;</span> <span class="dv">2007</span>) <span class="sc">%&gt;%</span> <span class="co">#remove data points from before 2008</span></span>
<span id="cb164-21"><a href="#cb164-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(year <span class="sc">&lt;</span> <span class="dv">2023</span>) <span class="sc">%&gt;%</span> <span class="co">#remove data points after 2022</span></span>
<span id="cb164-22"><a href="#cb164-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> year, <span class="at">y =</span> score)) <span class="sc">+</span> </span>
<span id="cb164-23"><a href="#cb164-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"#66c2a5"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">position =</span> pos) <span class="sc">+</span></span>
<span id="cb164-24"><a href="#cb164-24" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_smooth(method = "lm", color = "grey") + </span></span>
<span id="cb164-25"><a href="#cb164-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()  <span class="sc">+</span></span>
<span id="cb164-26"><a href="#cb164-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"total Altmetric score"</span>) <span class="sc">+</span></span>
<span id="cb164-27"><a href="#cb164-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"year"</span>) <span class="sc">+</span></span>
<span id="cb164-28"><a href="#cb164-28" aria-hidden="true" tabindex="-1"></a>  ggrepel<span class="sc">::</span><span class="fu">geom_text_repel</span>(<span class="at">data =</span> Altmetrics.results_df_highlyscored, <span class="fu">aes</span>(<span class="at">label =</span> journal), <span class="at">size =</span> <span class="dv">4</span>, <span class="at">max.overlaps =</span> <span class="dv">20</span>, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Plot of altmetric scores in context as journal percentile.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a>figure9C <span class="ot">&lt;-</span> Altmetrics.results_df <span class="sc">%&gt;%</span> </span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#replace(is.na(context.journal.pct), 0) %&gt;% #showing articles with no attention as 0</span></span>
<span id="cb165-3"><a href="#cb165-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(context.journal.pct)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have 0 scores</span></span>
<span id="cb165-4"><a href="#cb165-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(year)) <span class="sc">%&gt;%</span> <span class="co">#without articles that have 0 scores</span></span>
<span id="cb165-5"><a href="#cb165-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(year <span class="sc">&gt;</span> <span class="dv">2007</span>) <span class="sc">%&gt;%</span> <span class="co">#remove data points from before 2008</span></span>
<span id="cb165-6"><a href="#cb165-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(year <span class="sc">&lt;</span> <span class="dv">2023</span>) <span class="sc">%&gt;%</span> <span class="co">#remove data points after 2022</span></span>
<span id="cb165-7"><a href="#cb165-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> year, <span class="at">y =</span> <span class="fu">as.numeric</span>(context.journal.pct))) <span class="sc">+</span></span>
<span id="cb165-8"><a href="#cb165-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">position =</span> pos, <span class="at">color =</span> <span class="st">"#66c2a5"</span>) <span class="sc">+</span></span>
<span id="cb165-9"><a href="#cb165-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#geom_smooth(method = "lm", color = "grey") + </span></span>
<span id="cb165-10"><a href="#cb165-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb165-11"><a href="#cb165-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"percentile within journal context"</span>) <span class="sc">+</span></span>
<span id="cb165-12"><a href="#cb165-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"year"</span>) <span class="sc">+</span> </span>
<span id="cb165-13"><a href="#cb165-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">11</span>), <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Assemble into a multipanel figure and save as pdf.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a><span class="co">#assemble the panels</span></span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a>figure9 <span class="ot">&lt;-</span> figure9A <span class="sc">/</span> figure9B <span class="sc">/</span> figure9C <span class="sc">+</span></span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_layout</span>(<span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb166-5"><a href="#cb166-5" aria-hidden="true" tabindex="-1"></a><span class="co">#save the figure </span></span>
<span id="cb166-6"><a href="#cb166-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">plot =</span> figure9, <span class="fu">here</span>(<span class="st">"plots"</span>, <span class="st">"plot.article_impact.pdf"</span>), <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">16</span>, <span class="at">units =</span> <span class="st">"cm"</span>, <span class="at">dpi =</span> <span class="dv">300</span>, <span class="at">scale =</span> <span class="fl">1.6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>